<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.141">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bacterial filamentation: a bet for survival in stressful environments - 1&nbsp; Image processing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./02-experiment-analysis.html" rel="next">
<link href="./00-introduction.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bacterial filamentation: a bet for survival in stressful environments</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/jvelezmagic/bacterial-filamentation-research" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="./Bacterial-filamentation--a-bet-for-survival-in-stressful-environments.pdf" title="Download PDF" class="sidebar-tool px-1"><i class="bi bi-file-pdf"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
            <i class="bi bi-bi-linkedin pe-1"></i>
          LinkedIn
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00-introduction.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-image-processing.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-experiment-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-model-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Models to the rescue; filamentation abstraction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-discussion.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./98-appendices.html" class="sidebar-item-text sidebar-link">Appendix</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="toc-section-number">1.1</span>  Introduction</a></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="toc-section-number">1.2</span>  Preprocessing</a></li>
  <li><a href="#segmentation" id="toc-segmentation" class="nav-link" data-scroll-target="#segmentation"><span class="toc-section-number">1.3</span>  Segmentation</a></li>
  <li><a href="#tracking" id="toc-tracking" class="nav-link" data-scroll-target="#tracking"><span class="toc-section-number">1.4</span>  Tracking</a></li>
  <li><a href="#sec-manual-corrections" id="toc-sec-manual-corrections" class="nav-link" data-scroll-target="#sec-manual-corrections"><span class="toc-section-number">1.5</span>  Manual corrections</a></li>
  <li><a href="#data-extraction" id="toc-data-extraction" class="nav-link" data-scroll-target="#data-extraction"><span class="toc-section-number">1.6</span>  Data extraction</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/blob/main/01-image-processing.qmd" class="toc-action">View source</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/edit/main/01-image-processing.qmd" class="toc-action">Edit this page</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span id="sec-image-processing" class="quarto-section-identifier d-none d-lg-block"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
  </div>

<div class="quarto-title-meta">

      
    
  </div>
  

</header>

<section id="introduction" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1.1</span> Introduction</h2>
<p>With the progress of technology, optical and fluorescence microscopy has become a fundamental tool for the characterization and understanding of the bacterial world. Microscopy has allowed humanity to extend its senses to observe the unknown world with exciting new perspectives that they might never otherwise have envisioned. Furthermore, microscopy offers a clear advantage over other techniques used to characterize bacteria since it can acquire data from living cells in spatial resolution <span class="citation" data-cites="schermelleh2019">(<a href="references.html#ref-schermelleh2019" role="doc-biblioref">Schermelleh et al. 2019</a>)</span>.</p>
<p>Including the discovery of fluorescent proteins (<em>e.g.</em>, GFP and DsRed) and improvements in fluorescent reporters, it is possible to specifically label specific cellular components and track cellular functions <span class="citation" data-cites="Specht2017">(<a href="references.html#ref-Specht2017" role="doc-biblioref">Specht, Braselmann, and Palmer 2017</a>)</span>. On the other hand, mechanical and intellectual development of microfluidic research techniques provides an excellent opportunity to overcome bio-medical and chemical techniques <span class="citation" data-cites="convery2019">(<a href="references.html#ref-convery2019" role="doc-biblioref">Convery and Gadegaard 2019</a>)</span>. Collectively, it is possible to study communities of bacteria at the level of individual cells <span class="citation" data-cites="balaban2004a elowitz2002">(<a href="references.html#ref-balaban2004a" role="doc-biblioref">Balaban et al. 2004</a>; <a href="references.html#ref-elowitz2002" role="doc-biblioref">Elowitz et al. 2002</a>)</span>.</p>
<p>Although all this technological development has provided a significant advance for the scientific community, after acquiring fluorescence images, the extraction of quantitative properties from these images is crucial, but unfortunately, a difficult step for analyzing experiments. Not so long ago, image analysis in biology relied on manual quantification. However, manual analysis suffers from two main problems: 1) accuracy and 2) scalability (that is, analyzing miles or more images). Fortunately, improvements in image accuracy and computational image analysis capabilities are revolutionizing the quantification of biological processes through <span class="citation" data-cites="Caicedo2017 Smith2018">(<a href="references.html#ref-Caicedo2017" role="doc-biblioref">Caicedo et al. 2017</a>; <a href="references.html#ref-Smith2018" role="doc-biblioref">Smith et al. 2018</a>)</span>. Therefore, the manual correction required to analyze the experiments is minimal.</p>
<p>Here, we used a series of programs in <span class="math inline">\(\mu \mathrm{J}\)</span> (<a href="https://github.com/ccg-esb-lab/uJ" class="uri">https://github.com/ccg-esb-lab/uJ</a>), which consists of an <span class="math inline">\(\mathrm{ImageJ}\)</span> macro library (mainly) for quantifying unicellular bacterial dynamics in microfluidic devices <span class="citation" data-cites="schneider2012">(<a href="references.html#ref-schneider2012" role="doc-biblioref">Schneider, Rasband, and Eliceiri 2012</a>)</span>. The specific steps used are described below and are summarized in Figure <span class="quarto-unresolved-ref">?fig-something</span>.</p>
</section>
<section id="preprocessing" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="preprocessing"><span class="header-section-number">1.2</span> Preprocessing</h2>
<p>We exported the figures obtained by the NIS-Elements software (RRID:SCR_014329) from the microfluidics experiments in TIFF (Tagged Image File Format) format. Each figure was named as follows: <em>experimentxyc1t001</em> where <em>experiment</em> indicates the name assigned to the experiment, <em>xy</em> the trap number, <em>c</em> the fluorescence channel, and <em>t</em> the passage of time.</p>
<p>Subsequently, we compile the images, rename them and save them as images in different folders. We maintained the classification by fluorescence channels and phase contrast, and within the channel folder, it is the sub-classification by trap number.</p>
</section>
<section id="segmentation" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="segmentation"><span class="header-section-number">1.3</span> Segmentation</h2>
<p>To determine which parts of the photographs correspond to cells, we carry out an image segmentation analysis. Segmentation consists of classification at the pixel level, which allows us to define the pixels that give identity to the limit of a cell, its interior, and the image’s background (everything that is not a cell). A new image is generated from the above, known as the segmentation mask, containing only the pixels that identify cells.</p>
<p>To build the segmentation mask, we used <em>Deepcell</em> <span class="citation" data-cites="vanvalen2016">(<a href="references.html#ref-vanvalen2016" role="doc-biblioref">Van Valen et al. 2016</a>)</span>. <em>Deepcell</em> is a network trained with a robust set of images that people previously classified as cells. However, the generation of the segmentation masks is not absolved of errors (see also <a href="#sec-manual-corrections"><span>Section&nbsp;1.5</span></a>). Sometimes we must correct them manually due to 1) mistakenly identifying two or more cells as one, 2) identifying two or more cells when it is only one cell, and 3) failing to identify a cell.</p>
</section>
<section id="tracking" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="tracking"><span class="header-section-number">1.4</span> Tracking</h2>
<p>From the image segmentation, we obtain ROI files (region of interest), which contain coordinates of the position of individual cells in each photograph <span class="citation" data-cites="10.5555/1386553">(<a href="references.html#ref-10.5555/1386553" role="doc-biblioref">Brinkmann 2008</a>)</span>. Tracking is the tracking of a region of interest in a consecutive series of images. In this case, the tracking generates the identification of the lineages, that is, the ancestry of each cell.</p>
<p>We read the ROI files in Python through the <em>shapely</em> package, which efficiently reconstructs polygons, thus calculating the length of the cells <span class="citation" data-cites="10.5555/1593511 shapely2007">(<a href="references.html#ref-10.5555/1593511" role="doc-biblioref">Van Rossum and Drake 2009</a>; <a href="references.html#ref-shapely2007" role="doc-biblioref">Gillies et al. 2007--</a>)</span>. Also, in Python using ROI files, we track cells with the k-nearest neighbors algorithm that uses various properties such as fluorescence intensity, length, and shape of each cell, to identify cell lineages <span class="citation" data-cites="altman1992">(<a href="references.html#ref-altman1992" role="doc-biblioref">Altman 1992</a>)</span>.</p>
</section>
<section id="sec-manual-corrections" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="sec-manual-corrections"><span class="header-section-number">1.5</span> Manual corrections</h2>
<p>For cell-tracking manual correction, we used <em>Napari,</em> an open-source python-based tool designed to explore, annotate, and analyze large multidimensional images <span class="citation" data-cites="sofroniew2021a">(<a href="references.html#ref-sofroniew2021a" role="doc-biblioref">Sofroniew et al. 2021</a>)</span>. Our custom cell-viewer allows us to easy lineage data visualization, custom-plotting, and lineage-correction. Code for our cell-viewer is available on <a href="https://github.com/ccg-esb-lab/uJ/tree/master/single-channel" class="uri">https://github.com/ccg-esb-lab/uJ/tree/master/single-channel</a>.</p>
<p>We produced high-throughput data of thousands of cells with a single-cell resolution to the end of the lineages manual reconstruction. We obtained data about time-series of fluorescent intensity, morphological properties of individual cells (<em>e.g.</em>, elongation, duplication rate), and time-resolved population-level statistics (<em>e.g.</em>, probability of survival to the antibiotic shock).</p>
</section>
<section id="data-extraction" class="level2" data-number="1.6">
<h2 data-number="1.6" class="anchored" data-anchor-id="data-extraction"><span class="header-section-number">1.6</span> Data extraction</h2>
<p>We construct a file in columnar format through image processing that contains the information necessary to analyze each experiment (<em>i.e.</em>, chromosomal and plasmids) in its different traps (<em>i.e.</em>, XY identifier). See <a href="#tbl-data-columns-specifications">Table&nbsp;<span>1.1</span></a> for a full description of the output data. Subsequently, the table was analyzed in R for statistical computation and plotting (see <a href="02-experiment-analysis.html"><span>Chapter&nbsp;2</span></a>) <span class="citation" data-cites="R-base">(<a href="references.html#ref-R-base" role="doc-biblioref">R Core Team 2022</a>)</span>.</p>
<div id="tbl-data-columns-specifications" class="anchored">
<table class="table">
<caption>Table&nbsp;1.1: Resulting table from image processing. </caption>
<colgroup>
<col style="width: 29%">
<col style="width: 70%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Column</th>
<th style="text-align: right;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">experimentID</td>
<td style="text-align: right;">Unique identifier of the experiment.</td>
</tr>
<tr class="even">
<td style="text-align: left;">trapID</td>
<td style="text-align: right;">Unique identifier of the trap used.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">lineageID</td>
<td style="text-align: right;">Unique integer of the stem cell and its ancestry.</td>
</tr>
<tr class="even">
<td style="text-align: left;">cellID</td>
<td style="text-align: right;">Unique identification number for each cell existing since the beginning of the experiment or generated later.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">motherID</td>
<td style="text-align: right;">Represents the identification number of the stem cell that gave rise to the progeny.</td>
</tr>
<tr class="even">
<td style="text-align: left;">trackID</td>
<td style="text-align: right;">Indicates the x-y coordinates where the cell being tracked starts.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">roiID</td>
<td style="text-align: right;">Indicates the x-y position in which the cell is located, followed after each photograph.</td>
</tr>
<tr class="even">
<td style="text-align: left;">frame</td>
<td style="text-align: right;">Number of the photograph in the sequence of photographs taken, indicating the elapsed time (10 minutes per frame).</td>
</tr>
<tr class="odd">
<td style="text-align: left;">length</td>
<td style="text-align: right;">Cell length.</td>
</tr>
<tr class="even">
<td style="text-align: left;">division</td>
<td style="text-align: right;">Indicates cell division events, represented by the value 1 when they occur and 0 otherwise.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GFP</td>
<td style="text-align: right;">Represents the relative fluorescence intensity in each cell by green fluorescent protein (<em>i.e.</em>, GFP).</td>
</tr>
<tr class="even">
<td style="text-align: left;">DsRed</td>
<td style="text-align: right;">Represents the relative fluorescence intensity for cells generated by rhodamine’s internalization (<em>i.e.</em>, DsRed); an indicator of cell death events.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tracking_score</td>
<td style="text-align: right;">Determine how good or bad the tracking of a cell was.</td>
</tr>
<tr class="even">
<td style="text-align: left;">state</td>
<td style="text-align: right;">Indicates the state of the cell determined from its length and fluorescence thresholds. -1 for death, 0 for normal, and 1 for filamentation (see <a href="02-experiment-analysis.html#sec-experiment-general-preprocessing"><span>Section&nbsp;2.2</span></a> for detailed information).</td>
</tr>
</tbody>
</table>
</div>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-altman1992" class="csl-entry" role="doc-biblioentry">
Altman, N. S. 1992. <span>“An Introduction to Kernel and Nearest-Neighbor Nonparametric Regression.”</span> <em>The American Statistician</em> 46 (3): 175–85. <a href="https://doi.org/10.1080/00031305.1992.10475879">https://doi.org/10.1080/00031305.1992.10475879</a>.
</div>
<div id="ref-balaban2004a" class="csl-entry" role="doc-biblioentry">
Balaban, Nathalie Q., Jack Merrin, Remy Chait, Lukasz Kowalik, and Stanislas Leibler. 2004. <span>“Bacterial Persistence as a Phenotypic Switch.”</span> <em>Science</em> 305 (5690): 1622–25. <a href="https://doi.org/10.1126/science.1099390">https://doi.org/10.1126/science.1099390</a>.
</div>
<div id="ref-10.5555/1386553" class="csl-entry" role="doc-biblioentry">
Brinkmann, Ron. 2008. <em>The Art and Science of Digital Compositing, Second Edition: Techniques for Visual Effects, Animation and Motion Graphics (the Morgan Kaufmann Series in ... Morgan Kaufmann Series in Computer Graphics)</em>. 2nd ed. San Francisco, CA, USA: Morgan Kaufmann Publishers Inc.
</div>
<div id="ref-Caicedo2017" class="csl-entry" role="doc-biblioentry">
Caicedo, Juan C, Sam Cooper, Florian Heigwer, Scott Warchal, Peng Qiu, Csaba Molnar, Aliaksei S Vasilevich, et al. 2017. <span>“Data-Analysis Strategies for Image-Based Cell Profiling.”</span> <em>Nature Methods</em> 14 (9): 849–63. <a href="https://doi.org/10.1038/nmeth.4397">https://doi.org/10.1038/nmeth.4397</a>.
</div>
<div id="ref-convery2019" class="csl-entry" role="doc-biblioentry">
Convery, Neil, and Nikolaj Gadegaard. 2019. <span>“30 Years of Microfluidics.”</span> <em>Micro and Nano Engineering</em> 2 (March): 76–91. <a href="https://doi.org/10.1016/j.mne.2019.01.003">https://doi.org/10.1016/j.mne.2019.01.003</a>.
</div>
<div id="ref-elowitz2002" class="csl-entry" role="doc-biblioentry">
Elowitz, Michael B., Arnold J. Levine, Eric D. Siggia, and Peter S. Swain. 2002. <span>“Stochastic Gene Expression in a Single Cell.”</span> <em>Science</em> 297 (5584): 1183–86. <a href="https://doi.org/10.1126/science.1070919">https://doi.org/10.1126/science.1070919</a>.
</div>
<div id="ref-shapely2007" class="csl-entry" role="doc-biblioentry">
Gillies, Sean et al. 2007--. <span>“Shapely: Manipulation and Analysis of Geometric Objects.”</span> toblerity.org. <a href="https://github.com/Toblerity/Shapely">https://github.com/Toblerity/Shapely</a>.
</div>
<div id="ref-R-base" class="csl-entry" role="doc-biblioentry">
R Core Team. 2022. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-schermelleh2019" class="csl-entry" role="doc-biblioentry">
Schermelleh, Lothar, Alexia Ferrand, Thomas Huser, Christian Eggeling, Markus Sauer, Oliver Biehlmaier, and Gregor P. C. Drummen. 2019. <span>“Super-Resolution Microscopy Demystified.”</span> <em>Nature Cell Biology</em> 21 (1): 72–84. <a href="https://doi.org/10.1038/s41556-018-0251-8">https://doi.org/10.1038/s41556-018-0251-8</a>.
</div>
<div id="ref-schneider2012" class="csl-entry" role="doc-biblioentry">
Schneider, Caroline A, Wayne S Rasband, and Kevin W Eliceiri. 2012. <span>“NIH Image to ImageJ: 25 Years of Image Analysis.”</span> <em>Nature Methods</em> 9 (7): 671–75. <a href="https://doi.org/10.1038/nmeth.2089">https://doi.org/10.1038/nmeth.2089</a>.
</div>
<div id="ref-Smith2018" class="csl-entry" role="doc-biblioentry">
Smith, Kevin, Filippo Piccinini, Tamas Balassa, Krisztian Koos, Tivadar Danka, Hossein Azizpour, and Peter Horvath. 2018. <span>“Phenotypic Image Analysis Software Tools for Exploring and Understanding Big Image Data from Cell-Based Assays.”</span> <em>Cell Systems</em> 6 (6): 636–53. <a href="https://doi.org/10.1016/j.cels.2018.06.001">https://doi.org/10.1016/j.cels.2018.06.001</a>.
</div>
<div id="ref-sofroniew2021a" class="csl-entry" role="doc-biblioentry">
Sofroniew, Nicholas, Talley Lambert, Kira Evans, Juan Nunez-Iglesias, Grzegorz Bokota, Gonzalo Peña-Castellanos, Philip Winston, et al. 2021. <em>Napari/Napari: 0.4.12rc2</em>. Zenodo. <a href="https://doi.org/10.5281/ZENODO.3555620">https://doi.org/10.5281/ZENODO.3555620</a>.
</div>
<div id="ref-Specht2017" class="csl-entry" role="doc-biblioentry">
Specht, Elizabeth A., Esther Braselmann, and Amy E. Palmer. 2017. <span>“A Critical and Comparative Review of Fluorescent Tools for Live-Cell Imaging.”</span> <em>Annual Review of Physiology</em> 79 (1): 93–117. <a href="https://doi.org/10.1146/annurev-physiol-022516-034055">https://doi.org/10.1146/annurev-physiol-022516-034055</a>.
</div>
<div id="ref-10.5555/1593511" class="csl-entry" role="doc-biblioentry">
Van Rossum, Guido, and Fred L. Drake. 2009. <em>Python 3 Reference Manual</em>. Scotts Valley, CA: CreateSpace.
</div>
<div id="ref-vanvalen2016" class="csl-entry" role="doc-biblioentry">
Van Valen, David A., Takamasa Kudo, Keara M. Lane, Derek N. Macklin, Nicolas T. Quach, Mialy M. DeFelice, Inbal Maayan, Yu Tanouchi, Euan A. Ashley, and Markus W. Covert. 2016. <span>“Deep Learning Automates the Quantitative Analysis of Individual Cells in Live-Cell Imaging Experiments.”</span> Edited by Martin Meier-Schellersheim. <em>PLOS Computational Biology</em> 12 (11): e1005177. <a href="https://doi.org/10.1371/journal.pcbi.1005177">https://doi.org/10.1371/journal.pcbi.1005177</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./00-introduction.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Introduction</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./02-experiment-analysis.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">filters:</span><span class="co"> </span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">    - lightbox</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="an">lightbox:</span><span class="co"> auto</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu"># Image processing {#sec-image-processing}</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">## Introduction</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>With the progress of technology, optical and fluorescence microscopy has</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>become a fundamental tool for the characterization and understanding of</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>the bacterial world. Microscopy has allowed humanity to extend its</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>senses to observe the unknown world with exciting new perspectives that</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>they might never otherwise have envisioned. Furthermore, microscopy</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>offers a clear advantage over other techniques used to characterize</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>bacteria since it can acquire data from living cells in spatial</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>resolution <span class="co">[</span><span class="ot">@schermelleh2019</span><span class="co">]</span>.</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>Including the discovery of fluorescent proteins (*e.g.*, GFP and DsRed)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>and improvements in fluorescent reporters, it is possible to</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>specifically label specific cellular components and track cellular</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>functions <span class="co">[</span><span class="ot">@Specht2017</span><span class="co">]</span>. On the other hand, mechanical and intellectual</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>development of microfluidic research techniques provides an excellent</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>opportunity to overcome bio-medical and chemical techniques</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@convery2019</span><span class="co">]</span>. Collectively, it is possible to study communities of</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>bacteria at the level of individual cells <span class="co">[</span><span class="ot">@balaban2004a; @elowitz2002</span><span class="co">]</span>.</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>Although all this technological development has provided a significant</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>advance for the scientific community, after acquiring fluorescence</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>images, the extraction of quantitative properties from these images is</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>crucial, but unfortunately, a difficult step for analyzing experiments.</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>Not so long ago, image analysis in biology relied on manual</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>quantification. However, manual analysis suffers from two main</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>problems: 1) accuracy and 2) scalability (that is, analyzing miles or</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>more images). Fortunately, improvements in image accuracy and</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>computational image analysis capabilities are revolutionizing the</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>quantification of biological processes through [@Caicedo2017;</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>@Smith2018]. Therefore, the manual correction required to analyze the</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>experiments is minimal.</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>Here, we used a series of programs in $\mu \mathrm{J}$</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>(<span class="ot">&lt;https://github.com/ccg-esb-lab/uJ&gt;</span>), which consists of an</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>$\mathrm{ImageJ}$ macro library (mainly) for quantifying unicellular</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>bacterial dynamics in microfluidic devices <span class="co">[</span><span class="ot">@schneider2012</span><span class="co">]</span>. The</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>specific steps used are described below and are summarized in Figure</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>@fig-something.</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="fu">## Preprocessing</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>We exported the figures obtained by the NIS-Elements software</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>(RRID:SCR_014329) from the microfluidics experiments in TIFF (Tagged</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>Image File Format) format. Each figure was named as follows:</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>*experimentxyc1t001* where *experiment* indicates the name assigned to</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>the experiment, *xy* the trap number, *c* the fluorescence channel, and</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>*t* the passage of time.</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>Subsequently, we compile the images, rename them and save them as images</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>in different folders. We maintained the classification by fluorescence</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>channels and phase contrast, and within the channel folder, it is the</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>sub-classification by trap number.</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="fu">## Segmentation</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>To determine which parts of the photographs correspond to cells, we</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>carry out an image segmentation analysis. Segmentation consists of</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>classification at the pixel level, which allows us to define the pixels</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>that give identity to the limit of a cell, its interior, and the image's</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>background (everything that is not a cell). A new image is generated</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>from the above, known as the segmentation mask, containing only the</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>pixels that identify cells.</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>To build the segmentation mask, we used *Deepcell* <span class="co">[</span><span class="ot">@vanvalen2016</span><span class="co">]</span>.</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>*Deepcell* is a network trained with a robust set of images that people</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>previously classified as cells. However, the generation of the</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>segmentation masks is not absolved of errors (see also</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>@sec-manual-corrections). Sometimes we must correct them manually due</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>to 1) mistakenly identifying two or more cells as one, 2) identifying</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>two or more cells when it is only one cell, and 3) failing to identify a</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>cell.</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="fu">## Tracking</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>From the image segmentation, we obtain ROI files (region of interest),</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>which contain coordinates of the position of individual cells in each</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>photograph <span class="co">[</span><span class="ot">@10.5555/1386553</span><span class="co">]</span>. Tracking is the tracking of a region of</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>interest in a consecutive series of images. In this case, the tracking</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>generates the identification of the lineages, that is, the ancestry of</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>each cell.</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>We read the ROI files in Python through the *shapely* package, which</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>efficiently reconstructs polygons, thus calculating the length of the</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>cells <span class="co">[</span><span class="ot">@10.5555/1593511; @shapely2007</span><span class="co">]</span>. Also, in Python using ROI files,</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>we track cells with the k-nearest neighbors algorithm that uses various</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>properties such as fluorescence intensity, length, and shape of each</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>cell, to identify cell lineages <span class="co">[</span><span class="ot">@altman1992</span><span class="co">]</span>.</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="fu">## Manual corrections {#sec-manual-corrections}</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>For cell-tracking manual correction, we used *Napari,* an open-source</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>python-based tool designed to explore, annotate, and analyze large</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>multidimensional images <span class="co">[</span><span class="ot">@sofroniew2021a</span><span class="co">]</span>. Our custom cell-viewer allows</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>us to easy lineage data visualization, custom-plotting, and</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>lineage-correction. Code for our cell-viewer is available on</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://github.com/ccg-esb-lab/uJ/tree/master/single-channel&gt;</span>.</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>We produced high-throughput data of thousands of cells with a</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>single-cell resolution to the end of the lineages manual reconstruction.</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>We obtained data about time-series of fluorescent intensity,</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>morphological properties of individual cells (*e.g.*, elongation,</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>duplication rate), and time-resolved population-level statistics</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>(*e.g.*, probability of survival to the antibiotic shock).</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data extraction</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>We construct a file in columnar format through image processing that</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>contains the information necessary to analyze each experiment (*i.e.*,</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>chromosomal and plasmids) in its different traps (*i.e.*, XY</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>identifier). See @tbl-data-columns-specifications for a full description</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>of the output data. Subsequently, the table was analyzed in R for</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>statistical computation and plotting (see @sec-experiment-analysis)</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@R-base</span><span class="co">]</span>.</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>| Column         | Description                                                                                                                                                                                                               |</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>|:---------------------|---------------------------------------------------:|</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>| experimentID   | Unique identifier of the experiment.                                                                                                                                                                                      |</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>| trapID         | Unique identifier of the trap used.                                                                                                                                                                                       |</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>| lineageID      | Unique integer of the stem cell and its ancestry.                                                                                                                                                                         |</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>| cellID         | Unique identification number for each cell existing since the beginning of the experiment or generated later.                                                                                                             |</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>| motherID       | Represents the identification number of the stem cell that gave rise to the progeny.                                                                                                                                      |</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>| trackID        | Indicates the x-y coordinates where the cell being tracked starts.                                                                                                                                                        |</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>| roiID          | Indicates the x-y position in which the cell is located, followed after each photograph.                                                                                                                                  |</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>| frame          | Number of the photograph in the sequence of photographs taken, indicating the elapsed time (10 minutes per frame).                                                                                                        |</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>| length         | Cell length.                                                                                                                                                                                                              |</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>| division       | Indicates cell division events, represented by the value 1 when they occur and 0 otherwise.                                                                                                                               |</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>| GFP            | Represents the relative fluorescence intensity in each cell by green fluorescent protein (*i.e.*, GFP).                                                                                                                   |</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>| DsRed          | Represents the relative fluorescence intensity for cells generated by rhodamine's internalization (*i.e.*, DsRed); an indicator of cell death events.                                                                     |</span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>| tracking_score | Determine how good or bad the tracking of a cell was.                                                                                                                                                                     |</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>| state          | Indicates the state of the cell determined from its length and fluorescence thresholds. -1 for death, 0 for normal, and 1 for filamentation (see @sec-experiment-general-preprocessing for detailed information). |</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>: Resulting table from image processing.</span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>{#tbl-data-columns-specifications}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>