<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.38">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bacterial filamentation: a bet for survival in stressful environments - 1&nbsp; Image processing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./02-experiment-analysis.html" rel="next">
<link href="./00-introduction.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bacterial filamentation: a bet for survival in stressful environments</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/jvelezmagic/bacterial-filamentation-research" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="./Bacterial-filamentation--a-bet-for-survival-in-stressful-environments.pdf" title="Download PDF" class="sidebar-tool px-1"><i class="bi bi-file-pdf"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
            <i class="bi bi-bi-linkedin pe-1"></i>
          LinkedIn
          </a>
        </li>
    </ul>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00-introduction.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-image-processing.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-experiment-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-model-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Models to the rescue; filamentation abstraction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-discussion.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"> <span class="header-section-number">1.1</span> Introduction</a></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"> <span class="header-section-number">1.2</span> Preprocessing</a></li>
  <li><a href="#segmentation" id="toc-segmentation" class="nav-link" data-scroll-target="#segmentation"> <span class="header-section-number">1.3</span> Segmentation</a></li>
  <li><a href="#tracking" id="toc-tracking" class="nav-link" data-scroll-target="#tracking"> <span class="header-section-number">1.4</span> Tracking</a></li>
  <li><a href="#manual-corrections" id="toc-manual-corrections" class="nav-link" data-scroll-target="#manual-corrections"> <span class="header-section-number">1.5</span> Manual corrections</a></li>
  <li><a href="#data-extraction" id="toc-data-extraction" class="nav-link" data-scroll-target="#data-extraction"> <span class="header-section-number">1.6</span> Data extraction</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/blob/main/01-image-processing.qmd" class="toc-action">View source</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/edit/main/01-image-processing.qmd" class="toc-action">Edit this page</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="image-processing" class="quarto-section-identifier d-none d-lg-block"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="introduction" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1.1</span> Introduction</h2>
<p>With the progress of technology, optical and fluorescence microscopy has become a fundamental tool for the characterization and understanding of the bacterial world. Microscopy has allowed humanity to extend its senses to observe the unknown world with exciting new perspectives that they might never otherwise have envisioned. Furthermore, microscopy offers a clear advantage over other techniques used to characterize bacteria since it can acquire data from living cells in spatial resolution <span class="citation" data-cites="schermelleh2019">(<a href="references.html#ref-schermelleh2019" role="doc-biblioref">Schermelleh et al. 2019</a>)</span>.</p>
<p>Including the discovery of fluorescent proteins (<em>e.g.</em>, GFP and DsRed) and improvements in fluorescent reporters, it is possible to specifically label specific cellular components and track cellular functions <span class="citation" data-cites="Specht2017">(<a href="references.html#ref-Specht2017" role="doc-biblioref">Specht, Braselmann, and Palmer 2017</a>)</span>. On the other hand, mechanical and intellectual development of microfluidic research techniques provides an excellent opportunity to overcome bio-medical and chemical techniques <span class="citation" data-cites="convery2019">(<a href="references.html#ref-convery2019" role="doc-biblioref">Convery and Gadegaard 2019</a>)</span>. Collectively, it is possible to study communities of bacteria at the level of individual cells <span class="citation" data-cites="balaban2004a elowitz2002">(<a href="references.html#ref-balaban2004a" role="doc-biblioref">Balaban et al. 2004</a>; <a href="references.html#ref-elowitz2002" role="doc-biblioref">Elowitz et al. 2002</a>)</span>.</p>
<p>Although all this technological development has provided a significant advance for the scientific community, after acquiring fluorescence images, the extraction of quantitative properties from these images is crucial, but unfortunately, a difficult step for analyzing experiments. Not so long ago, image analysis in biology relied on manual quantification. However, manual analysis suffers from two main problems: 1) accuracy and 2) scalability (that is, analyzing miles or more images). Fortunately, improvements in image accuracy and computational image analysis capabilities are revolutionizing the quantification of biological processes through <span class="citation" data-cites="Caicedo2017 Smith2018">(<a href="references.html#ref-Caicedo2017" role="doc-biblioref">Caicedo et al. 2017</a>; <a href="references.html#ref-Smith2018" role="doc-biblioref">Smith et al. 2018</a>)</span>. Therefore, the manual correction required to analyze the experiments is minimal.</p>
<p>Here, we used a series of programs in <span class="math inline">\(\mu \mathrm{J}\)</span> (<a href="https://github.com/ccg-esb-lab/uJ" class="uri">https://github.com/ccg-esb-lab/uJ</a>), which consists of an <span class="math inline">\(\mathrm{ImageJ}\)</span> macro library (mainly) for quantifying unicellular bacterial dynamics in microfluidic devices <span class="citation" data-cites="schneider2012">(<a href="references.html#ref-schneider2012" role="doc-biblioref">Schneider, Rasband, and Eliceiri 2012</a>)</span>. The specific steps used are described below and are summarized in Figure <span class="citation" data-cites="ref">(<a href="references.html#ref-ref" role="doc-biblioref"><strong>ref?</strong></a>)</span>(fig:).</p>
</section>
<section id="preprocessing" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="preprocessing"><span class="header-section-number">1.2</span> Preprocessing</h2>
<p>We exported the figures obtained by the NIS-Elements software (RRID:SCR_014329) from the microfluidics experiments in TIFF (Tagged Image File Format) format. Each figure was named as follows: <em>experimentxyc1t001</em> where <em>experiment</em> indicates the name assigned to the experiment, <em>xy</em> the trap number, <em>c</em> the fluorescence channel, and <em>t</em> the passage of time.</p>
<p>Subsequently, we compile the images, rename them and save them as images in different folders. We maintained the classification by fluorescence channels and phase contrast, and within the channel folder, it is the sub-classification by trap number.</p>
</section>
<section id="segmentation" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="segmentation"><span class="header-section-number">1.3</span> Segmentation</h2>
<p>To determine which parts of the photographs correspond to cells, we carry out an image segmentation analysis. Segmentation consists of classification at the pixel level, which allows us to define the pixels that give identity to the limit of a cell, its interior, and the image’s background (everything that is not a cell). A new image is generated from the above, known as the segmentation mask, containing only the pixels that identify cells.</p>
<p>To build the segmentation mask, we used <em>Deepcell</em> <span class="citation" data-cites="vanvalen2016">(<a href="references.html#ref-vanvalen2016" role="doc-biblioref">Van Valen et al. 2016</a>)</span>. <em>Deepcell</em> is a network trained with a robust set of images that people previously classified as cells. However, the generation of the segmentation masks is not absolved of errors (see also Section @ref(manual-corrections)). Sometimes we must correct them manually due to 1) mistakenly identifying two or more cells as one, 2) identifying two or more cells when it is only one cell, and 3) failing to identify a cell.</p>
</section>
<section id="tracking" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="tracking"><span class="header-section-number">1.4</span> Tracking</h2>
<p>From the image segmentation, we obtain ROI files (region of interest), which contain coordinates of the position of individual cells in each photograph <span class="citation" data-cites="10.5555/1386553">(<a href="references.html#ref-10.5555/1386553" role="doc-biblioref">Brinkmann 2008</a>)</span>. Tracking is the tracking of a region of interest in a consecutive series of images. In this case, the tracking generates the identification of the lineages, that is, the ancestry of each cell.</p>
<p>We read the ROI files in Python through the <em>shapely</em> package, which efficiently reconstructs polygons, thus calculating the length of the cells <span class="citation" data-cites="10.5555/1593511 shapely2007">(<a href="references.html#ref-10.5555/1593511" role="doc-biblioref"><strong>10.5555/1593511?</strong></a>; <a href="references.html#ref-shapely2007" role="doc-biblioref"><strong>shapely2007?</strong></a>)</span>. Also, in Python using ROI files, we track cells with the k-nearest neighbors algorithm that uses various properties such as fluorescence intensity, length, and shape of each cell, to identify cell lineages <span class="citation" data-cites="altman1992">(<a href="references.html#ref-altman1992" role="doc-biblioref">Altman 1992</a>)</span>.</p>
</section>
<section id="manual-corrections" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="manual-corrections"><span class="header-section-number">1.5</span> Manual corrections</h2>
<p>For cell-tracking manual correction, we used <em>Napari,</em> an open-source python-based tool designed to explore, annotate, and analyze large multidimensional images <span class="citation" data-cites="sofroniew2021a">(<a href="references.html#ref-sofroniew2021a" role="doc-biblioref"><strong>sofroniew2021a?</strong></a>)</span>. Our custom cell-viewer allows us to easy lineage data visualization, custom-plotting, and lineage-correction. Code for our cell-viewer is available on <a href="https://github.com/ccg-esb-lab/uJ/tree/master/single-channel" class="uri">https://github.com/ccg-esb-lab/uJ/tree/master/single-channel</a>.</p>
<p>We produced high-throughput data of thousands of cells with a single-cell resolution to the end of the lineages manual reconstruction. We obtained data about time-series of fluorescent intensity, morphological properties of individual cells (<em>e.g.</em>, elongation, duplication rate), and time-resolved population-level statistics (<em>e.g.</em>, probability of survival to the antibiotic shock).</p>
</section>
<section id="data-extraction" class="level2" data-number="1.6">
<h2 data-number="1.6" class="anchored" data-anchor-id="data-extraction"><span class="header-section-number">1.6</span> Data extraction</h2>
<p>We construct a file in columnar format through image processing that contains the information necessary to analyze each experiment (<em>i.e.</em>, chromosomal and plasmids) in its different traps (<em>i.e.</em>, XY identifier). See <a href="#tbl-data-columns-specifications">Table&nbsp;<span>1.1</span></a> for a full description of the output data. Subsequently, the table was analyzed in R for statistical computation and plotting (see Chapter @ref(experiment-analysis)) <span class="citation" data-cites="R-base">(<a href="references.html#ref-R-base" role="doc-biblioref"><strong>R-base?</strong></a>)</span>.</p>
<div id="tbl-data-columns-specifications" class="anchored">
<table class="table">
<caption>Table 1.1: Resulting table from image processing.</caption>
<colgroup>
<col style="width: 35%">
<col style="width: 64%">
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>experimentID</td>
<td>Unique identifier of the experiment.</td>
</tr>
<tr class="even">
<td>trapID</td>
<td>Unique identifier of the trap used.</td>
</tr>
<tr class="odd">
<td>lineageID</td>
<td>Unique integer of the stem cell and its ancestry.</td>
</tr>
<tr class="even">
<td>cellID</td>
<td>Unique identification number for each cell existing since the beginning of the experiment or generated later.</td>
</tr>
<tr class="odd">
<td>motherID</td>
<td>Represents the identification number of the stem cell that gave rise to the progeny.</td>
</tr>
<tr class="even">
<td>trackID</td>
<td>Indicates the x-y coordinates where the cell being tracked starts.</td>
</tr>
<tr class="odd">
<td>roiID</td>
<td>Indicates the x-y position in which the cell is located, followed after each photograph.</td>
</tr>
<tr class="even">
<td>frame</td>
<td>Number of the photograph in the sequence of photographs taken, indicating the elapsed time (10 minutes per frame).</td>
</tr>
<tr class="odd">
<td>length</td>
<td>Cell length.</td>
</tr>
<tr class="even">
<td>division</td>
<td>Indicates cell division events, represented by the value 1 when they occur and 0 otherwise.</td>
</tr>
<tr class="odd">
<td>GFP</td>
<td>Represents the relative fluorescence intensity in each cell by green fluorescent protein (<em>i.e.</em>, GFP).</td>
</tr>
<tr class="even">
<td>DsRed</td>
<td>Represents the relative fluorescence intensity for cells generated by rhodamine’s internalization (<em>i.e.</em>, DsRed); an indicator of cell death events.</td>
</tr>
<tr class="odd">
<td>tracking_score</td>
<td>Determine how good or bad the tracking of a cell was.</td>
</tr>
<tr class="even">
<td>state</td>
<td>Indicates the state of the cell determined from its length and fluorescence thresholds. -1 for death, 0 for normal, and 1 for filamentation (see Section @ref(experiment-general-preprocessing) for detailed information).</td>
</tr>
</tbody>
</table>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-altman1992" class="csl-entry" role="doc-biblioentry">
Altman, N. S. 1992. <span>“An Introduction to Kernel and Nearest-Neighbor Nonparametric Regression.”</span> <em>The American Statistician</em> 46 (3): 175–85. <a href="https://doi.org/10.1080/00031305.1992.10475879">https://doi.org/10.1080/00031305.1992.10475879</a>.
</div>
<div id="ref-balaban2004a" class="csl-entry" role="doc-biblioentry">
Balaban, Nathalie Q., Jack Merrin, Remy Chait, Lukasz Kowalik, and Stanislas Leibler. 2004. <span>“Bacterial Persistence as a Phenotypic Switch.”</span> <em>Science</em> 305 (5690): 1622–25. <a href="https://doi.org/10.1126/science.1099390">https://doi.org/10.1126/science.1099390</a>.
</div>
<div id="ref-10.5555/1386553" class="csl-entry" role="doc-biblioentry">
Brinkmann, Ron. 2008. <em>The Art and Science of Digital Compositing, Second Edition: Techniques for Visual Effects, Animation and Motion Graphics (the Morgan Kaufmann Series in ... Morgan Kaufmann Series in Computer Graphics)</em>. 2nd ed. San Francisco, CA, USA: Morgan Kaufmann Publishers Inc.
</div>
<div id="ref-Caicedo2017" class="csl-entry" role="doc-biblioentry">
Caicedo, Juan C, Sam Cooper, Florian Heigwer, Scott Warchal, Peng Qiu, Csaba Molnar, Aliaksei S Vasilevich, et al. 2017. <span>“Data-Analysis Strategies for Image-Based Cell Profiling.”</span> <em>Nature Methods</em> 14 (9): 849–63. <a href="https://doi.org/10.1038/nmeth.4397">https://doi.org/10.1038/nmeth.4397</a>.
</div>
<div id="ref-convery2019" class="csl-entry" role="doc-biblioentry">
Convery, Neil, and Nikolaj Gadegaard. 2019. <span>“30 Years of Microfluidics.”</span> <em>Micro and Nano Engineering</em> 2 (March): 76–91. <a href="https://doi.org/10.1016/j.mne.2019.01.003">https://doi.org/10.1016/j.mne.2019.01.003</a>.
</div>
<div id="ref-elowitz2002" class="csl-entry" role="doc-biblioentry">
Elowitz, Michael B., Arnold J. Levine, Eric D. Siggia, and Peter S. Swain. 2002. <span>“Stochastic Gene Expression in a Single Cell.”</span> <em>Science</em> 297 (5584): 1183–86. <a href="https://doi.org/10.1126/science.1070919">https://doi.org/10.1126/science.1070919</a>.
</div>
<div id="ref-schermelleh2019" class="csl-entry" role="doc-biblioentry">
Schermelleh, Lothar, Alexia Ferrand, Thomas Huser, Christian Eggeling, Markus Sauer, Oliver Biehlmaier, and Gregor P. C. Drummen. 2019. <span>“Super-Resolution Microscopy Demystified.”</span> <em>Nature Cell Biology</em> 21 (1): 72–84. <a href="https://doi.org/10.1038/s41556-018-0251-8">https://doi.org/10.1038/s41556-018-0251-8</a>.
</div>
<div id="ref-schneider2012" class="csl-entry" role="doc-biblioentry">
Schneider, Caroline A, Wayne S Rasband, and Kevin W Eliceiri. 2012. <span>“NIH Image to ImageJ: 25 Years of Image Analysis.”</span> <em>Nature Methods</em> 9 (7): 671–75. <a href="https://doi.org/10.1038/nmeth.2089">https://doi.org/10.1038/nmeth.2089</a>.
</div>
<div id="ref-Smith2018" class="csl-entry" role="doc-biblioentry">
Smith, Kevin, Filippo Piccinini, Tamas Balassa, Krisztian Koos, Tivadar Danka, Hossein Azizpour, and Peter Horvath. 2018. <span>“Phenotypic Image Analysis Software Tools for Exploring and Understanding Big Image Data from Cell-Based Assays.”</span> <em>Cell Systems</em> 6 (6): 636–53. <a href="https://doi.org/10.1016/j.cels.2018.06.001">https://doi.org/10.1016/j.cels.2018.06.001</a>.
</div>
<div id="ref-Specht2017" class="csl-entry" role="doc-biblioentry">
Specht, Elizabeth A., Esther Braselmann, and Amy E. Palmer. 2017. <span>“A Critical and Comparative Review of Fluorescent Tools for Live-Cell Imaging.”</span> <em>Annual Review of Physiology</em> 79 (1): 93–117. <a href="https://doi.org/10.1146/annurev-physiol-022516-034055">https://doi.org/10.1146/annurev-physiol-022516-034055</a>.
</div>
<div id="ref-vanvalen2016" class="csl-entry" role="doc-biblioentry">
Van Valen, David A., Takamasa Kudo, Keara M. Lane, Derek N. Macklin, Nicolas T. Quach, Mialy M. DeFelice, Inbal Maayan, Yu Tanouchi, Euan A. Ashley, and Markus W. Covert. 2016. <span>“Deep Learning Automates the Quantitative Analysis of Individual Cells in Live-Cell Imaging Experiments.”</span> Edited by Martin Meier-Schellersheim. <em>PLOS Computational Biology</em> 12 (11): e1005177. <a href="https://doi.org/10.1371/journal.pcbi.1005177">https://doi.org/10.1371/journal.pcbi.1005177</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./00-introduction.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Introduction</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./02-experiment-analysis.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>