# Experiment analysis {#sec-experiment-analysis}

```{r}
#| label: find-preprocessing-files-02-experiment-analysis
experiment_analysis_files <- fs::dir_ls(
    path = here::here("data-preprocessing", "02-experiment-analysis"),
    type = "file",
    glob = "*qmd"
)
```

```{r child=experiment_analysis_files}
#| label: execute-preprocessing-files-02-experiment-analysis
```

```{r}
#| label: libraries
library(tidyverse)
library(tidymodels)
library(tidytext)
library(plotly)
library(ggpubr)
library(GGally)
library(ggdist)
library(embed)
library(here)
library(fs)
```

```{r}
#| label: read-processed-data
cells_file <- here("data", "processed", "cells_summary.csv")
cells_raw_df <- read_csv(
  file = cells_file,
  show_col_types = FALSE
) %>% 
  select(-over_ds_red_id) %>%
  rename(divided = divided_id)
```

```{r}
#| label: create-factors
cells_df <- cells_raw_df %>% 
  mutate(
    filamented_id = factor(
      x = filamented_id,
      levels = c(FALSE, TRUE), 
      labels = c("Not filamented", "Filamented")
    ),
    survived = factor(
      x = survived,
      levels = c(FALSE, TRUE),
      labels = c("Not survived", "Survived")
    ),
    cell_status = interaction(
      filamented_id,
      survived,
      sep = " - "
    ),
    cell_status = paste0(
      filamented_id,
      " - ",
      survived
    ),
    cell_status = factor(cell_status)
  ) %>% 
  relocate(where(is.character), where(is.factor), where(is.logical))
```

```{r}
#| label: set-default=plot-style
theme_set(
  theme_bw() +
  theme(
    legend.position = "top",
    strip.background = element_blank()
  )
)
```

```{r}
#| label: utility-functions
parse_metrics_column <- function(.data, metric_column) {
  .data %>% 
    mutate(
      {{ metric_column }} := str_remove(
        string = {{ metric_column }},
        pattern = "(.+)_"
      ) %>% 
        factor(
          levels = c("first", "sos", "last"),
          labels = c("Initial", "SOS", "End")
        ) %>%
        identity()
    )
}
```

## Introduction

The previous chapter (see @sec-image-processing) detailed the
steps necessary to extract data from a set of microfluidic images
through image analysis techniques and fluorescence microscopy. Each step
was instrumental in creating a dataset that was easy to explore and ask
questions. With the help of computational biology, systems biology, and
data analysis techniques, we could process these files to help us in the
search to find the role of filamentation in cell survival.

Both the ideas and concepts of computational biology and systems biology
contributed to the development of this analysis. In principle,
computational biology originated after the origin of computer science
with the British mathematician and logistician Alan Turing (regularly
known as the father of computing) [@turing1950]. Over time, systems
biology emerged as an area that synergistically combines models and
experimental data to understand biological processes [@bruggeman]. Thus,
giving a step towards creating models that, in general, are
phenomenological but that sometimes serve to discover new ideas about
the process under study. Ideas and aspects of the study of biological
sciences that otherwise could be unthinkable without the computer's
power.

Here, we divide the experimental analysis into two main parts: 1) at the
cell level or measurements at specific points in time and 2) at the
population level and time series. The first level allowed us to identify
the individual contribution of each variable understudy to determine
cell survival. The second level allowed us to understand how the
population behaves according to the passage of time in the face of
exposure to a harmful agent (in this case, beta-lactam antibiotics).
Together, both visions of the same study phenomenon allowed us to
extract the main ideas for postulating a mathematical model that seeks
to show how filamentation is a factor for cell survival in stressful
environments (see @sec-model-analysis).

## General preprocessing of data {#sec-experiment-general-preprocessing}

The raw data processing consisted mainly of creating two levels of
observation for the cells of both chromosomal strains and multicopy
plasmids. The first level is at a cell granularity, that is, point
properties. The second level consists of the cells over time, thus
observing properties at the population level. We did this because it
would allow us to understand what factors are affecting filamentation
and why.

We normalized the fluorescence values of DsRed and GFP for both
experiments based on the values observed before exposure to antibiotics.
It allowed us to have a basis to work with and compare expressions
between cells. In the case of DsRed environment drug concentration, we
also applied a logarithmic transformation to observe subtle changes in
fluorescence intensity that would allow us to detect cell dead.

Ultimately, we decided to classify cells into four fundamental groups
based on whether the cell filamented and survived (see
@fig-cell-distribution-across-experiments). We
define a *filamented cell* as a cell with more than two standard
deviations from the mean concerning the lengths observed before
introducing antibiotics into the system. On the other hand, although
there are multiple ways to define death from single-cell observations
[@trevors2012; @kroemer2008], we considered a *cell dead or missing*
when we stopped having information about it, either because of
fluorescence in the red channel was above a given threshold (resulting
from an increase in cell membrane permeability and the introduction of
fluorescent dye into the cell) or because it left the field of
observation. Therefore, a *surviving cell* is defined as a cell observed
before and after exposure to the antibiotic and does not surpass the
DsRed threshold.

```{r}
#| label: fig-cell-distribution-across-experiments
#| fig-scap: Cell classification and its distribution across experiments.
#| fig-cap: >
#|   **Cell classification and its distribution across experiments.**
#|   We define a *filamented cell* as a cell whose length exceeded two
#|   standard deviations from the mean at any time during the experiment.
#|   A *surviving cell* is a cell that was observed before and after exposure
#|   to the antibiotic. Accordingly, we removed from the analysis those
#|   cells that died before or were born after the exposure of the experiment.
#|   Therefore, we delimited the effect caused by the exposure to
#|   the antibiotic.
#|   
p_cells_distribution <- cells_df %>% 
  count(experiment_id, cell_status) %>%
  group_by(experiment_id) %>% 
  mutate(
    percentage = n / sum(n) * 100,
    ymax = cumsum(percentage),
    ymin = c(0, head(ymax, -1)),
    labels = paste0(format(percentage, digits = 2), "%"),
    labels_position = (ymax + ymin) / 2,
    total_label = paste0("Total:\n", format(sum(n), big.mark = ","), " cells")
  ) %>% 
  ungroup() %>% 
  identity() %>% 
    ggplot(
    aes(
      ymin = ymin,
      ymax = ymax,
      xmin = 3,
      xmax = 4
    )
  ) +
  geom_rect(
    size = 1.5,
    color = "white",
    aes(fill = cell_status)
  ) +
  geom_label(
    x = 2,
    aes(
      y = labels_position,
      label = labels
    ),
    label.size = NA,
    size = 3.5,
  ) +
  geom_text(
    aes(x = -Inf, y = -Inf, label = total_label),
    hjust = 0.5, 
    vjust = 0.5
  ) +
  facet_grid(. ~ experiment_id) +
  coord_polar(theta = "y") +
  xlim(c(-1, 4)) +
  guides(
    fill = guide_legend(ncol = 2)
  ) +
  theme_void() +
  theme(
    legend.position = "bottom"
  ) +
  labs(
    fill = "Cell status"
  ) +
  NULL

p_cells_distribution
```

## Results

### Cell length and the amount of GFP are crucial in determining cell survival {#sec-length-gfp-crucial}

We evaluated the DsRed, GFP, and length values for each cell at
different time points: initial, filamentation, and end. This
preprocessing allowed us to observe and quantify each cell at critical
times in the experiment and eliminate noise or signals outside the scope
of this investigation.

We define the *initial time* as the first time we observed the cell in
the experiment. *Filamentation time* equals when a cell reaches the
filamentation threshold (see @fig-length-temporal-distribution) for the first
time. We defined the *end time* as the time of the last observation of
the cell. We decided to bound the end time for surviving cells to one
frame (10 min) after the end of antibiotic exposure so that the observed
signal would reflect the final stress responses.

When we compared the distributions of DsRed, GFP, and length for both
experiments, we observed its changes in its role for cell survival. In
@fig-dsred-temporal-distribution, we show
that indistinctly and, as expected, surviving cells managed to eliminate
the antibiotic by the end time. In contrast, dead cells presented higher
levels of antibiotics (measured by proxy through the mean DsRed
intensity of the cell).

```{r}
#| label: fig-dsred-temporal-distribution
#| fig-scap: DsRed temporal distribution.
#| fig-cap: >
#|   **DsRed temporal distribution.**
#|   To evaluate the incident effect of the antibiotic marked by DsRed
#|   on cells by class, we show its values at three key moments: start,
#|   filamentation (SOS), and end. The upper asterisks represent the
#|   significance value when comparing a group X to the filamented and
#|   surviving cell reference. Asterisks in a line indicate whether or not
#|   there is a significant difference in the survival of non-filamented cells.
#|   The black dots represent the mean of each group, and the lines that join
#|   them are a comparative guide. The extent of the black bars represents the
#|   distribution of the data. Although, at the initial time, we observe
#|   multiple significant differences, this is likely due to the intrinsic
#|   noise of the system since, as expected, the values are close to zero.
#|   We observed a difference between the surviving and non-filamented
#|   cells for the chromosomal strain for the SOS time, but the same did not
#|   occur for the plasmid strain. The final amount of DsRed makes a clear
#|   difference between survival and death.
#|   
p_temporal_dsred_distributution <- cells_df %>% 
  pivot_longer(
    cols = contains("ds_red"),
    names_to = "metric",
    values_to = "value"
  ) %>% 
  parse_metrics_column(metric) %>% 
  filter(!is.na(value)) %>% 
  identity() %>% 
  ggplot(aes(x = cell_status, y = value, fill = cell_status)) +
  stat_eye() +
  stat_summary(fun=median, geom="line", aes(group=1))  +
  stat_compare_means(
    method = "t.test",
    comparisons = list(c("Not filamented - Survived", "Not filamented - Not survived")),
    label = "p.signif",
    label.y = c(0.4),
    hide.ns = TRUE
  ) +
  # stat_compare_means(
  #   method = "anova",
  #   label.y.npc = 0.93
  # ) + # Add global annova p-value
  stat_compare_means(
    label = "p.signif",
    method = "t.test",
    ref.group = "Filamented - Survived",
    hide.ns = TRUE,
    label.y.npc = 0.75
    ) +
  stat_gradientinterval(position = "dodge", fill_type = "segments") +
  facet_grid(experiment_id ~ metric) +
  guides(
    color = guide_legend(ncol = 2),
    fill = guide_legend(ncol = 2)
  ) +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank()
  ) +
  labs(
    fill = "Cell status",
    y = "DsRed value"
  ) +
  NULL

p_temporal_dsred_distributution
```

On the other hand, GFP observations in
@fig-gfp-temporal-distribution showed us two
essential things for cell classification: 1) The chromosomal strain did
not exhibit noticeable changes in GFP levels, and 2) filamented cells
were those that had low fluorescent intensities (low plasmid
copy-number) at the beginning of the experiment. For the final
observation times, GFP measurements indicated that among the cells that
did not filament, the ones that survived exhibited a reduced GFP
expression concerning cells killed by the antibiotic. Meanwhile, for the
filamented cells, whether surviving or dead, their GFP measurements
indicated no difference at the beginning or the end of the experiment,
suggesting the presence of other determinants of cell survival.

```{r}
#| label: fig-gfp-temporal-distribution
#| fig-scap: GFP temporal distribution.
#| fig-cap: >
#|   **GFP temporal distribution.**
#|   To evaluate the incident effect of the GFP on cells by class, we used
#|   the same notation as in @fig-dsred-temporal-distribution.
#|   The chromosomal strain exhibits variability in GFP at different
#|   time points, mainly due to experimental noise resulting from low
#|   fluorescent intensity values. As expected,
#|   in the plasmid strain, filamented cells had a lower initial GFP.
#|   At the time of filamentation, there appear to be differences in
#|   fluorescence between surviving and dead cells. However, in the end time,
#|   we observed that the surviving non-filamented cells have lower GFP
#|   values than the non-filamented dead cells and alive filamented cells.
#|
p_temporal_gfp_distributution <- cells_df %>% 
  pivot_longer(
    cols = contains("gfp"),
    names_to = "metric",
    values_to = "value"
  ) %>% 
  parse_metrics_column(metric) %>% 
  filter(!is.na(value)) %>% 
  identity() %>% 
  ggplot(aes(x = cell_status, y = value, fill = cell_status)) +
  stat_eye() +
  stat_summary(fun=median, geom="line", aes(group=1))  +
  stat_compare_means(
    method = "t.test",
    comparisons = list(c("Not filamented - Survived", "Not filamented - Not survived")),
    label = "p.signif",
    label.y = c(2.0),
    hide.ns = TRUE
  ) +
  stat_compare_means(
    method = "anova",
    label.y.npc = 0.93
  ) + # Add global annova p-value
  stat_compare_means(
    label = "p.signif",
    method = "t.test",
    ref.group = "Filamented - Survived",
    hide.ns = TRUE,
    label.y.npc = 0.75
    ) +
  stat_gradientinterval(position = "dodge", fill_type = "segments") +
  facet_grid(experiment_id ~ metric) +
  guides(
    color = guide_legend(ncol = 2),
    fill = guide_legend(ncol = 2)
  ) +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank()
  ) +
  labs(
    fill = "Cell status",
    y = "GFP value"
  ) +
  NULL

p_temporal_gfp_distributution
```

Cell length was one of the factors that GFP expression levels could not
explain for cell survival. In @fig-length-temporal-distribution, we show that the
conclusions regarding filamentation were applicable for both chromosomal
or plasmid strains. For the initial times, filamented and survived cells
were shorter in length than those that died but longer than not
filamented cells of both classes, while non-filamented cells did not
differ from each other. We observed no length differences between cells
at filamentation time. Thus, survival could depend on other factors,
such as growth rate. At the final time, the results were well-defined.
Surviving cells had a greater length relative to their non-surviving
pair (*i.e.*, dead filamented and non-filamented cells). However, for
filamented cells, surviving cells represent a distribution of higher
final length values in general but not as extensive as their dead
counterpart. Which we could explain as a length limit to which cells can
grow without dying. Nevertheless, we had no information to evaluate such
a hypothesis.

```{r}
#| label: fig-length-temporal-distribution
#| fig-scap: Length temporal distribution.
#| fig-cap: >
#|   **Length temporal distribution.**
#|   To evaluate the incident effect of length on cells by class,
#|   we use the same notation as in @fig-dsred-temporal-distribution.
#|   The observations for both strains, chromosomal or plasmid, are the same.
#|   In the beginning, the surviving filamented cells already have a difference
#|   in length concerning the rest of the classes. At the time of
#|   filamentation, there is no difference to help determine whether the
#|   cell will survive or not. Finally, in the final time, it seems that
#|   the surviving filamented cells have a greater length than the rest
#|   of the groups. However, this length is moderate compared to the excess
#|   length shown by non-surviving filamented cells. On the other hand,
#|   we highlighted the growth of the surviving non-filamented cells.
#|   Therefore, although they did not reach a length for us to classify as
#|   filamented, the cells did resort to filamentation.
#|
p_temporal_length_distributution <- cells_df %>% 
  pivot_longer(
    cols = contains("length"),
    names_to = "metric",
    values_to = "value"
  ) %>% 
  parse_metrics_column(metric) %>% 
  filter(!is.na(value)) %>% 
  identity() %>% 
  ggplot(aes(x = cell_status, y = value, fill = cell_status)) +
  geom_hline(aes(yintercept = filamentation_threshold), linetype = "dashed", alpha = 1 / 2) +
  stat_eye() +
  stat_summary(fun=median, geom="line", aes(group=1))  +
  stat_compare_means(
    method = "t.test",
    comparisons = list(
      c("Not filamented - Survived", "Not filamented - Not survived")
    ),
    label = "p.signif",
    label.y = c(60),
    hide.ns = TRUE
  ) +
  stat_compare_means(
    method = "anova",
    label.y.npc = 0.43,
    label.x.npc = 0.3
  ) + # Add global annova p-value
  stat_compare_means(
    label = "p.signif",
    method = "t.test",
    ref.group = "Filamented - Survived",
    hide.ns = TRUE,
    label.y.npc = 0.3
  ) +
  #stat_gradientinterval(position = "dodge", fill_type = "segments") +
  facet_grid(experiment_id ~ metric) +
  coord_cartesian(ylim = c(0, 150)) +
  guides(
    color = guide_legend(ncol = 2),
    fill = guide_legend(ncol = 2)
  ) +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank()
  ) +
  labs(
    fill = "Cell status",
    y = "Length value"
  ) +
  NULL

p_temporal_length_distributution
```


Once we observed the effects of GFP expression levels and lengths in
determining whether a cell lives or dies, we projected the cells onto
the plane and painted them with their class status (See
@fig-cell-distribution-across-experiments) to
determine whether these two variables contained the necessary
information to cluster the data correctly. In @fig-just-initial-values, 
we show the initial GFP
and length values projection. While, with some work, we could
contextually place the results in Figures
@fig-gfp-temporal-distribution and
@fig-length-temporal-distribution, the initial
values did not appear to determine the classes. Therefore, we explored
the final versus initial values differences in
@fig-metric-differences. With this new
representation of the cells in the plane, we contextualized the
statistical results presented in Figures
@fig-gfp-temporal-distribution and
@fig-length-temporal-distribution. Besides, it
showed us that differences in length (*i.e.*, filamentation) and
reductions in GFP expression are essential in determining cell survival.
Though, the clustering of cells by state is not completely separated,
which means that other variables are affection the experimental results
in cell survival.

```{r}
#| label: fig-just-initial-values
#| fig-scap: Experiment initial values.
#| fig-cap: >
#|   **Experiment initial values.**
#|   By positioning a cell in space based on its initial length and GFP
#|   values, we can see that class separation occurs, but not as a strong
#|   signal. Therefore, we concluded that although the initial state
#|   influences the result, this is not everything. For this, we have
#|   the example of the length changes throughout the experiment caused
#|   by filamentation. In this graph, the GFP scale is at log10 to help
#|   us observe those minor differences between the experiments.
#|
p_initial_values <- cells_df %>% 
  ggplot(aes(x = log(gfp_first), y = length_first, color = cell_status)) +
  geom_point(alpha = 1/2, size = 0.5) +
  facet_wrap(. ~ experiment_id, scales = "free") +
  guides(
    color = guide_legend(ncol = 2, override.aes = list(alpha = 1, size = 1)),
    fill = guide_legend(ncol = 2)
  ) +
  labs(
    x = "Initial normalized GFP (log10)",
    y = "Initial length",
    color = "Cell status"
  ) +
  NULL

p_initial_values
```

```{r}
#| label: fig-metric-differences
#| fig-scap: Experiment initial values differences.
#| fig-cap: >
#|   **Experiment initial values differences.**
#|   By comparing the metric differences of the last observation and
#|   the first observation of a cell, we can separate mainly the
#|   surviving filamented cells from those that did not do it in both
#|   experiments (green dots). Meanwhile, cells with plasmids form a
#|   small accumulation of surviving cells that did not produce
#|   filament (purple dots). However, this has made a breakthrough in
#|   understanding what is affecting cell survival. There are still
#|   variables that we can include to understand this phenomenon better.
#|   
p_metric_differences <- cells_df %>% 
  ggplot(aes(x = log(gfp_last) - log(gfp_first), y = length_last - length_first, color = cell_status)) +
  geom_point(alpha = 1/20) +
  facet_grid(~experiment_id) +
  guides(
    color = guide_legend(ncol = 2, override.aes = list(alpha = 1)),
    fill = guide_legend(ncol = 2)
  ) +
  labs(
    x = "End GFP - Initial GFP",
    y = "End length - Initial length",
    color = "Cell status"
  ) +
  NULL

p_metric_differences
```

## Number of divisions and cell age do not appear to play a clear role in determining cell survival

In @sec-length-gfp-crucial, we explored how GFP variability
and cell length influence cell survival. However,
@fig-just-initial-values and @fig-metric-differences showed us the
possibility of other factors relevant to the phenomenon under study. As
some papers in the literature suggest, some of these other factors may be
cell division and chronological age (*i.e.*, how much time has passed since
the last cell division at the time of exposure to a toxic agent)
[@moger-reischer2019; @roostalu2008; @heinrich2015]. Therefore, we chose
to observe these two metrics in experiments at a purely qualitative
level, i.e., without the inclusion of, e.g., metrics of membrane or cell
cycle properties [@Joseleau-Petit1999].

Although we expected to see a small contribution, either by the number
of divisions or cell age, in @fig-number-divisions and 
@fig-time-since-last-division, we could not observe
a precise effect of these variables on cell survival. Patterns that,
although they could have an explanation or biological significance, we
decided to omit as relevant in the characterization of our cells, since
the signal was not clear. However, we derived from this analysis a
slightly simpler variable that tells us whether a cell underwent a cell
division event or not. So it gives us a more generalized picture of the
contribution of division to cell survival (see
@fig-plasmid-pca-variable-contribution).

```{r}
#| label: fig-number-divisions
#| fig-scap: Cell's number of divisions.
#| fig-cap: >
#|   **Cell's number of divisions.**
#|   Chromosomal cells exhibited more divisions for surviving classes
#|   and non-surviving filamented cells (*i.e.*, purple, green, and
#|   red dots) relative to unchanged behavior in plasmid cells.
#|   Therefore, its contribution to filamentation remains uncertain.
#|   
p_number_divisions <-cells_df %>% 
  ggplot(aes(x = cell_status, y = n_divisions, fill = cell_status)) +
  stat_eye(position = "dodge") +
  stat_compare_means(
    label = "p.signif",
    method = "t.test",
    ref.group = "Filamented - Survived",
    label.y.npc = 0.75,
    hide.ns = TRUE
  ) +
  stat_compare_means(
    method = "t.test",
    comparisons = list(c("Not filamented - Survived", "Not filamented - Not survived")),
    label = "p.signif",
    label.y = 7,
    hide.ns = TRUE
  ) +
  facet_grid(. ~ experiment_id) +
  coord_cartesian(ylim = c(0, 10)) +
  guides(
    fill = guide_legend(ncol = 2)
  ) +
  theme(
    panel.grid.major = element_blank(),
    axis.text.x = element_blank()
  ) +
  labs(
    x = "Cell status",
    y = "Number of divisions",
    fill = "Cell status"
  ) +
  NULL

p_number_divisions
```

```{r}
#| label: fig-time-since-last-division
#| fig-scap: Time elapsed since the last division at the beginning of the experiment.
#| fig-cap: >
#|   **Time elapsed since the last division at the beginning of the experiment.**
#|   The mean time of the last division before starting the experiment
#|   indicates that it did not influence the final result for chromosomal
#|   cells. There is a slight difference between the filamented-not survived
#|   cells and the rest for cells with plasmids. However, the signal does not
#|   appear to be strong on the survival role. Therefore, we conclude that we
#|   have no evidence to support that the time of the last division at the
#|   beginning of the experiment influences the final classification results.
#|   
p_time_since_last_division <- cells_df %>% 
  filter(!is.na(time_since_last_division_to_experiment_start)) %>% 
  ggplot(
    aes(
      x = experiment_id,
      y = time_since_last_division_to_experiment_start,
      fill = cell_status
    )
  ) +
  stat_eye(position = "dodge") +
  labs(
    x = "Experiment",
    y = "Time since last division to experiment start",
    fill = "Cell status"
  ) +
  guides(
    color = guide_legend(ncol = 2),
    fill = guide_legend(ncol = 2)
  ) +
  NULL

p_time_since_last_division
```

### Time to reach filamentation matters in determining cell survival

In @fig-dsred-temporal-distribution,
@fig-gfp-temporal-distribution, and
@fig-length-temporal-distribution, we showed how, at
the time of filamentation, DsRed and GFP levels appeared indifferent to
the cells. Therefore, we hypothesized that a possible variable that
could determine cell survival could be its time to activate its
anti-stress response system that causes filamentation. Furthermore, we
also guided our hypothesis by previous reports showing us how the gene
expression level can induce filamentation with tight temporal
coordination [x].

While, for our analyses, we did not measure the concentration of
antibiotic that triggers filamentation per se, we indirectly quantified
its effect by using the time it took for a cell to reach a length at
which it is already considered a filamentating cell. Furthermore, to
recognize that the observed effect was a product of the experiment, we
decided to keep only filamented cells just once antibiotic exposure
began.

@fig-time-to-filamentation-filtered shows how
filamentation times are narrower for chromosomal cells than for
plasmid-bearing cells. Then, we hypothesize that the effect could come
from the heterogeneity in the plasmid copy number in the population.
Also, interestingly, we observed that, for both experiments, cells that
survived had longer filamentation times than the cells that died. These
differences in response times suggest the following: 1) if the cell
grows too fast, it will reach a limit and start to accumulate
antibiotics constantly, and 2) if the cell grows too fast, it is likely
that the cost of maintaining an ample length for prolonged periods of
exposure will become counterproductive.

```{r}
#| label: fig-time-to-filamentation-filtered
#| fig-scap: Time to filamentation filtered.
#| fig-cap: >
#|  **Time to filamentation filtered.**
#|  To quantify the effect of filamented to survive, we filtered those
#|  cells that filamented during the experiment. In this way, we
#|  normalize the start times for the calculation of the filamentation
#|  time. For both strains, the filamentation time had a more significant
#|  delay in the surviving cells.
#|
p_time_to_filamentation_filtered <- cells_df %>% 
  filter(
    filamented_id == "Filamented",
    time_sos > antibiotic_start_time
  ) %>% 
  mutate(
    time_to_sos = time_sos - antibiotic_start_time,
  ) %>%
  ggplot(aes( x = experiment_id, y = time_to_sos, fill = survived)) +
  stat_eye(position = "dodge") +
  stat_compare_means(
    label = "p.signif"
  ) +
  labs(
    x = "Experiment",
    y = "Time to filamentation (minutes)",
    fill = "Cell status"
  )

p_time_to_filamentation_filtered
```


In Figure @fig-initial-values-with-time, we decided
to project the results of Figure
@fig-time-to-filamentation-filtered in a space
similar to the one described in Figure
@fig-just-initial-values). Thus, we separated our
data into cells that survived and cells that did not, and painted them
when it took them to reach their filamented state. We realized that,
indeed, by adding this temporal component to the initial variables of
length and GFP, we could separate surviving cells from dead cells to a
greater degree. However, it may still not be enough, and there are still
many other variables that play a crucial role in understanding the
ecology of stress and how some cells will be survivors or not.

```{r}
#| label: fig-initial-values-with-time
#| fig-scap: Experiment initial values with time to filamentation.
#| fig-cap: >
#|   **Experiment initial values with time to filamentation.**
#|   As in @fig-just-initial-values, including the time it will take for
#|   cells to filament allows us to understand the phenomenon of survival
#|   better. Cells that filamented and survived generally have a much
#|   higher delay than their non-filamented peers for both strains
#|   (see @fig-time-to-filamentation-filtered).
#|
p_initial_values_with_time <- cells_df %>% 
  filter(
    filamented_id == "Filamented",
    time_sos > antibiotic_start_time
  ) %>% 
  mutate(time_to_sos = time_sos - antibiotic_start_time) %>%
  ggplot(aes(x = length_first, y = log(gfp_first), z = time_to_sos)) +
  stat_summary_2d() +
  facet_grid(experiment_id ~ survived) +
  scale_fill_viridis_c(option = "magma") +
  labs(
    x = "Initial length",
    y = "Initial GFP",
    fill = "Time to filamentation (minutes)"
  ) +
  NULL

p_initial_values_with_time
```
