<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bacterial filamentation: a bet for survival in stressful environments - 2&nbsp; Experiment analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./03-model-analysis.html" rel="next">
<link href="./01-image-processing.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bacterial filamentation: a bet for survival in stressful environments</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/jvelezmagic/bacterial-filamentation-research" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="./Bacterial-filamentation--a-bet-for-survival-in-stressful-environments.pdf" title="Download PDF" class="sidebar-tool px-1"><i class="bi bi-file-pdf"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
            <i class="bi bi-bi-linkedin pe-1"></i>
          LinkedIn
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00-introduction.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-image-processing.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-experiment-analysis.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-model-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Models to the rescue; filamentation abstraction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-discussion.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./98-appendices.html" class="sidebar-item-text sidebar-link">Appendix</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="toc-section-number">2.1</span>  Introduction</a></li>
  <li><a href="#sec-experiment-general-preprocessing" id="toc-sec-experiment-general-preprocessing" class="nav-link" data-scroll-target="#sec-experiment-general-preprocessing"><span class="toc-section-number">2.2</span>  General preprocessing of data</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="toc-section-number">2.3</span>  Results</a>
  <ul>
  <li><a href="#sec-length-gfp-crucial" id="toc-sec-length-gfp-crucial" class="nav-link" data-scroll-target="#sec-length-gfp-crucial"><span class="toc-section-number">2.3.1</span>  Cell length and the amount of GFP are crucial in determining cell survival</a></li>
  <li><a href="#number-of-divisions-and-cell-age-do-not-appear-to-play-a-clear-role-in-determining-cell-survival" id="toc-number-of-divisions-and-cell-age-do-not-appear-to-play-a-clear-role-in-determining-cell-survival" class="nav-link" data-scroll-target="#number-of-divisions-and-cell-age-do-not-appear-to-play-a-clear-role-in-determining-cell-survival"><span class="toc-section-number">2.3.2</span>  Number of divisions and cell age do not appear to play a clear role in determining cell survival</a></li>
  <li><a href="#time-to-reach-filamentation-matters-in-determining-cell-survival" id="toc-time-to-reach-filamentation-matters-in-determining-cell-survival" class="nav-link" data-scroll-target="#time-to-reach-filamentation-matters-in-determining-cell-survival"><span class="toc-section-number">2.3.3</span>  Time to reach filamentation matters in determining cell survival</a></li>
  <li><a href="#sec-unsupervised-classification" id="toc-sec-unsupervised-classification" class="nav-link" data-scroll-target="#sec-unsupervised-classification"><span class="toc-section-number">2.3.4</span>  Increasing the system’s complexity and analyzing it in an unsupervised way allows a correct classification of cell states</a>
  <ul class="collapse">
  <li><a href="#principal-component-analysis-pca-emphasizes-the-importance-of-cell-length-and-its-gfp-in-cell-survival" id="toc-principal-component-analysis-pca-emphasizes-the-importance-of-cell-length-and-its-gfp-in-cell-survival" class="nav-link" data-scroll-target="#principal-component-analysis-pca-emphasizes-the-importance-of-cell-length-and-its-gfp-in-cell-survival"><span class="toc-section-number">2.3.4.1</span>  Principal Component Analysis (PCA) emphasizes the importance of cell length and its GFP in cell survival</a></li>
  <li><a href="#uniform-manifold-approximation-and-projection-umap-correctly-represents-the-local-structure-of-cell-states" id="toc-uniform-manifold-approximation-and-projection-umap-correctly-represents-the-local-structure-of-cell-states" class="nav-link" data-scroll-target="#uniform-manifold-approximation-and-projection-umap-correctly-represents-the-local-structure-of-cell-states"><span class="toc-section-number">2.3.4.2</span>  Uniform Manifold Approximation and Projection (UMAP) correctly represents the local structure of cell states</a></li>
  </ul></li>
  <li><a href="#population-dynamics-reveal-how-filamentation-contributes-cell-survival" id="toc-population-dynamics-reveal-how-filamentation-contributes-cell-survival" class="nav-link" data-scroll-target="#population-dynamics-reveal-how-filamentation-contributes-cell-survival"><span class="toc-section-number">2.3.5</span>  Population dynamics reveal how filamentation contributes cell survival</a></li>
  <li><a href="#heterogeneity-in-plasmid-copy-number-allows-various-forms-of-survival-in-addition-to-filamentation" id="toc-heterogeneity-in-plasmid-copy-number-allows-various-forms-of-survival-in-addition-to-filamentation" class="nav-link" data-scroll-target="#heterogeneity-in-plasmid-copy-number-allows-various-forms-of-survival-in-addition-to-filamentation"><span class="toc-section-number">2.3.6</span>  Heterogeneity in plasmid copy-number allows various forms of survival in addition to filamentation</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="toc-section-number">2.4</span>  Discussion</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/blob/main/02-experiment-analysis.qmd" class="toc-action">View source</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/jvelezmagic/bacterial-filamentation-research/edit/main/02-experiment-analysis.qmd" class="toc-action">Edit this page</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span id="sec-experiment-analysis" class="quarto-section-identifier d-none d-lg-block"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="introduction" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">2.1</span> Introduction</h2>
<p>The previous chapter (see <a href="01-image-processing.html"><span>Chapter&nbsp;1</span></a>) detailed the steps necessary to extract data from a set of microfluidic images through image analysis techniques and fluorescence microscopy. Each step was instrumental in creating a dataset that was easy to explore and ask questions. With the help of computational biology, systems biology, and data analysis techniques, we could process these files to help us in the search to find the role of filamentation in cell survival.</p>
<p>Computational biology and systems biology contributed to the development of this analysis. In principle, computational biology originated after the origin of computer science with the British mathematician and logistician Alan Turing (regularly known as the father of computing) <span class="citation" data-cites="turing1950">(<a href="99-references.html#ref-turing1950" role="doc-biblioref">TURING 1950</a>)</span>. Over time, systems biology emerged as an area that synergistically combines models and experimental data to understand biological processes <span class="citation" data-cites="bruggeman">(<a href="99-references.html#ref-bruggeman" role="doc-biblioref">Bruggeman et al. 2007</a>)</span>. Thus, giving a step towards creating models that, in general, are phenomenological but sometimes serve to discover new ideas about the process under study. Without the computer’s power, modern ideas and aspects of studying biological sciences would otherwise be unthinkable.</p>
<p>Here, we divide the experimental analysis into two main parts: 1) at the cell level or measurements at specific points in time and 2) at the population level and time series. The first level allowed us to identify the individual contribution of each variable understudy to determine cell survival. The second level allowed us to understand how the population behaves according to the passage of time in the face of exposure to a harmful agent (in this case, beta-lactam antibiotics). Together, both visions of the same study phenomenon allowed us to extract the main ideas for postulating a mathematical model that seeks to show how filamentation is a factor for cell survival in stressful environments (see <a href="03-model-analysis.html"><span>Chapter&nbsp;3</span></a>).</p>
</section>
<section id="sec-experiment-general-preprocessing" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="sec-experiment-general-preprocessing"><span class="header-section-number">2.2</span> General preprocessing of data</h2>
<p>The raw data processing consisted mainly of creating two levels of observation for the cells of both chromosomal strains and multicopy plasmids. The first level is at a cell granularity, point properties. The second level consists of the cells over time, thus observing properties at the population level. We did this because it would allow us to understand what factors affect filamentation and why.</p>
<p>We normalized the fluorescence values of DsRed and GFP for both experiments based on the values observed before antibiotic exposure. It allowed us to have a basis to work with and compare expressions between cells. In the case of DsRed environment drug concentration, we also applied a logarithmic transformation to observe subtle changes in fluorescence intensity that would allow us to detect cell death.</p>
<p>Ultimately, we decided to classify cells into four fundamental groups based on whether the cell filamented and survived (see <a href="#fig-cell-distribution-across-experiments">Figure&nbsp;<span>2.1</span></a>). We define a <em>filamented cell</em> as a cell with more than two standard deviations from the mean concerning the lengths observed before introducing antibiotics into the system. On the other hand, although there are multiple ways to define death from single-cell observations <span class="citation" data-cites="trevors2012 kroemer2008">(<a href="99-references.html#ref-trevors2012" role="doc-biblioref">Trevors 2012</a>; <a href="99-references.html#ref-kroemer2008" role="doc-biblioref">Kroemer et al. 2008</a>)</span>, we considered a <em>cell dead or missing</em> when we stopped having information about it, either because of fluorescence in the red channel was above a given threshold (resulting from an increase in cell membrane permeability and the introduction of fluorescent dye into the cell) or because it left the field of observation. Therefore, we defined a <em>surviving cell</em> as a cell observed before and after antibiotic exposure that did not surpass the DsRed death threshold.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-cell-distribution-across-experiments" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-cell-distribution-across-experiments-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.1: <strong>Cell classification and its distribution across experiments.</strong> We define a <em>filamented cell</em> as a cell whose length exceeded two standard deviations from the mean at any time during the experiment. A <em>surviving cell</em> is a cell we observed before and after exposure to the antibiotic and did not surpass the DsRed death threshold. Accordingly, we removed from the analysis those cells that died before or were born after antibiotic exposure. Therefore, we delimited the effect caused by antibiotic exposure.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="results" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="results"><span class="header-section-number">2.3</span> Results</h2>
<section id="sec-length-gfp-crucial" class="level3" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="sec-length-gfp-crucial"><span class="header-section-number">2.3.1</span> Cell length and the amount of GFP are crucial in determining cell survival</h3>
<p>We evaluated the DsRed, GFP, and length values for each cell at different time points: initial, filamentation, and end. This preprocessing allowed us to observe and quantify each cell at critical times in the experiment and eliminate noise or signals outside the scope of this investigation.</p>
<p>We define the <em>initial time</em> as the first time we observed the cell in the experiment. <em>Filamentation time</em> equals when a cell reaches the filamentation threshold (see <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>) for the first time. We defined the <em>end time</em> as the time of the last observation of the cell. We decided to bound the end time for surviving cells to one frame (10 min) after the end of antibiotic exposure so that the observed signal would reflect the final stress responses.</p>
<p>When we compared the distributions of DsRed, GFP, and length for both experiments, we observed the changes in their role in cell survival. In <a href="#fig-dsred-temporal-distribution">Figure&nbsp;<span>2.2</span></a>, we show that indistinctly and, as expected, surviving cells managed to eliminate the antibiotic by the end time. In contrast, dead cells presented higher levels of antibiotics (measured by proxy through the mean DsRed intensity of the cell).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-dsred-temporal-distribution" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-dsred-temporal-distribution-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.2: <strong>DsRed temporal distribution.</strong> To evaluate the incident effect of the antibiotic marked by DsRed on cells by class, we show its values at three key moments: start, filamentation (SOS), and end. The upper asterisks represent the significance value when comparing a group X to the filamented and surviving cell reference. Asterisks in a line indicate whether or not there is a significant difference in the survival of non-filamented cells. Dots represent the mean of each group. The line bars represents the distribution of the data. Although, at the initial time, we observe multiple significant differences, this is likely due to the intrinsic noise of the system since, as expected, the values are close to zero. We observed a difference between the surviving and non-filamented cells for the chromosomal strain for the SOS time, but the same did not occur for the plasmid strain. The final amount of DsRed makes a clear difference between survival and death.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>On the other hand, GFP observations in <a href="#fig-gfp-temporal-distribution">Figure&nbsp;<span>2.3</span></a> showed us that filamented cells had low fluorescent intensities (low plasmid copy-number) at the beginning of the experiment. In comparison, the chromosomal strain did not exhibit noticeable changes in GFP levels. For the final observation times, GFP measurements indicated that among the cells that did not filament, the ones that survived exhibited a reduced GFP expression concerning cells killed by the antibiotic. Meanwhile, for the filamented cells, whether surviving or dead, their GFP measurements indicated no difference at the beginning or the end of the experiment, suggesting the presence of other determinants of cell survival.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-gfp-temporal-distribution" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-gfp-temporal-distribution-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.3: <strong>GFP temporal distribution.</strong> To evaluate the incident effect of the GFP on cells by class, we used the same notation as in <a href="#fig-dsred-temporal-distribution">Figure&nbsp;<span>2.2</span></a>. The chromosomal strain exhibits variability in GFP at different time points, mainly due to experimental noise resulting from low fluorescent intensity values. As expected, filamented cells had a lower initial GFP in the plasmid strain. At the time of filamentation, there appear to be differences in fluorescence between surviving and dead cells. However, in the end time, we observed that the surviving non-filamented cells have lower GFP values than the non-filamented dead cells and alive filamented cells.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Cell length was one of the factors that GFP expression levels could not explain for cell survival. In <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>, we show that the conclusions regarding filamentation were applicable for both chromosomal and plasmid strains. For the initial times, filamented and survived cells were shorter in length than those that died but longer than not filamented cells of both classes, while non-filamented cells did not differ. We observed no length differences between cells at filamentation time. Thus, survival could depend on other factors, such as growth rate. In the final time, the results were well-defined. Surviving cells had a greater length than their non-surviving pair (<em>i.e.</em>, dead filamented and non-filamented cells). However, for filamented cells, surviving cells generally represent a distribution of higher final length values but are not as extensive as their dead counterpart. Which we could explain as a length limit to which cells can grow without dying. Nevertheless, we had no information to evaluate such a hypothesis.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-length-temporal-distribution" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-length-temporal-distribution-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.4: <strong>Length temporal distribution.</strong> To evaluate the incident effect of length on cells by class, we use the same notation as in <a href="#fig-dsred-temporal-distribution">Figure&nbsp;<span>2.2</span></a>. The observations for both strains, chromosomal or plasmid, are the same. In the beginning, the surviving filamented cells already have a difference in length from the rest of the classes. At the time of filamentation, there is no difference to help determine whether the cell will survive or not. Finally, in the final time, it seems that the surviving filamented cells have a greater length than the rest of the groups. However, this length is moderate compared to the excess length shown by non-surviving filamented cells. On the other hand, we highlighted the growth of the surviving non-filamented cells. Therefore, although they did not reach a length for us to classify as filamented, the cells did resort to filamentation.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Once we observed the effects of GFP expression levels and lengths in determining whether a cell lives or dies, we projected the cells onto the plane. We painted them with their class status (See <a href="#fig-cell-distribution-across-experiments">Figure&nbsp;<span>2.1</span></a>) to determine whether these two variables contained the necessary information to cluster the data correctly. In <a href="#fig-just-initial-values">Figure&nbsp;<span>2.5</span></a>, we show the initial GFP and length values projection. While, with some work, we could contextually place the results in <a href="#fig-gfp-temporal-distribution">Figure&nbsp;<span>2.3</span></a> and <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>, the initial values did not appear to determine the classes. Therefore, we explored the final versus initial values differences in <a href="#fig-metric-differences">Figure&nbsp;<span>2.6</span></a>. With this new representation of the cells in the plane, we contextualized the statistical results presented in <a href="#fig-gfp-temporal-distribution">Figure&nbsp;<span>2.3</span></a> and <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>. Besides, it showed us that differences in length (<em>i.e.</em>, filamentation) and reductions in GFP expression are essential in determining cell survival. Though the clustering of cell status is not entirely separated, other variables affect the experimental results in cell survival.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-just-initial-values" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-just-initial-values-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.5: <strong>Experiment’s initial values.</strong> By positioning a cell in space based on its initial length and GFP values, we can see that class separation occurs, but not as a strong signal. Therefore, we concluded that although the initial state influences the result, this is not everything. For this, we have the example of the length changes throughout the experiment caused by filamentation. In this graph, the GFP scale is at log10 to help us observe those minor differences between the experiments.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-metric-differences" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-metric-differences-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.6: <strong>Experiment’s initial values differences.</strong> By comparing the metric differences of the last observation and the first observation of a cell, we can separate mainly the surviving filamented cells from those that did not do it in both experiments (blue dots). Meanwhile, cells with plasmids form a small accumulation of surviving cells that did not produce filament (green dots). However, this has made a breakthrough in understanding what is affecting cell survival. There are still variables that we can include to understand this phenomenon better.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="number-of-divisions-and-cell-age-do-not-appear-to-play-a-clear-role-in-determining-cell-survival" class="level3" data-number="2.3.2">
<h3 data-number="2.3.2" class="anchored" data-anchor-id="number-of-divisions-and-cell-age-do-not-appear-to-play-a-clear-role-in-determining-cell-survival"><span class="header-section-number">2.3.2</span> Number of divisions and cell age do not appear to play a clear role in determining cell survival</h3>
<p>In <a href="#sec-length-gfp-crucial"><span>Section&nbsp;2.3.1</span></a>, we explored the effect on cell survival through GFP variability and cell length.However, <a href="#fig-just-initial-values">Figure&nbsp;<span>2.5</span></a> and <a href="#fig-metric-differences">Figure&nbsp;<span>2.6</span></a> showed us the possibility of other factors relevant to the phenomenon under study. As some papers in the literature suggest, some of these other factors may be cell division and chronological age (<em>i.e.</em>, how much time has passed since the last cell division at the time of exposure to a toxic agent) <span class="citation" data-cites="moger-reischer2019 roostalu2008 heinrich2015">(<a href="99-references.html#ref-moger-reischer2019" role="doc-biblioref">Moger-Reischer and Lennon 2019</a>; <a href="99-references.html#ref-roostalu2008" role="doc-biblioref">Roostalu et al. 2008</a>; <a href="99-references.html#ref-heinrich2015" role="doc-biblioref">Heinrich, Leslie, and Jonas 2015</a>)</span>. Therefore, we chose to observe these two metrics in experiments at a purely qualitative level, i.e., without the inclusion of, e.g., metrics of membrane or cell cycle properties <span class="citation" data-cites="Joseleau-Petit1999">(<a href="99-references.html#ref-Joseleau-Petit1999" role="doc-biblioref">Joseleau-Petit, Vinella, and D’Ari 1999</a>)</span>.</p>
<p>Although we expected to see a small contribution, either by the number of divisions or cell age, in <a href="#fig-number-divisions">Figure&nbsp;<span>2.7</span></a> and <a href="#fig-time-since-last-division">Figure&nbsp;<span>2.8</span></a>, we could not observe a precise effect of these variables on cell survival. Although they could have an explanation or biological significance, we decided to omit as relevant in the characterization of our cells, since the signal was not clear. However, we derived from this analysis a slightly simpler variable that tells us whether a cell underwent a cell division event or not. So it gives us a more generalized picture of the contribution of division to cell survival (see <a href="#fig-plasmid-pca-variable-contribution">Figure&nbsp;<span>2.14</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-number-divisions" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-number-divisions-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.7: <strong>Cell’s number of divisions.</strong> Both chromosomal and plasmid cells exhibited a wider distribution of divisions for the surviving cells against non-surviving cells. However, we did not appreciate a significant change between the chromosome filamented cells. Therefore, the number of cell divisions’ contribution to filamentation remains uncertain.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-time-since-last-division" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-time-since-last-division-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.8: <strong>Time elapsed since the last division at the beginning of the experiment.</strong> The mean time of the last division before starting the experiment indicates that it did not influence the final result for chromosomal cells. There is a slight difference between the filamented-not survived cells and the rest for cells with plasmids. However, the signal does not appear to be strong on the survival role. Therefore, we conclude that we have no evidence to support that the time of the last division at the beginning of the experiment influences the final classification results.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="time-to-reach-filamentation-matters-in-determining-cell-survival" class="level3" data-number="2.3.3">
<h3 data-number="2.3.3" class="anchored" data-anchor-id="time-to-reach-filamentation-matters-in-determining-cell-survival"><span class="header-section-number">2.3.3</span> Time to reach filamentation matters in determining cell survival</h3>
<p>In <a href="#fig-dsred-temporal-distribution">Figure&nbsp;<span>2.2</span></a>, <a href="#fig-gfp-temporal-distribution">Figure&nbsp;<span>2.3</span></a>, and <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>, we showed how, at the time of filamentation, DsRed and GFP levels appeared indifferent to the cells. Therefore, we hypothesized that a possible variable determining cell survival could be its time to activate its anti-stress response system that causes filamentation. Furthermore, we also guided our hypothesis by previous reports showing us how the gene expression level can induce filamentation with tight temporal coordination [x].</p>
<p>While, for our analyses, we did not measure the concentration of antibiotic that triggers filamentation per se, we indirectly quantified its effect by using the time it took for a cell to reach a length at which it is already considered a filamentating cell. Furthermore, to recognize that the observed effect was a product of the experiment, we decided to keep only filamented cells just once antibiotic exposure began.</p>
<p><a href="#fig-time-to-filamentation-filtered">Figure&nbsp;<span>2.9</span></a> shows how filamentation times are narrower for chromosomal cells than for plasmid-bearing cells. Then, we hypothesize that the effect could come from the heterogeneity in the plasmid copy number in the population. Also, interestingly, we observed that, for both experiments, cells that survived had longer filamentation times than those that died. These differences in response times suggest the following: 1) if the cell grows too fast, it will reach a limit and start to accumulate antibiotics constantly, and 2) if the cell grows too fast, likely, the cost of maintaining an ample length for prolonged periods of exposure will become counterproductive.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-time-to-filamentation-filtered" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-time-to-filamentation-filtered-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.9: <strong>Time to filamentation filtered.</strong> We only keep cells that filamented during the antibiotic exposure to quantify their time to filamentation and its effect on survival. In this way, we normalize the start times for the calculation of the filamentation time. For both strains, the filamentation time had a more significant delay in the surviving cells.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In Figure <a href="#fig-initial-values-with-time">Figure&nbsp;<span>2.10</span></a>, we decided to project the results of Figure <a href="#fig-time-to-filamentation-filtered">Figure&nbsp;<span>2.9</span></a> in a space similar to the one described in Figure <a href="#fig-just-initial-values">Figure&nbsp;<span>2.5</span></a>). Thus, we separated our data into cells that survived and cells that did not and painted them when it took them to reach their filamented state. We realized that, by adding this temporal component to the initial variables of length and GFP, we could separate surviving cells from dead cells to a greater degree. However, it may still not be enough, and there are still many other variables that play a crucial role in understanding the ecology of stress and how some cells will be survivors or not.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-initial-values-with-time" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-initial-values-with-time-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.10: <strong>Experiment initial values with time to filamentation.</strong> As in <a href="#fig-just-initial-values">Figure&nbsp;<span>2.5</span></a>, including the time it will take for cells to filament allows us to better understand the phenomenon of survival. Cells that filamented and survived generally have a much higher delay than their non-filamented peers for both strains (see <a href="#fig-time-to-filamentation-filtered">Figure&nbsp;<span>2.9</span></a>).</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="sec-unsupervised-classification" class="level3" data-number="2.3.4">
<h3 data-number="2.3.4" class="anchored" data-anchor-id="sec-unsupervised-classification"><span class="header-section-number">2.3.4</span> Increasing the system’s complexity and analyzing it in an unsupervised way allows a correct classification of cell states</h3>
<p>In the experiments, we observed the importance of GFP filamentation and variability for cell survival. Similarly, we realized that other variables must be affecting the final results. Filamentation and GFP variability alone did not fully recapitulate the expected behavior of the data. That is, the target variables did not capture the system’s heterogeneity.</p>
<p>The inability to reproduce cell classification led us to question two things: 1) the possibility that our sorting was wrong beforehand, and 2) we did not have enough variables to capture the study phenomenon. We decided to take the unsupervised learning way to answer these subjects because it allows us to project our data without prior knowledge.</p>
<p>We opted for the path of dimensionality reduction techniques where each variable or feature is equivalent to one dimension. The essence of dimensionality reduction is that it is not feasible to analyze each dimension with many dimensions. Furthermore, dimensionality reduction helps us counteract several problems, such as reducing the complexity of a model, reducing the possibility of overfitting a model, removing all correlated variables, and visualizing our data in a two- or three-dimensional space for better appreciation. Improved visualization and identification of essential variables are the main reasons to guide and complement our research with this technique.</p>
<section id="principal-component-analysis-pca-emphasizes-the-importance-of-cell-length-and-its-gfp-in-cell-survival" class="level4" data-number="2.3.4.1">
<h4 data-number="2.3.4.1" class="anchored" data-anchor-id="principal-component-analysis-pca-emphasizes-the-importance-of-cell-length-and-its-gfp-in-cell-survival"><span class="header-section-number">2.3.4.1</span> Principal Component Analysis (PCA) emphasizes the importance of cell length and its GFP in cell survival</h4>
<div class="cell">

</div>
<p>The first dimensionality reduction technique we decided to use was Principal Component Analysis (PCA) <span class="citation" data-cites="pearson1901 hotelling1936">(<a href="99-references.html#ref-pearson1901" role="doc-biblioref">Pearson 1901</a>; <a href="99-references.html#ref-hotelling1936" role="doc-biblioref">Hotelling 1936</a>)</span>. Scientist mainly uses PCA to create predictive models or in Exploratory Data Analysis (EDA). In our case, we only use it as an EDA.</p>
<p>For chromosomal and plasmid strain, in <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a> and <a href="#fig-plasmid-pca-new-coordinates">Figure&nbsp;<span>2.12</span></a>, we show the projection of the first two principal components (PCs), respectively. <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a> separates the manually annotated classes, surviving cells separated from non-surviving cells. However, for <a href="#fig-plasmid-pca-new-coordinates">Figure&nbsp;<span>2.12</span></a>, the class separation was a bit rougher but allowed us to separate the surviving filament cells from the dead ones.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-chromosome-pca-new-coordinates" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-chromosome-pca-new-coordinates-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.11: <strong>Principal Component Analysis of chromosomal strain.</strong> When integrating the information of different variables in a dimensionality reduction analysis, we observed a clear separation between the surviving cells and those that did not. The contributions that determined this phenomenon come mainly from the last amount of DsRed, GFP, and cell length (see <a href="#fig-chromosome-pca-variable-contribution">Figure&nbsp;<span>2.13</span></a>). Although it seems obvious, it effectively confirms that the temporal classification that we carry out makes sense. Longer length represents a greater uptake of antibiotics, but in a much larger volume, so the net effect is an internal reduction of antibiotics (see <a href="03-model-analysis.html#fig-cell-dimensions-relationship">Figure&nbsp;<span>3.1</span></a>).</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-plasmid-pca-new-coordinates" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-plasmid-pca-new-coordinates-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.12: <strong>Principal Component Analysis of plasmid strain.</strong> By integrating the information from different variables in a dimensionality reduction analysis, we observed a clear separation between the filamented and non-filamented cells. Said class separation is given by component 2 (Y-axis), which is determined primarily by the initial and final lengths of the cells (see <a href="#fig-plasmid-pca-variable-contribution">Figure&nbsp;<span>2.14</span></a>). Furthermore, the classification also allows us to separate those filamented cells that died from those that survived. Therefore, despite the increase in the system’s complexity, length plays a role in determining survival.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>For their part, in <a href="#fig-chromosome-pca-variable-contribution">Figure&nbsp;<span>2.13</span></a> and <a href="#fig-plasmid-pca-variable-contribution">Figure&nbsp;<span>2.14</span></a>, we show the total contribution of each variable per PC for the chromosomal and plasmid strain, respectively. Finding that filamentation plays a crucial role in determining cell survival. For example, for PC2, we appreciated how the variable end DsRed directed the dots to the positive side, while the variable end and start length directed the dots to the opposing side. Therefore, we can support that filamentation has a role in moving cells away from having higher amounts of DsRed.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-chromosome-pca-variable-contribution" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-chromosome-pca-variable-contribution-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.13: <strong>Variables contribution of Principal Component Analysis of chromosomal strain.</strong> In <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a>, we see that the classes we created manually reflected what we observed when performing a reduction of dimensions analysis. Here we show the individual contribution of each variable for the first two components. The variables that most affected components 1 and 2 (X-axis and Y-axis, respectively) are the final measurements of DsRed, GFP, length, and the initial amount of GFP. Given that they are chromosomal strains, we should note that this variability could be produced by intrinsic experimental noise that we could not remove. With that in mind, having the DsRed and the final length highlights the inherent role of cells by having increased their size.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-plasmid-pca-variable-contribution" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-plasmid-pca-variable-contribution-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.14: <strong>Variables contribution of Principal Component Analysis of plasmid strain.</strong> In <a href="#fig-plasmid-pca-new-coordinates">Figure&nbsp;<span>2.12</span></a>, we saw that we could separate the filamented cells from the non-filamented ones. The reduction analysis also shows a slight difference between surviving and dead cells within the small group of filamented cells. Here we offer the individual contribution of each variable for the first two components. For the first component (x-axis in <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a>), the initial and final GFP measurements mainly received the component’s variability. We expected this component’s importance since it is a chromosomal strain, so we hope its inherent variation will be inherited. On the other hand, the second component (Y-axis in <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a>) was determined by the length of the cell. Factors that, in the chromosomal strain (see <a href="#fig-chromosome-pca-variable-contribution">Figure&nbsp;<span>2.13</span></a>), determined with the help of DsRed the separation between surviving and dead cells.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="uniform-manifold-approximation-and-projection-umap-correctly-represents-the-local-structure-of-cell-states" class="level4" data-number="2.3.4.2">
<h4 data-number="2.3.4.2" class="anchored" data-anchor-id="uniform-manifold-approximation-and-projection-umap-correctly-represents-the-local-structure-of-cell-states"><span class="header-section-number">2.3.4.2</span> Uniform Manifold Approximation and Projection (UMAP) correctly represents the local structure of cell states</h4>
<p>Staying with only a one-dimensionality reduction technique was not an option, so we used the UMAP technique <span class="citation" data-cites="mcinnes2018umap">(<a href="99-references.html#ref-mcinnes2018umap" role="doc-biblioref">McInnes, Healy, and Melville 2018</a>)</span>. We mainly decided to use UMAP for clustering purposes and see if the annotated clusters corresponded to the manually annotated ones. UMAP has certain advantages for these purposes, e.g., it preserves the global structure across the whole space, so the distances between clusters matter.</p>
<p>In <a href="#fig-chromosome-umap-new-coordinates">Figure&nbsp;<span>2.15</span></a> and <a href="#fig-plasmid-umap-new-coordinates">Figure&nbsp;<span>2.16</span></a>, we show how, using the same variables used in the “PCA” section, UMAP accomplished clustering the four proposed classes correctly. Interestingly, in <a href="#fig-chromosome-umap-new-coordinates">Figure&nbsp;<span>2.15</span></a>, UMAP formed three general groups and four for <a href="#fig-plasmid-umap-new-coordinates">Figure&nbsp;<span>2.16</span></a>. However, in general, UMAP clustered the surviving cells from those that did not survive. On investigating why this separation occurred, we found that the large groups coalesced into one another if we eliminated the division variable. So, in a way, the division also has a role in determining survival, but it is not essential or at least not over-represented in our data.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-chromosome-umap-new-coordinates" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-chromosome-umap-new-coordinates-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.15: <strong>UMAP coordinates of chromosome strain.</strong> We represented the cells in a low-dimensional space. This new projection allowed it to group the cells that survived and those that did not. Therefore, as in PCA <a href="#fig-chromosome-pca-new-coordinates">Figure&nbsp;<span>2.11</span></a>, this technique supports the manual classification that we carry out.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-plasmid-umap-new-coordinates" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-plasmid-umap-new-coordinates-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.16: <strong>UMAP coordinates of plasmid strain.</strong> As in <a href="#fig-chromosome-umap-new-coordinates">Figure&nbsp;<span>2.15</span></a>, the representation in a low-dimensional space helped classify the cells into four groups, two survivors and two non-survivors. The variable <em>division</em> marks the separation of classes. The <em>division</em> variable indicates whether a cell divided during its lifetime or not. Together, the UMAP represents the manually assigned classes.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="population-dynamics-reveal-how-filamentation-contributes-cell-survival" class="level3" data-number="2.3.5">
<h3 data-number="2.3.5" class="anchored" data-anchor-id="population-dynamics-reveal-how-filamentation-contributes-cell-survival"><span class="header-section-number">2.3.5</span> Population dynamics reveal how filamentation contributes cell survival</h3>
<div class="cell">

</div>
<p>From the full tracking dataset, we evaluated how the different cell states behaved over time—for example, understanding how the cells absorbed antibiotics or how they elongated in time. In contrast to the dataset generated in the <a href="#sec-length-gfp-crucial"><span>Section&nbsp;2.3.1</span></a>, we did not truncate the results 10 minutes after the antibiotic exposure. In this way, we could observe cell behavior before and after the presence of the toxic agent.</p>
<p>In <a href="#fig-status-with-dead">Figure&nbsp;<span>2.17</span></a>, we observed a small fraction of filamentous cells without exposure to the toxic agent in both cell strains. However, after antibiotic exposure at minute 60, we observed increases in the proportion of filamented cells. It is interesting to note how filamented cells grew after antibiotic exposure for the chromosomal strain. We speculate that this post-antibiotic growth exists because, once the SOS system that triggers filamentation is activated, the system continues to grow until it reaches a limit regardless of whether the damaging agent is still present <span class="citation" data-cites="justiceMorphologicalPlasticityBacterial2008 mückl2018">(<a href="99-references.html#ref-justiceMorphologicalPlasticityBacterial2008" role="doc-biblioref">Justice et al. 2008</a>; <a href="99-references.html#ref-mückl2018" role="doc-biblioref">Mückl et al. 2018</a>)</span>. Moreover, we observed how the cells start to divide again after some time because the proportion of non-filament cells starts to grow while the filament cells start to divide. We observed the same effects for the plasmid strain. However, the number of filament cells expected was much lower by experimental design.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-status-with-dead" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-status-with-dead-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.17: <strong>Population status over time.</strong> We calculate how many cells of each type existed for each time point: non-filamented and filamented living cells (green and orange areas, respectively) and dead cells (red area; we considered <em>dead</em> cells as those that existed at one time and then stopped tracking). The gray vertical lines represent each experiment’s start and end of antibiotic exposure. The experiment was finalized with the resolution of the cells when they returned to their non-filamented state. The effect of filamentation and its spread after exposure to the antibiotic is evident for the chromosomal strain. For its part, for the plasmid strain, it is observed how the filamented cells begin to appear slowly. Their proportion is as expected, given that the population had a wide distribution of GFP that allowed them to combat exposure to the antibiotic.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In <a href="#fig-metrics-over-time">Figure&nbsp;<span>2.18</span></a>, we showed that once antibiotics exposure began, those cells that died had a much faster increase in DsRed than those that did manage to live, regardless of whether they were filamented. On the other hand, surviving cells maintained their relatively stable DsRed levels. We noted that length was critical for the surviving cells for the chromosomal strain by turning to the GFP and length variables for a temporal explanation. Even cells categorized as non-filamented reached the filamentation threshold minutes after antibiotic exposure. However, the distinction between live or dead filamented cells was not as evident as expected. As for cells with plasmids, the effect on GFP for surviving cells was maintained for filamented cells and decreased for non-filamented cells. For the filament cells that died, we showed that they had, on average, a much longer initial length than the surviving cells. We also consider it necessary to understand which variables affect cell survival.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-metrics-over-time" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-metrics-over-time-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.18: <strong>Population measurements over time.</strong> The colored lines symbolize the average value of each metric at each instant of time, while its surrounding shaded area represents the 95% confidence interval. The vertical lines represent the start and end of antibiotic exposure. The horizontal line in the length metric symbolizes the threshold to consider a cell filament. Regarding the GFP metric, the behavior is relatively stable for the chromosomal strain. We observed a faster increase of DsRed for the non-surviving populations in both experiments. In contrast, for the plasmid strain, a decline in GFP is observed for the population that did not survive. For the length metric, it is interesting to note how the chromosome cells that did not filament continued to grow past the filamentation threshold once the exposure to the antibiotic in the chromosomal strain had ended. On the other hand, the filamented and dead cells seem to have a greater length from the beginning for the plasmid strain.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heterogeneity-in-plasmid-copy-number-allows-various-forms-of-survival-in-addition-to-filamentation" class="level3" data-number="2.3.6">
<h3 data-number="2.3.6" class="anchored" data-anchor-id="heterogeneity-in-plasmid-copy-number-allows-various-forms-of-survival-in-addition-to-filamentation"><span class="header-section-number">2.3.6</span> Heterogeneity in plasmid copy-number allows various forms of survival in addition to filamentation</h3>
<p>We are confident that filamentation has a fundamental role in determining cell survival, as we have shown so far. However, for plasmid cells, we have a component of our complete interest; heterogeneity. Each cell can possess a different plasmid copy number; thus, each could show a different behavior under stress <span class="citation" data-cites="sanmillan2016">(<a href="99-references.html#ref-sanmillan2016" role="doc-biblioref">San Millan et al. 2016</a>)</span>. For instance, heterogeneity can produce resistant cells that do not suffer damage, susceptible cells, and cells that form filaments to mitigate environmental stress.</p>
<p>To study the effect of variability in plasmid copy number on the survival probability of the population, we decided to group cells by the proportion of initial GFP with respect to the population maximum. We defined 100% of the population as the number of total cells at the onset of antibiotic exposure. <a href="#fig-proportion-living-cells-gfp-by-row">Figure&nbsp;<span>2.19</span></a> shows how the cells with the highest amount of GFP remained unchanged once antibiotic exposure began, while the rest of the cells started to decrease their percentage of surviving cells. However, the decrease was not linear. On the contrary, we observed a bi-modal distribution in the reduction of live cells. An average GFP point provided higher survival than a point below or above the average (except for cells very close to the population maximum).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-proportion-living-cells-gfp-by-row" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-proportion-living-cells-gfp-by-row-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.19: <strong>Population survivals binned by initial GFP over time.</strong> We categorized the cells’ GFP into ranges of proportions 0.05 concerning the maximum amount of GFP in the population. 100% cells per bin of GFP was taken as the number of cells one frame before the start of exposure to the antibiotic (minute 50). Therefore, dark to light colors represent a generation of new cells, and light to dark colors the death of cells. The black vertical bars represent the start and end of the antibiotic exposure. Bars size and color on the right represent the percentage of the living cells 10 minutes after the end of the experiment. As shown in <a href="#fig-gfp-survival-probability">Figure&nbsp;<span>2.20</span></a>, we showed that the surviving cells appear to follow something similar to a bimodal distribution. More cells survive with a moderate amount of GFP or with an amount close to the maximum of the population.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Therefore, what we observed was a bimodal distribution for GFP-dependent cell survival. In order to show this effect more clearly, in <a href="#fig-gfp-survival-probability">Figure&nbsp;<span>2.20</span></a>, we plotted the survival probability for each GFP bin without normalizing for the population maximum. This new plot allowed us to observe how the bimodal survival distribution occurs for cells that did not grow as filaments, whereas cells that filament increase their survival probability gradually as they have more initial GFP (see also <a href="#fig-gfp-temporal-distribution">Figure&nbsp;<span>2.3</span></a>).-distribution).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-gfp-survival-probability" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-gfp-survival-probability-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.20: <strong>Plasmid initial GFP survival probability.</strong> We calculated the survival probability after comparing the population distributions of GFP with those of the cells that managed to survive. To assess survival by GFP, we only used plasmid cells. For non-filamented cells (blue dots), a bell forms with an upturned tail. On the other hand, for the filamented cells (red dots), a continuous increase in survival is shown just when it seems that the probability of the non-filamented cells has decreased. In global, much GFP has higher resistance, but an average GFP value without filamentation also increases the probability of survival.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>As in <a href="#fig-gfp-survival-probability">Figure&nbsp;<span>2.20</span></a>, in <a href="#fig-length-survival-probability">Figure&nbsp;<span>2.21</span></a>, we show the survival probability given an initial length. We observe that survival is higher for cells that did not grow as filaments if the initial length was less than the average. In contrast, for filamented cells, the survival probability increased as cell length was longer at the beginning of the experiment (see also <a href="#fig-length-temporal-distribution">Figure&nbsp;<span>2.4</span></a>). However, it is noteworthy that the probability of survival had a limit in which a higher initial length meant a lower probability of survival (see red dotted lines in <a href="#fig-length-survival-probability">Figure&nbsp;<span>2.21</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-length-survival-probability" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="02-experiment-analysis_files/figure-html/fig-length-survival-probability-1.svg" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2.21: <strong>Plasmid initial length survival probability.</strong> We calculated the survival probability after comparing the population distributions of length with those of the cells that managed to survive. For non-filamented cells (blue dots), the survival probability is higher for those cells with small initial lengths, while It seems to decrease with a more extensive initial size. For their part, for filamented cells (red dots), the probability of survival increases according to their length but then declines when the cells are too long at first (see red dotted line). Therefore, in general, a small and moderate length or an initial length already filamented from the beginning increases the chances of survival.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="discussion" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="discussion"><span class="header-section-number">2.4</span> Discussion</h2>
<p>Here, we evaluated different variables that could determine cell survival upon exposure to toxic agents by studying two experimental populations of <em>E. coli</em>, one strain with a resistance gene on the chromosome and the other on multicopy plasmids. We identified two variables that are predominantly responsible for cell survival: cell length and GFP amount related to the cell’s inherent resistance to the toxic agent and heterogeneity in response times.</p>
<p>On the other hand, as other studies have already mentioned <span class="citation" data-cites="heinrich2015 wang2009">(<a href="99-references.html#ref-heinrich2015" role="doc-biblioref">Heinrich, Leslie, and Jonas 2015</a>; <a href="99-references.html#ref-wang2009" role="doc-biblioref">Wang and Levin 2009</a>)</span>, we examined cell activity and youth in a minimalistic way. While the distribution of the number of divisions exemplifies a broader and more uniform range for the surviving cells, the cells that died tended to have fewer divisions. However, for the study of cellular youth at the time of exposure to the toxic agent, the results did not show a clear pattern of behavior for cell fate determination. Therefore, it would be interesting to study cellular youth at a higher level of complexity in future studies to understand its contribution to cell survival.</p>
<p>Interestingly, when we used temporal measurements of cell length, GFP, DsRed, and if a cell divided, we could recapitulate, for the most part, the fates of cellular states (see <a href="#sec-length-gfp-crucial"><span>Section&nbsp;2.3.1</span></a> and <a href="#sec-unsupervised-classification"><span>Section&nbsp;2.3.4</span></a>). Thus, increasing the system’s complexity led to better clustering of cell states, but not how these factors interact biologically in determining cell survival. Therefore, we decided to postulate a mathematical model that helps us understand the critical components of cell survival.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-bruggeman" class="csl-entry" role="doc-biblioentry">
Bruggeman, Frank J., Jorrit J. Hornberg, Fred C. Boogerd, and Hans V. Westerhoff. 2007. <span>“Introduction to Systems Biology.”</span> In, 1–19. Birkhäuser Basel. <a href="https://doi.org/10.1007/978-3-7643-7439-6_1">https://doi.org/10.1007/978-3-7643-7439-6_1</a>.
</div>
<div id="ref-heinrich2015" class="csl-entry" role="doc-biblioentry">
Heinrich, Kristina, David J. Leslie, and Kristina Jonas. 2015. <span>“Modulation of Bacterial Proliferation as a Survival Strategy.”</span> In, 127–71. Elsevier. <a href="https://doi.org/10.1016/bs.aambs.2015.02.004">https://doi.org/10.1016/bs.aambs.2015.02.004</a>.
</div>
<div id="ref-hotelling1936" class="csl-entry" role="doc-biblioentry">
Hotelling, Harold. 1936. <span>“Relations Between Two Sets of Variates.”</span> <em>Biometrika</em> 28 (3/4): 321. <a href="https://doi.org/10.2307/2333955">https://doi.org/10.2307/2333955</a>.
</div>
<div id="ref-Joseleau-Petit1999" class="csl-entry" role="doc-biblioentry">
Joseleau-Petit, Daniel̀e, Daniel Vinella, and Richard D’Ari. 1999. <span>“Metabolic Alarms and Cell Division in Escherichia Coli.”</span> <em>Journal of Bacteriology</em> 181 (1): 9–14. <a href="https://doi.org/10.1128/jb.181.1.9-14.1999">https://doi.org/10.1128/jb.181.1.9-14.1999</a>.
</div>
<div id="ref-justiceMorphologicalPlasticityBacterial2008" class="csl-entry" role="doc-biblioentry">
Justice, Sheryl S., David A. Hunstad, Lynette Cegelski, and Scott J. Hultgren. 2008. <span>“Morphological Plasticity as a Bacterial Survival Strategy.”</span> <em>Nature Reviews Microbiology</em> 6 (2): 162–68. <a href="https://doi.org/10.1038/nrmicro1820">https://doi.org/10.1038/nrmicro1820</a>.
</div>
<div id="ref-kroemer2008" class="csl-entry" role="doc-biblioentry">
Kroemer, G, L Galluzzi, P Vandenabeele, J Abrams, E S Alnemri, E H Baehrecke, M V Blagosklonny, et al. 2008. <span>“Classification of Cell Death: Recommendations of the Nomenclature Committee on Cell Death 2009.”</span> <em>Cell Death &amp; Differentiation</em> 16 (1): 3–11. <a href="https://doi.org/10.1038/cdd.2008.150">https://doi.org/10.1038/cdd.2008.150</a>.
</div>
<div id="ref-mcinnes2018umap" class="csl-entry" role="doc-biblioentry">
McInnes, Leland, John Healy, and James Melville. 2018. <span>“Umap: Uniform Manifold Approximation and Projection for Dimension Reduction.”</span> <em>arXiv Preprint arXiv:1802.03426</em>.
</div>
<div id="ref-moger-reischer2019" class="csl-entry" role="doc-biblioentry">
Moger-Reischer, Roy Z., and Jay T. Lennon. 2019. <span>“Microbial Ageing and Longevity.”</span> <em>Nature Reviews Microbiology</em> 17 (11): 679–90. <a href="https://doi.org/10.1038/s41579-019-0253-y">https://doi.org/10.1038/s41579-019-0253-y</a>.
</div>
<div id="ref-mückl2018" class="csl-entry" role="doc-biblioentry">
Mückl, Andrea, Matthaeus Schwarz-Schilling, Katrin Fischer, and Friedrich C. Simmel. 2018. <span>“Filamentation and Restoration of Normal Growth in Escherichia Coli Using a Combined CRISPRi sgRNA/Antisense RNA Approach.”</span> Edited by Mark Isalan. <em>PLOS ONE</em> 13 (9): e0198058. <a href="https://doi.org/10.1371/journal.pone.0198058">https://doi.org/10.1371/journal.pone.0198058</a>.
</div>
<div id="ref-pearson1901" class="csl-entry" role="doc-biblioentry">
Pearson, Karl. 1901. <span>“LIII. On Lines and Planes of Closest Fit to Systems of Points in Space.”</span> <em>The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science</em> 2 (11): 559–72. <a href="https://doi.org/10.1080/14786440109462720">https://doi.org/10.1080/14786440109462720</a>.
</div>
<div id="ref-roostalu2008" class="csl-entry" role="doc-biblioentry">
Roostalu, Johanna, Arvi Jõers, Hannes Luidalepp, Niilo Kaldalu, and Tanel Tenson. 2008. <span>“Cell Division in Escherichia Colicultures Monitored at Single Cell Resolution.”</span> <em>BMC Microbiology</em> 8 (1). <a href="https://doi.org/10.1186/1471-2180-8-68">https://doi.org/10.1186/1471-2180-8-68</a>.
</div>
<div id="ref-sanmillan2016" class="csl-entry" role="doc-biblioentry">
San Millan, Alvaro, Jose Antonio Escudero, Danna R. Gifford, Didier Mazel, and R. Craig MacLean. 2016. <span>“Multicopy Plasmids Potentiate the Evolution of Antibiotic Resistance in Bacteria.”</span> <em>Nature Ecology &amp; Evolution</em> 1 (1). <a href="https://doi.org/10.1038/s41559-016-0010">https://doi.org/10.1038/s41559-016-0010</a>.
</div>
<div id="ref-trevors2012" class="csl-entry" role="doc-biblioentry">
Trevors, J.T. 2012. <span>“Can Dead Bacterial Cells Be Defined and Are Genes Expressed After Cell Death?”</span> <em>Journal of Microbiological Methods</em> 90 (1): 25–28. <a href="https://doi.org/10.1016/j.mimet.2012.04.004">https://doi.org/10.1016/j.mimet.2012.04.004</a>.
</div>
<div id="ref-turing1950" class="csl-entry" role="doc-biblioentry">
TURING, A. M. 1950. <span>“I.<span></span>COMPUTING MACHINERY AND INTELLIGENCE.”</span> <em>Mind</em> LIX (236): 433–60. <a href="https://doi.org/10.1093/mind/lix.236.433">https://doi.org/10.1093/mind/lix.236.433</a>.
</div>
<div id="ref-wang2009" class="csl-entry" role="doc-biblioentry">
Wang, Jue D., and Petra A. Levin. 2009. <span>“Metabolism, Cell Growth and the Bacterial Cell Cycle.”</span> <em>Nature Reviews Microbiology</em> 7 (11): 822–27. <a href="https://doi.org/10.1038/nrmicro2202">https://doi.org/10.1038/nrmicro2202</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./01-image-processing.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./03-model-analysis.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Models to the rescue; filamentation abstraction</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Experiment analysis {#sec-experiment-analysis}</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: find-preprocessing-files-02-experiment-analysis</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>experiment_analysis_files <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">path =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data-preprocessing"</span>, <span class="st">"02-experiment-analysis"</span>),</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">glob =</span> <span class="st">"*qmd"</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="in">```{r child=experiment_analysis_files}</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: execute-preprocessing-files-02-experiment-analysis</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r clear-environment}</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: libraries</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidytext)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(embed)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-processed-data</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>cells_file <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed"</span>, <span class="st">"cells_summary.csv"</span>)</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>cells_raw_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> cells_file,</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>over_ds_red_id) <span class="sc">%&gt;%</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">divided =</span> divided_id)</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: create-factors</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>cells_df <span class="ot">&lt;-</span> cells_raw_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">filamented_id =</span> <span class="fu">factor</span>(</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> filamented_id,</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>), </span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not filamented"</span>, <span class="st">"Filamented"</span>)</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">survived =</span> <span class="fu">factor</span>(</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> survived,</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>),</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not survived"</span>, <span class="st">"Survived"</span>)</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status =</span> <span class="fu">interaction</span>(</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>      filamented_id,</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>      survived,</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>      <span class="at">sep =</span> <span class="st">" - "</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status =</span> <span class="fu">paste0</span>(</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>      filamented_id,</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>      <span class="st">" - "</span>,</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>      survived</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status =</span> <span class="fu">factor</span>(cell_status)</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="fu">where</span>(is.character), <span class="fu">where</span>(is.factor), <span class="fu">where</span>(is.logical))</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: population-read-data</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>lineages_file <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed"</span>, <span class="st">"lineages.csv"</span>)</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>lineages_raw_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> lineages_file,</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">glimpse</span>()</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: population-create-factors</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>lineages_processed_1_df <span class="ot">&lt;-</span> lineages_raw_df <span class="sc">%&gt;%</span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">filamented_id =</span> <span class="fu">factor</span>(</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> filamented_id,</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>),</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not filamented"</span>, <span class="st">"Filamented"</span>)</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>    <span class="at">filamented_at_time =</span> <span class="fu">factor</span>(</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> filamented_at_time,</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>),</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not filamented"</span>, <span class="st">"Filamented"</span>)</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>    <span class="at">survived =</span> <span class="fu">factor</span>(</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> survived,</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>),</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not survived"</span>, <span class="st">"Survived"</span>)</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status =</span> <span class="fu">interaction</span>(</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>      filamented_id,</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>      survived,</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>      <span class="at">sep =</span> <span class="st">" - "</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.character</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.factor</span>()</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">glimpse</span>()</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-default-plot-style</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>cell_status_pallete <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Filamented - Not survived"</span> <span class="ot">=</span> <span class="st">"#dd5129"</span>, </span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Filamented - Survived"</span> <span class="ot">=</span> <span class="st">"#0f7ba2"</span>, </span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Not filamented - Survived"</span> <span class="ot">=</span> <span class="st">"#43b284"</span>, </span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Not filamented - Not survived"</span><span class="ot">=</span> <span class="st">"#fab255"</span></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>cell_status_legend_order <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Not filamented - Survived"</span>,</span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Not filamented - Not survived"</span>,</span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Filamented - Survived"</span>,</span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Filamented - Not survived"</span></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: utility-functions</span></span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>parse_metrics_column <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, metric_column) {</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>      {{ metric_column }} <span class="sc">:</span><span class="er">=</span> <span class="fu">str_remove</span>(</span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>        <span class="at">string =</span> {{ metric_column }},</span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>        <span class="at">pattern =</span> <span class="st">"(.+)_"</span></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>        <span class="fu">factor</span>(</span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>          <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"first"</span>, <span class="st">"sos"</span>, <span class="st">"last"</span>),</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>          <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Initial"</span>, <span class="st">"SOS"</span>, <span class="st">"End"</span>)</span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>        <span class="fu">identity</span>()</span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a><span class="fu">## Introduction</span></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>The previous chapter (see @sec-image-processing) detailed the steps</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>necessary to extract data from a set of microfluidic images through</span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>image analysis techniques and fluorescence microscopy. Each step was</span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>instrumental in creating a dataset that was easy to explore and ask</span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>questions. With the help of computational biology, systems biology, and</span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>data analysis techniques, we could process these files to help us in the</span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a>search to find the role of filamentation in cell survival.</span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a>Computational biology and systems biology contributed to the development</span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a>of this analysis. In principle, computational biology originated after</span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a>the origin of computer science with the British mathematician and</span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>logistician Alan Turing (regularly known as the father of computing)</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@turing1950</span><span class="co">]</span>. Over time, systems biology emerged as an area that</span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a>synergistically combines models and experimental data to understand</span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a>biological processes <span class="co">[</span><span class="ot">@bruggeman</span><span class="co">]</span>. Thus, giving a step towards creating</span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>models that, in general, are phenomenological but sometimes serve to</span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a>discover new ideas about the process under study. Without the computer's</span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a>power, modern ideas and aspects of studying biological sciences would</span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a>otherwise be unthinkable.</span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a>Here, we divide the experimental analysis into two main parts: 1) at the</span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a>cell level or measurements at specific points in time and 2) at the</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a>population level and time series. The first level allowed us to identify</span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>the individual contribution of each variable understudy to determine</span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>cell survival. The second level allowed us to understand how the</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>population behaves according to the passage of time in the face of</span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>exposure to a harmful agent (in this case, beta-lactam antibiotics).</span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a>Together, both visions of the same study phenomenon allowed us to</span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>extract the main ideas for postulating a mathematical model that seeks</span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>to show how filamentation is a factor for cell survival in stressful</span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>environments (see @sec-model-analysis).</span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a><span class="fu">## General preprocessing of data {#sec-experiment-general-preprocessing}</span></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a>The raw data processing consisted mainly of creating two levels of</span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a>observation for the cells of both chromosomal strains and multicopy</span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a>plasmids. The first level is at a cell granularity, point properties.</span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>The second level consists of the cells over time, thus observing</span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a>properties at the population level. We did this because it would allow</span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a>us to understand what factors affect filamentation and why.</span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a>We normalized the fluorescence values of DsRed and GFP for both</span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>experiments based on the values observed before antibiotic exposure. It</span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a>allowed us to have a basis to work with and compare expressions between</span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>cells. In the case of DsRed environment drug concentration, we also</span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a>applied a logarithmic transformation to observe subtle changes in</span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>fluorescence intensity that would allow us to detect cell death.</span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>Ultimately, we decided to classify cells into four fundamental groups</span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>based on whether the cell filamented and survived (see</span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>@fig-cell-distribution-across-experiments). We define a *filamented</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>cell* as a cell with more than two standard deviations from the mean</span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>concerning the lengths observed before introducing antibiotics into the</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>system. On the other hand, although there are multiple ways to define</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>death from single-cell observations <span class="co">[</span><span class="ot">@trevors2012; @kroemer2008</span><span class="co">]</span>, we</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>considered a *cell dead or missing* when we stopped having information</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a>about it, either because of fluorescence in the red channel was above a</span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a>given threshold (resulting from an increase in cell membrane</span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a>permeability and the introduction of fluorescent dye into the cell) or</span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a>because it left the field of observation. Therefore, we defined a</span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a>*surviving cell* as a cell observed before and after antibiotic exposure</span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a>that did not surpass the DsRed death threshold.</span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-cell-distribution-across-experiments</span></span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Cell classification and its distribution across experiments.</span></span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Cell classification and its distribution across experiments.**</span></span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We define a *filamented cell* as a cell whose length exceeded two standard</span></span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a><span class="co">#|   deviations from the mean at any time during the experiment. A *surviving cell*</span></span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a><span class="co">#|   is a cell we observed before and after exposure to the antibiotic and did not</span></span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a><span class="co">#|   surpass the DsRed death threshold. Accordingly, we removed from the analysis</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a><span class="co">#|   those cells that died before or were born after antibiotic exposure. </span></span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Therefore, we delimited the effect caused by antibiotic exposure.</span></span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a><span class="co">#|   </span></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a>p_cells_distribution <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(experiment_id, cell_status) <span class="sc">%&gt;%</span></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(experiment_id) <span class="sc">%&gt;%</span> </span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a>    <span class="at">percentage =</span> n <span class="sc">/</span> <span class="fu">sum</span>(n) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymax =</span> <span class="fu">cumsum</span>(percentage),</span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">head</span>(ymax, <span class="sc">-</span><span class="dv">1</span>)),</span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">paste0</span>(<span class="fu">format</span>(percentage, <span class="at">digits =</span> <span class="dv">2</span>), <span class="st">"%"</span>),</span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels_position =</span> (ymax <span class="sc">+</span> ymin) <span class="sc">/</span> <span class="dv">2</span>,</span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a>    <span class="at">total_label =</span> <span class="fu">paste0</span>(<span class="st">"Total:</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">format</span>(<span class="fu">sum</span>(n), <span class="at">big.mark =</span> <span class="st">","</span>), <span class="st">" cells"</span>)</span>
<span id="cb1-268"><a href="#cb1-268" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-269"><a href="#cb1-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(</span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a>      <span class="at">ymin =</span> ymin,</span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a>      <span class="at">ymax =</span> ymax,</span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a>      <span class="at">xmin =</span> <span class="dv">3</span>,</span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a>      <span class="at">xmax =</span> <span class="dv">4</span></span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(</span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb1-281"><a href="#cb1-281" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"white"</span>,</span>
<span id="cb1-282"><a href="#cb1-282" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">fill =</span> cell_status)</span>
<span id="cb1-283"><a href="#cb1-283" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-284"><a href="#cb1-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label</span>(</span>
<span id="cb1-285"><a href="#cb1-285" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">2</span>,</span>
<span id="cb1-286"><a href="#cb1-286" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb1-287"><a href="#cb1-287" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> labels_position,</span>
<span id="cb1-288"><a href="#cb1-288" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> labels</span>
<span id="cb1-289"><a href="#cb1-289" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-290"><a href="#cb1-290" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.size =</span> <span class="cn">NA</span>,</span>
<span id="cb1-291"><a href="#cb1-291" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">3.5</span>,</span>
<span id="cb1-292"><a href="#cb1-292" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-293"><a href="#cb1-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb1-294"><a href="#cb1-294" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">y =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">label =</span> total_label),</span>
<span id="cb1-295"><a href="#cb1-295" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="fl">0.5</span>, </span>
<span id="cb1-296"><a href="#cb1-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="fl">0.5</span></span>
<span id="cb1-297"><a href="#cb1-297" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-298"><a href="#cb1-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> experiment_id) <span class="sc">+</span></span>
<span id="cb1-299"><a href="#cb1-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_polar</span>(<span class="at">theta =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb1-300"><a href="#cb1-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb1-301"><a href="#cb1-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-302"><a href="#cb1-302" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-303"><a href="#cb1-303" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-304"><a href="#cb1-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb1-305"><a href="#cb1-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-306"><a href="#cb1-306" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb1-307"><a href="#cb1-307" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-308"><a href="#cb1-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-309"><a href="#cb1-309" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-310"><a href="#cb1-310" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-311"><a href="#cb1-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-312"><a href="#cb1-312" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-313"><a href="#cb1-313" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-314"><a href="#cb1-314" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-315"><a href="#cb1-315" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-316"><a href="#cb1-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-317"><a href="#cb1-317" aria-hidden="true" tabindex="-1"></a>p_cells_distribution</span>
<span id="cb1-318"><a href="#cb1-318" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-319"><a href="#cb1-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-320"><a href="#cb1-320" aria-hidden="true" tabindex="-1"></a><span class="fu">## Results</span></span>
<span id="cb1-321"><a href="#cb1-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-322"><a href="#cb1-322" aria-hidden="true" tabindex="-1"></a><span class="fu">### Cell length and the amount of GFP are crucial in determining cell survival {#sec-length-gfp-crucial}</span></span>
<span id="cb1-323"><a href="#cb1-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-324"><a href="#cb1-324" aria-hidden="true" tabindex="-1"></a>We evaluated the DsRed, GFP, and length values for each cell at</span>
<span id="cb1-325"><a href="#cb1-325" aria-hidden="true" tabindex="-1"></a>different time points: initial, filamentation, and end. This</span>
<span id="cb1-326"><a href="#cb1-326" aria-hidden="true" tabindex="-1"></a>preprocessing allowed us to observe and quantify each cell at critical</span>
<span id="cb1-327"><a href="#cb1-327" aria-hidden="true" tabindex="-1"></a>times in the experiment and eliminate noise or signals outside the scope</span>
<span id="cb1-328"><a href="#cb1-328" aria-hidden="true" tabindex="-1"></a>of this investigation.</span>
<span id="cb1-329"><a href="#cb1-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-330"><a href="#cb1-330" aria-hidden="true" tabindex="-1"></a>We define the *initial time* as the first time we observed the cell in</span>
<span id="cb1-331"><a href="#cb1-331" aria-hidden="true" tabindex="-1"></a>the experiment. *Filamentation time* equals when a cell reaches the</span>
<span id="cb1-332"><a href="#cb1-332" aria-hidden="true" tabindex="-1"></a>filamentation threshold (see @fig-length-temporal-distribution) for the</span>
<span id="cb1-333"><a href="#cb1-333" aria-hidden="true" tabindex="-1"></a>first time. We defined the *end time* as the time of the last</span>
<span id="cb1-334"><a href="#cb1-334" aria-hidden="true" tabindex="-1"></a>observation of the cell. We decided to bound the end time for surviving</span>
<span id="cb1-335"><a href="#cb1-335" aria-hidden="true" tabindex="-1"></a>cells to one frame (10 min) after the end of antibiotic exposure so that</span>
<span id="cb1-336"><a href="#cb1-336" aria-hidden="true" tabindex="-1"></a>the observed signal would reflect the final stress responses.</span>
<span id="cb1-337"><a href="#cb1-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-338"><a href="#cb1-338" aria-hidden="true" tabindex="-1"></a>When we compared the distributions of DsRed, GFP, and length for both</span>
<span id="cb1-339"><a href="#cb1-339" aria-hidden="true" tabindex="-1"></a>experiments, we observed the changes in their role in cell survival. In</span>
<span id="cb1-340"><a href="#cb1-340" aria-hidden="true" tabindex="-1"></a>@fig-dsred-temporal-distribution, we show that indistinctly and, as</span>
<span id="cb1-341"><a href="#cb1-341" aria-hidden="true" tabindex="-1"></a>expected, surviving cells managed to eliminate the antibiotic by the end</span>
<span id="cb1-342"><a href="#cb1-342" aria-hidden="true" tabindex="-1"></a>time. In contrast, dead cells presented higher levels of antibiotics</span>
<span id="cb1-343"><a href="#cb1-343" aria-hidden="true" tabindex="-1"></a>(measured by proxy through the mean DsRed intensity of the cell).</span>
<span id="cb1-344"><a href="#cb1-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-347"><a href="#cb1-347" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-348"><a href="#cb1-348" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-dsred-temporal-distribution</span></span>
<span id="cb1-349"><a href="#cb1-349" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: DsRed temporal distribution.</span></span>
<span id="cb1-350"><a href="#cb1-350" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-351"><a href="#cb1-351" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **DsRed temporal distribution.**</span></span>
<span id="cb1-352"><a href="#cb1-352" aria-hidden="true" tabindex="-1"></a><span class="co">#|   To evaluate the incident effect of the antibiotic marked by DsRed</span></span>
<span id="cb1-353"><a href="#cb1-353" aria-hidden="true" tabindex="-1"></a><span class="co">#|   on cells by class, we show its values at three key moments: start,</span></span>
<span id="cb1-354"><a href="#cb1-354" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamentation (SOS), and end. The upper asterisks represent the</span></span>
<span id="cb1-355"><a href="#cb1-355" aria-hidden="true" tabindex="-1"></a><span class="co">#|   significance value when comparing a group X to the filamented and</span></span>
<span id="cb1-356"><a href="#cb1-356" aria-hidden="true" tabindex="-1"></a><span class="co">#|   surviving cell reference. Asterisks in a line indicate whether or not</span></span>
<span id="cb1-357"><a href="#cb1-357" aria-hidden="true" tabindex="-1"></a><span class="co">#|   there is a significant difference in the survival of non-filamented cells.</span></span>
<span id="cb1-358"><a href="#cb1-358" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Dots represent the mean of each group. The line bars represents the</span></span>
<span id="cb1-359"><a href="#cb1-359" aria-hidden="true" tabindex="-1"></a><span class="co">#|   distribution of the data. Although, at the initial time, we observe</span></span>
<span id="cb1-360"><a href="#cb1-360" aria-hidden="true" tabindex="-1"></a><span class="co">#|   multiple significant differences, this is likely due to the intrinsic</span></span>
<span id="cb1-361"><a href="#cb1-361" aria-hidden="true" tabindex="-1"></a><span class="co">#|   noise of the system since, as expected, the values are close to zero.</span></span>
<span id="cb1-362"><a href="#cb1-362" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We observed a difference between the surviving and non-filamented</span></span>
<span id="cb1-363"><a href="#cb1-363" aria-hidden="true" tabindex="-1"></a><span class="co">#|   cells for the chromosomal strain for the SOS time, but the same did not</span></span>
<span id="cb1-364"><a href="#cb1-364" aria-hidden="true" tabindex="-1"></a><span class="co">#|   occur for the plasmid strain. The final amount of DsRed makes a clear</span></span>
<span id="cb1-365"><a href="#cb1-365" aria-hidden="true" tabindex="-1"></a><span class="co">#|   difference between survival and death.</span></span>
<span id="cb1-366"><a href="#cb1-366" aria-hidden="true" tabindex="-1"></a><span class="co">#|   </span></span>
<span id="cb1-367"><a href="#cb1-367" aria-hidden="true" tabindex="-1"></a>p_temporal_dsred_distributution <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-368"><a href="#cb1-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb1-369"><a href="#cb1-369" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">contains</span>(<span class="st">"ds_red"</span>),</span>
<span id="cb1-370"><a href="#cb1-370" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"metric"</span>,</span>
<span id="cb1-371"><a href="#cb1-371" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb1-372"><a href="#cb1-372" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-373"><a href="#cb1-373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">parse_metrics_column</span>(metric) <span class="sc">%&gt;%</span> </span>
<span id="cb1-374"><a href="#cb1-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-375"><a href="#cb1-375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-376"><a href="#cb1-376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> cell_status, <span class="at">y =</span> value, <span class="at">fill =</span> cell_status, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-377"><a href="#cb1-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_eye</span>() <span class="sc">+</span></span>
<span id="cb1-378"><a href="#cb1-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-379"><a href="#cb1-379" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-380"><a href="#cb1-380" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"Not filamented - Survived"</span>, <span class="st">"Not filamented - Not survived"</span>)),</span>
<span id="cb1-381"><a href="#cb1-381" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-382"><a href="#cb1-382" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y =</span> <span class="fu">c</span>(<span class="fl">0.4</span>),</span>
<span id="cb1-383"><a href="#cb1-383" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span></span>
<span id="cb1-384"><a href="#cb1-384" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-385"><a href="#cb1-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-386"><a href="#cb1-386" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"anova"</span>,</span>
<span id="cb1-387"><a href="#cb1-387" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.x.npc =</span> <span class="fl">0.10</span>,</span>
<span id="cb1-388"><a href="#cb1-388" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.93</span></span>
<span id="cb1-389"><a href="#cb1-389" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="co"># Add global annova p-value</span></span>
<span id="cb1-390"><a href="#cb1-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-391"><a href="#cb1-391" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-392"><a href="#cb1-392" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-393"><a href="#cb1-393" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref.group =</span> <span class="st">"Filamented - Survived"</span>,</span>
<span id="cb1-394"><a href="#cb1-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-395"><a href="#cb1-395" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.80</span></span>
<span id="cb1-396"><a href="#cb1-396" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-397"><a href="#cb1-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> metric) <span class="sc">+</span></span>
<span id="cb1-398"><a href="#cb1-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-399"><a href="#cb1-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-400"><a href="#cb1-400" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-401"><a href="#cb1-401" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-402"><a href="#cb1-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-403"><a href="#cb1-403" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-404"><a href="#cb1-404" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-405"><a href="#cb1-405" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-406"><a href="#cb1-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-407"><a href="#cb1-407" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-408"><a href="#cb1-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-409"><a href="#cb1-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-410"><a href="#cb1-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-411"><a href="#cb1-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"DsRed value"</span></span>
<span id="cb1-412"><a href="#cb1-412" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-413"><a href="#cb1-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-414"><a href="#cb1-414" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-415"><a href="#cb1-415" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-416"><a href="#cb1-416" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-417"><a href="#cb1-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-418"><a href="#cb1-418" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-419"><a href="#cb1-419" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-420"><a href="#cb1-420" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-421"><a href="#cb1-421" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-422"><a href="#cb1-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-423"><a href="#cb1-423" aria-hidden="true" tabindex="-1"></a>p_temporal_dsred_distributution</span>
<span id="cb1-424"><a href="#cb1-424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-425"><a href="#cb1-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-426"><a href="#cb1-426" aria-hidden="true" tabindex="-1"></a>On the other hand, GFP observations in @fig-gfp-temporal-distribution</span>
<span id="cb1-427"><a href="#cb1-427" aria-hidden="true" tabindex="-1"></a>showed us that filamented cells had low fluorescent intensities (low</span>
<span id="cb1-428"><a href="#cb1-428" aria-hidden="true" tabindex="-1"></a>plasmid copy-number) at the beginning of the experiment. In comparison,</span>
<span id="cb1-429"><a href="#cb1-429" aria-hidden="true" tabindex="-1"></a>the chromosomal strain did not exhibit noticeable changes in GFP levels.</span>
<span id="cb1-430"><a href="#cb1-430" aria-hidden="true" tabindex="-1"></a>For the final observation times, GFP measurements indicated that among</span>
<span id="cb1-431"><a href="#cb1-431" aria-hidden="true" tabindex="-1"></a>the cells that did not filament, the ones that survived exhibited a</span>
<span id="cb1-432"><a href="#cb1-432" aria-hidden="true" tabindex="-1"></a>reduced GFP expression concerning cells killed by the antibiotic.</span>
<span id="cb1-433"><a href="#cb1-433" aria-hidden="true" tabindex="-1"></a>Meanwhile, for the filamented cells, whether surviving or dead, their</span>
<span id="cb1-434"><a href="#cb1-434" aria-hidden="true" tabindex="-1"></a>GFP measurements indicated no difference at the beginning or the end of</span>
<span id="cb1-435"><a href="#cb1-435" aria-hidden="true" tabindex="-1"></a>the experiment, suggesting the presence of other determinants of cell</span>
<span id="cb1-436"><a href="#cb1-436" aria-hidden="true" tabindex="-1"></a>survival.</span>
<span id="cb1-437"><a href="#cb1-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-440"><a href="#cb1-440" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-441"><a href="#cb1-441" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-gfp-temporal-distribution</span></span>
<span id="cb1-442"><a href="#cb1-442" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: GFP temporal distribution.</span></span>
<span id="cb1-443"><a href="#cb1-443" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-444"><a href="#cb1-444" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **GFP temporal distribution.**</span></span>
<span id="cb1-445"><a href="#cb1-445" aria-hidden="true" tabindex="-1"></a><span class="co">#|   To evaluate the incident effect of the GFP on cells by class, we used</span></span>
<span id="cb1-446"><a href="#cb1-446" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the same notation as in @fig-dsred-temporal-distribution.</span></span>
<span id="cb1-447"><a href="#cb1-447" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The chromosomal strain exhibits variability in GFP at different</span></span>
<span id="cb1-448"><a href="#cb1-448" aria-hidden="true" tabindex="-1"></a><span class="co">#|   time points, mainly due to experimental noise resulting from low</span></span>
<span id="cb1-449"><a href="#cb1-449" aria-hidden="true" tabindex="-1"></a><span class="co">#|   fluorescent intensity values. As expected, filamented cells had</span></span>
<span id="cb1-450"><a href="#cb1-450" aria-hidden="true" tabindex="-1"></a><span class="co">#|   a lower initial GFP in the plasmid strain.</span></span>
<span id="cb1-451"><a href="#cb1-451" aria-hidden="true" tabindex="-1"></a><span class="co">#|   At the time of filamentation, there appear to be differences in</span></span>
<span id="cb1-452"><a href="#cb1-452" aria-hidden="true" tabindex="-1"></a><span class="co">#|   fluorescence between surviving and dead cells. However, in the end time,</span></span>
<span id="cb1-453"><a href="#cb1-453" aria-hidden="true" tabindex="-1"></a><span class="co">#|   we observed that the surviving non-filamented cells have lower GFP</span></span>
<span id="cb1-454"><a href="#cb1-454" aria-hidden="true" tabindex="-1"></a><span class="co">#|   values than the non-filamented dead cells and alive filamented cells.</span></span>
<span id="cb1-455"><a href="#cb1-455" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-456"><a href="#cb1-456" aria-hidden="true" tabindex="-1"></a>p_temporal_gfp_distributution <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-457"><a href="#cb1-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb1-458"><a href="#cb1-458" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">contains</span>(<span class="st">"gfp"</span>),</span>
<span id="cb1-459"><a href="#cb1-459" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"metric"</span>,</span>
<span id="cb1-460"><a href="#cb1-460" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb1-461"><a href="#cb1-461" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-462"><a href="#cb1-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">parse_metrics_column</span>(metric) <span class="sc">%&gt;%</span> </span>
<span id="cb1-463"><a href="#cb1-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-464"><a href="#cb1-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-465"><a href="#cb1-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> cell_status, <span class="at">y =</span> value, <span class="at">fill =</span> cell_status, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-466"><a href="#cb1-466" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_eye</span>() <span class="sc">+</span></span>
<span id="cb1-467"><a href="#cb1-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-468"><a href="#cb1-468" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-469"><a href="#cb1-469" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"Not filamented - Survived"</span>, <span class="st">"Not filamented - Not survived"</span>)),</span>
<span id="cb1-470"><a href="#cb1-470" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-471"><a href="#cb1-471" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y =</span> <span class="fu">c</span>(<span class="fl">2.0</span>),</span>
<span id="cb1-472"><a href="#cb1-472" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span></span>
<span id="cb1-473"><a href="#cb1-473" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-474"><a href="#cb1-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-475"><a href="#cb1-475" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"anova"</span>,</span>
<span id="cb1-476"><a href="#cb1-476" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.x.npc =</span> <span class="fl">0.10</span>,</span>
<span id="cb1-477"><a href="#cb1-477" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.93</span></span>
<span id="cb1-478"><a href="#cb1-478" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="co"># Add global annova p-value</span></span>
<span id="cb1-479"><a href="#cb1-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-480"><a href="#cb1-480" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-481"><a href="#cb1-481" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-482"><a href="#cb1-482" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref.group =</span> <span class="st">"Filamented - Survived"</span>,</span>
<span id="cb1-483"><a href="#cb1-483" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-484"><a href="#cb1-484" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.75</span></span>
<span id="cb1-485"><a href="#cb1-485" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-486"><a href="#cb1-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> metric) <span class="sc">+</span></span>
<span id="cb1-487"><a href="#cb1-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-488"><a href="#cb1-488" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-489"><a href="#cb1-489" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-490"><a href="#cb1-490" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-491"><a href="#cb1-491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-492"><a href="#cb1-492" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-493"><a href="#cb1-493" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-494"><a href="#cb1-494" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-495"><a href="#cb1-495" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-496"><a href="#cb1-496" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-497"><a href="#cb1-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-498"><a href="#cb1-498" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-499"><a href="#cb1-499" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-500"><a href="#cb1-500" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"GFP value"</span></span>
<span id="cb1-501"><a href="#cb1-501" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-502"><a href="#cb1-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-503"><a href="#cb1-503" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-504"><a href="#cb1-504" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-505"><a href="#cb1-505" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-506"><a href="#cb1-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-507"><a href="#cb1-507" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-508"><a href="#cb1-508" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-509"><a href="#cb1-509" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-510"><a href="#cb1-510" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-511"><a href="#cb1-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-512"><a href="#cb1-512" aria-hidden="true" tabindex="-1"></a>p_temporal_gfp_distributution</span>
<span id="cb1-513"><a href="#cb1-513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-514"><a href="#cb1-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-515"><a href="#cb1-515" aria-hidden="true" tabindex="-1"></a>Cell length was one of the factors that GFP expression levels could not</span>
<span id="cb1-516"><a href="#cb1-516" aria-hidden="true" tabindex="-1"></a>explain for cell survival. In @fig-length-temporal-distribution, we show</span>
<span id="cb1-517"><a href="#cb1-517" aria-hidden="true" tabindex="-1"></a>that the conclusions regarding filamentation were applicable for both</span>
<span id="cb1-518"><a href="#cb1-518" aria-hidden="true" tabindex="-1"></a>chromosomal and plasmid strains. For the initial times, filamented and</span>
<span id="cb1-519"><a href="#cb1-519" aria-hidden="true" tabindex="-1"></a>survived cells were shorter in length than those that died but longer</span>
<span id="cb1-520"><a href="#cb1-520" aria-hidden="true" tabindex="-1"></a>than not filamented cells of both classes, while non-filamented cells</span>
<span id="cb1-521"><a href="#cb1-521" aria-hidden="true" tabindex="-1"></a>did not differ. We observed no length differences between cells at</span>
<span id="cb1-522"><a href="#cb1-522" aria-hidden="true" tabindex="-1"></a>filamentation time. Thus, survival could depend on other factors, such</span>
<span id="cb1-523"><a href="#cb1-523" aria-hidden="true" tabindex="-1"></a>as growth rate. In the final time, the results were well-defined.</span>
<span id="cb1-524"><a href="#cb1-524" aria-hidden="true" tabindex="-1"></a>Surviving cells had a greater length than their non-surviving pair</span>
<span id="cb1-525"><a href="#cb1-525" aria-hidden="true" tabindex="-1"></a>(*i.e.*, dead filamented and non-filamented cells). However, for</span>
<span id="cb1-526"><a href="#cb1-526" aria-hidden="true" tabindex="-1"></a>filamented cells, surviving cells generally represent a distribution of</span>
<span id="cb1-527"><a href="#cb1-527" aria-hidden="true" tabindex="-1"></a>higher final length values but are not as extensive as their dead</span>
<span id="cb1-528"><a href="#cb1-528" aria-hidden="true" tabindex="-1"></a>counterpart. Which we could explain as a length limit to which cells can</span>
<span id="cb1-529"><a href="#cb1-529" aria-hidden="true" tabindex="-1"></a>grow without dying. Nevertheless, we had no information to evaluate such</span>
<span id="cb1-530"><a href="#cb1-530" aria-hidden="true" tabindex="-1"></a>a hypothesis.</span>
<span id="cb1-531"><a href="#cb1-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-534"><a href="#cb1-534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-535"><a href="#cb1-535" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-length-temporal-distribution</span></span>
<span id="cb1-536"><a href="#cb1-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Length temporal distribution.</span></span>
<span id="cb1-537"><a href="#cb1-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-538"><a href="#cb1-538" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Length temporal distribution.**</span></span>
<span id="cb1-539"><a href="#cb1-539" aria-hidden="true" tabindex="-1"></a><span class="co">#|   To evaluate the incident effect of length on cells by class,</span></span>
<span id="cb1-540"><a href="#cb1-540" aria-hidden="true" tabindex="-1"></a><span class="co">#|   we use the same notation as in @fig-dsred-temporal-distribution.</span></span>
<span id="cb1-541"><a href="#cb1-541" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The observations for both strains, chromosomal or plasmid, are the same.</span></span>
<span id="cb1-542"><a href="#cb1-542" aria-hidden="true" tabindex="-1"></a><span class="co">#|   In the beginning, the surviving filamented cells already have a difference</span></span>
<span id="cb1-543"><a href="#cb1-543" aria-hidden="true" tabindex="-1"></a><span class="co">#|   in length from the rest of the classes. At the time of</span></span>
<span id="cb1-544"><a href="#cb1-544" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamentation, there is no difference to help determine whether the</span></span>
<span id="cb1-545"><a href="#cb1-545" aria-hidden="true" tabindex="-1"></a><span class="co">#|   cell will survive or not. Finally, in the final time, it seems that</span></span>
<span id="cb1-546"><a href="#cb1-546" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the surviving filamented cells have a greater length than the rest</span></span>
<span id="cb1-547"><a href="#cb1-547" aria-hidden="true" tabindex="-1"></a><span class="co">#|   of the groups. However, this length is moderate compared to the excess</span></span>
<span id="cb1-548"><a href="#cb1-548" aria-hidden="true" tabindex="-1"></a><span class="co">#|   length shown by non-surviving filamented cells. On the other hand,</span></span>
<span id="cb1-549"><a href="#cb1-549" aria-hidden="true" tabindex="-1"></a><span class="co">#|   we highlighted the growth of the surviving non-filamented cells.</span></span>
<span id="cb1-550"><a href="#cb1-550" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Therefore, although they did not reach a length for us to classify as</span></span>
<span id="cb1-551"><a href="#cb1-551" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamented, the cells did resort to filamentation.</span></span>
<span id="cb1-552"><a href="#cb1-552" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-553"><a href="#cb1-553" aria-hidden="true" tabindex="-1"></a>p_temporal_length_distributution <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-554"><a href="#cb1-554" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb1-555"><a href="#cb1-555" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">contains</span>(<span class="st">"length"</span>),</span>
<span id="cb1-556"><a href="#cb1-556" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"metric"</span>,</span>
<span id="cb1-557"><a href="#cb1-557" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb1-558"><a href="#cb1-558" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-559"><a href="#cb1-559" aria-hidden="true" tabindex="-1"></a>  <span class="fu">parse_metrics_column</span>(metric) <span class="sc">%&gt;%</span> </span>
<span id="cb1-560"><a href="#cb1-560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-561"><a href="#cb1-561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-562"><a href="#cb1-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> cell_status, <span class="at">y =</span> value, <span class="at">fill =</span> cell_status, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-563"><a href="#cb1-563" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> filamentation_threshold), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">alpha =</span> <span class="dv">1</span> <span class="sc">/</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb1-564"><a href="#cb1-564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_eye</span>() <span class="sc">+</span></span>
<span id="cb1-565"><a href="#cb1-565" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-566"><a href="#cb1-566" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-567"><a href="#cb1-567" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparisons =</span> <span class="fu">list</span>(</span>
<span id="cb1-568"><a href="#cb1-568" aria-hidden="true" tabindex="-1"></a>      <span class="fu">c</span>(<span class="st">"Not filamented - Survived"</span>, <span class="st">"Not filamented - Not survived"</span>)</span>
<span id="cb1-569"><a href="#cb1-569" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-570"><a href="#cb1-570" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-571"><a href="#cb1-571" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">60</span>),</span>
<span id="cb1-572"><a href="#cb1-572" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span></span>
<span id="cb1-573"><a href="#cb1-573" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-574"><a href="#cb1-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-575"><a href="#cb1-575" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"anova"</span>,</span>
<span id="cb1-576"><a href="#cb1-576" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.43</span>,</span>
<span id="cb1-577"><a href="#cb1-577" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.x.npc =</span> <span class="fl">0.3</span></span>
<span id="cb1-578"><a href="#cb1-578" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="co"># Add global annova p-value</span></span>
<span id="cb1-579"><a href="#cb1-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-580"><a href="#cb1-580" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-581"><a href="#cb1-581" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-582"><a href="#cb1-582" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref.group =</span> <span class="st">"Filamented - Survived"</span>,</span>
<span id="cb1-583"><a href="#cb1-583" aria-hidden="true" tabindex="-1"></a>    <span class="at">hide.ns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-584"><a href="#cb1-584" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.3</span></span>
<span id="cb1-585"><a href="#cb1-585" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-586"><a href="#cb1-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> metric) <span class="sc">+</span></span>
<span id="cb1-587"><a href="#cb1-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">150</span>)) <span class="sc">+</span></span>
<span id="cb1-588"><a href="#cb1-588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-589"><a href="#cb1-589" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-590"><a href="#cb1-590" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-591"><a href="#cb1-591" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-592"><a href="#cb1-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-593"><a href="#cb1-593" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-594"><a href="#cb1-594" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-595"><a href="#cb1-595" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-596"><a href="#cb1-596" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-597"><a href="#cb1-597" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-598"><a href="#cb1-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-599"><a href="#cb1-599" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-600"><a href="#cb1-600" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-601"><a href="#cb1-601" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Length value"</span></span>
<span id="cb1-602"><a href="#cb1-602" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-603"><a href="#cb1-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-604"><a href="#cb1-604" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-605"><a href="#cb1-605" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-606"><a href="#cb1-606" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-607"><a href="#cb1-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-608"><a href="#cb1-608" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-609"><a href="#cb1-609" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-610"><a href="#cb1-610" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-611"><a href="#cb1-611" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-612"><a href="#cb1-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-613"><a href="#cb1-613" aria-hidden="true" tabindex="-1"></a>p_temporal_length_distributution</span>
<span id="cb1-614"><a href="#cb1-614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-615"><a href="#cb1-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-616"><a href="#cb1-616" aria-hidden="true" tabindex="-1"></a>Once we observed the effects of GFP expression levels and lengths in</span>
<span id="cb1-617"><a href="#cb1-617" aria-hidden="true" tabindex="-1"></a>determining whether a cell lives or dies, we projected the cells onto</span>
<span id="cb1-618"><a href="#cb1-618" aria-hidden="true" tabindex="-1"></a>the plane. We painted them with their class status (See</span>
<span id="cb1-619"><a href="#cb1-619" aria-hidden="true" tabindex="-1"></a>@fig-cell-distribution-across-experiments) to determine whether these</span>
<span id="cb1-620"><a href="#cb1-620" aria-hidden="true" tabindex="-1"></a>two variables contained the necessary information to cluster the data</span>
<span id="cb1-621"><a href="#cb1-621" aria-hidden="true" tabindex="-1"></a>correctly. In @fig-just-initial-values, we show the initial GFP and</span>
<span id="cb1-622"><a href="#cb1-622" aria-hidden="true" tabindex="-1"></a>length values projection. While, with some work, we could contextually</span>
<span id="cb1-623"><a href="#cb1-623" aria-hidden="true" tabindex="-1"></a>place the results in @fig-gfp-temporal-distribution and</span>
<span id="cb1-624"><a href="#cb1-624" aria-hidden="true" tabindex="-1"></a>@fig-length-temporal-distribution, the initial values did not appear to</span>
<span id="cb1-625"><a href="#cb1-625" aria-hidden="true" tabindex="-1"></a>determine the classes. Therefore, we explored the final versus initial</span>
<span id="cb1-626"><a href="#cb1-626" aria-hidden="true" tabindex="-1"></a>values differences in @fig-metric-differences. With this new</span>
<span id="cb1-627"><a href="#cb1-627" aria-hidden="true" tabindex="-1"></a>representation of the cells in the plane, we contextualized the</span>
<span id="cb1-628"><a href="#cb1-628" aria-hidden="true" tabindex="-1"></a>statistical results presented in @fig-gfp-temporal-distribution and</span>
<span id="cb1-629"><a href="#cb1-629" aria-hidden="true" tabindex="-1"></a>@fig-length-temporal-distribution. Besides, it showed us that</span>
<span id="cb1-630"><a href="#cb1-630" aria-hidden="true" tabindex="-1"></a>differences in length (*i.e.*, filamentation) and reductions in GFP</span>
<span id="cb1-631"><a href="#cb1-631" aria-hidden="true" tabindex="-1"></a>expression are essential in determining cell survival. Though the</span>
<span id="cb1-632"><a href="#cb1-632" aria-hidden="true" tabindex="-1"></a>clustering of cell status is not entirely separated, other variables</span>
<span id="cb1-633"><a href="#cb1-633" aria-hidden="true" tabindex="-1"></a>affect the experimental results in cell survival.</span>
<span id="cb1-634"><a href="#cb1-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-637"><a href="#cb1-637" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-638"><a href="#cb1-638" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-just-initial-values</span></span>
<span id="cb1-639"><a href="#cb1-639" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Experiment's initial values.</span></span>
<span id="cb1-640"><a href="#cb1-640" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-641"><a href="#cb1-641" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Experiment's initial values.**</span></span>
<span id="cb1-642"><a href="#cb1-642" aria-hidden="true" tabindex="-1"></a><span class="co">#|   By positioning a cell in space based on its initial length and GFP</span></span>
<span id="cb1-643"><a href="#cb1-643" aria-hidden="true" tabindex="-1"></a><span class="co">#|   values, we can see that class separation occurs, but not as a strong</span></span>
<span id="cb1-644"><a href="#cb1-644" aria-hidden="true" tabindex="-1"></a><span class="co">#|   signal. Therefore, we concluded that although the initial state</span></span>
<span id="cb1-645"><a href="#cb1-645" aria-hidden="true" tabindex="-1"></a><span class="co">#|   influences the result, this is not everything. For this, we have</span></span>
<span id="cb1-646"><a href="#cb1-646" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the example of the length changes throughout the experiment caused</span></span>
<span id="cb1-647"><a href="#cb1-647" aria-hidden="true" tabindex="-1"></a><span class="co">#|   by filamentation. In this graph, the GFP scale is at log10 to help</span></span>
<span id="cb1-648"><a href="#cb1-648" aria-hidden="true" tabindex="-1"></a><span class="co">#|   us observe those minor differences between the experiments.</span></span>
<span id="cb1-649"><a href="#cb1-649" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-650"><a href="#cb1-650" aria-hidden="true" tabindex="-1"></a>p_initial_values <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-651"><a href="#cb1-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(gfp_first), <span class="at">y =</span> length_first, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-652"><a href="#cb1-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb1-653"><a href="#cb1-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> experiment_id, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb1-654"><a href="#cb1-654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-655"><a href="#cb1-655" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">1</span>)),</span>
<span id="cb1-656"><a href="#cb1-656" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-657"><a href="#cb1-657" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-658"><a href="#cb1-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-659"><a href="#cb1-659" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Initial normalized GFP (log10)"</span>,</span>
<span id="cb1-660"><a href="#cb1-660" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Initial length"</span>,</span>
<span id="cb1-661"><a href="#cb1-661" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-662"><a href="#cb1-662" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-663"><a href="#cb1-663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-664"><a href="#cb1-664" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-665"><a href="#cb1-665" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-666"><a href="#cb1-666" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-667"><a href="#cb1-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-668"><a href="#cb1-668" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-669"><a href="#cb1-669" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-670"><a href="#cb1-670" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-671"><a href="#cb1-671" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-672"><a href="#cb1-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-673"><a href="#cb1-673" aria-hidden="true" tabindex="-1"></a>p_initial_values</span>
<span id="cb1-674"><a href="#cb1-674" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-675"><a href="#cb1-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-678"><a href="#cb1-678" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-679"><a href="#cb1-679" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-metric-differences</span></span>
<span id="cb1-680"><a href="#cb1-680" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Experiment's initial values differences.</span></span>
<span id="cb1-681"><a href="#cb1-681" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-682"><a href="#cb1-682" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Experiment's initial values differences.**</span></span>
<span id="cb1-683"><a href="#cb1-683" aria-hidden="true" tabindex="-1"></a><span class="co">#|   By comparing the metric differences of the last observation and</span></span>
<span id="cb1-684"><a href="#cb1-684" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the first observation of a cell, we can separate mainly the</span></span>
<span id="cb1-685"><a href="#cb1-685" aria-hidden="true" tabindex="-1"></a><span class="co">#|   surviving filamented cells from those that did not do it in both</span></span>
<span id="cb1-686"><a href="#cb1-686" aria-hidden="true" tabindex="-1"></a><span class="co">#|   experiments (blue dots). Meanwhile, cells with plasmids form a</span></span>
<span id="cb1-687"><a href="#cb1-687" aria-hidden="true" tabindex="-1"></a><span class="co">#|   small accumulation of surviving cells that did not produce</span></span>
<span id="cb1-688"><a href="#cb1-688" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filament (green dots). However, this has made a breakthrough in</span></span>
<span id="cb1-689"><a href="#cb1-689" aria-hidden="true" tabindex="-1"></a><span class="co">#|   understanding what is affecting cell survival. There are still</span></span>
<span id="cb1-690"><a href="#cb1-690" aria-hidden="true" tabindex="-1"></a><span class="co">#|   variables that we can include to understand this phenomenon better.</span></span>
<span id="cb1-691"><a href="#cb1-691" aria-hidden="true" tabindex="-1"></a><span class="co">#|   </span></span>
<span id="cb1-692"><a href="#cb1-692" aria-hidden="true" tabindex="-1"></a>p_metric_differences <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-693"><a href="#cb1-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(gfp_last) <span class="sc">-</span> <span class="fu">log</span>(gfp_first), <span class="at">y =</span> length_last <span class="sc">-</span> length_first, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-694"><a href="#cb1-694" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb1-695"><a href="#cb1-695" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>experiment_id) <span class="sc">+</span></span>
<span id="cb1-696"><a href="#cb1-696" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-697"><a href="#cb1-697" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>)),</span>
<span id="cb1-698"><a href="#cb1-698" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-699"><a href="#cb1-699" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-700"><a href="#cb1-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-701"><a href="#cb1-701" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"End GFP - Initial GFP"</span>,</span>
<span id="cb1-702"><a href="#cb1-702" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"End length - Initial length"</span>,</span>
<span id="cb1-703"><a href="#cb1-703" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-704"><a href="#cb1-704" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-705"><a href="#cb1-705" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-706"><a href="#cb1-706" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-707"><a href="#cb1-707" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-708"><a href="#cb1-708" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-709"><a href="#cb1-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-710"><a href="#cb1-710" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-711"><a href="#cb1-711" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-712"><a href="#cb1-712" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-713"><a href="#cb1-713" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-714"><a href="#cb1-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-715"><a href="#cb1-715" aria-hidden="true" tabindex="-1"></a>p_metric_differences</span>
<span id="cb1-716"><a href="#cb1-716" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-717"><a href="#cb1-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-720"><a href="#cb1-720" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-721"><a href="#cb1-721" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb1-722"><a href="#cb1-722" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-histogram-gpf-intensity</span></span>
<span id="cb1-723"><a href="#cb1-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Histograms of fluorescent intensity for classified cells.</span></span>
<span id="cb1-724"><a href="#cb1-724" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-725"><a href="#cb1-725" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Histograms of fluorescent intensity for classified cells.**</span></span>
<span id="cb1-726"><a href="#cb1-726" aria-hidden="true" tabindex="-1"></a><span class="co">#|   A) Cells in MG:GT exhibit a fluorescent distribution with low variance</span></span>
<span id="cb1-727"><a href="#cb1-727" aria-hidden="true" tabindex="-1"></a><span class="co">#|   and with no significant differences in mean GFP between cells that</span></span>
<span id="cb1-728"><a href="#cb1-728" aria-hidden="true" tabindex="-1"></a><span class="co">#|   produced filaments and were killed (red) or survived (blue), as well</span></span>
<span id="cb1-729"><a href="#cb1-729" aria-hidden="true" tabindex="-1"></a><span class="co">#|   as for cells that did not produce filaments and died (orange), and those</span></span>
<span id="cb1-730"><a href="#cb1-730" aria-hidden="true" tabindex="-1"></a><span class="co">#|   that survived drug exposure (green). B) GFP distributions of the</span></span>
<span id="cb1-731"><a href="#cb1-731" aria-hidden="true" tabindex="-1"></a><span class="co">#|   plasmid-bearing population exhibit large variance. Cells that</span></span>
<span id="cb1-732"><a href="#cb1-732" aria-hidden="true" tabindex="-1"></a><span class="co">#|   survived showed increased mean fluorescence relative to cells that were</span></span>
<span id="cb1-733"><a href="#cb1-733" aria-hidden="true" tabindex="-1"></a><span class="co">#|   killed. For surviving cells, mean GFP was significantly lower for cells</span></span>
<span id="cb1-734"><a href="#cb1-734" aria-hidden="true" tabindex="-1"></a><span class="co">#|   that did not produce filaments with respect to cells that triggered the</span></span>
<span id="cb1-735"><a href="#cb1-735" aria-hidden="true" tabindex="-1"></a><span class="co">#|   SOS response system.</span></span>
<span id="cb1-736"><a href="#cb1-736" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-737"><a href="#cb1-737" aria-hidden="true" tabindex="-1"></a>cells_df <span class="sc">|&gt;</span> </span>
<span id="cb1-738"><a href="#cb1-738" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb1-739"><a href="#cb1-739" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> gfp_first,</span>
<span id="cb1-740"><a href="#cb1-740" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> ..scaled..,</span>
<span id="cb1-741"><a href="#cb1-741" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> cell_status,</span>
<span id="cb1-742"><a href="#cb1-742" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> cell_status</span>
<span id="cb1-743"><a href="#cb1-743" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb1-744"><a href="#cb1-744" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(</span>
<span id="cb1-745"><a href="#cb1-745" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="dv">1</span> <span class="sc">/</span> <span class="dv">4</span>,</span>
<span id="cb1-746"><a href="#cb1-746" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-747"><a href="#cb1-747" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> .) <span class="sc">+</span></span>
<span id="cb1-748"><a href="#cb1-748" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb1-749"><a href="#cb1-749" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb1-750"><a href="#cb1-750" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-751"><a href="#cb1-751" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb1-752"><a href="#cb1-752" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb1-753"><a href="#cb1-753" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_percent</span>()</span>
<span id="cb1-754"><a href="#cb1-754" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-755"><a href="#cb1-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-756"><a href="#cb1-756" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-757"><a href="#cb1-757" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-758"><a href="#cb1-758" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-759"><a href="#cb1-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-760"><a href="#cb1-760" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-761"><a href="#cb1-761" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb1-762"><a href="#cb1-762" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-763"><a href="#cb1-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-764"><a href="#cb1-764" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"GFP"</span>,</span>
<span id="cb1-765"><a href="#cb1-765" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Scaled density"</span>,</span>
<span id="cb1-766"><a href="#cb1-766" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-767"><a href="#cb1-767" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-768"><a href="#cb1-768" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-769"><a href="#cb1-769" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-770"><a href="#cb1-770" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-771"><a href="#cb1-771" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-772"><a href="#cb1-772" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-773"><a href="#cb1-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-774"><a href="#cb1-774" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-775"><a href="#cb1-775" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-776"><a href="#cb1-776" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-777"><a href="#cb1-777" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-778"><a href="#cb1-778" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-779"><a href="#cb1-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-780"><a href="#cb1-780" aria-hidden="true" tabindex="-1"></a><span class="fu">### Number of divisions and cell age do not appear to play a clear role in determining cell survival</span></span>
<span id="cb1-781"><a href="#cb1-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-782"><a href="#cb1-782" aria-hidden="true" tabindex="-1"></a>In @sec-length-gfp-crucial, we explored the effect on cell survival</span>
<span id="cb1-783"><a href="#cb1-783" aria-hidden="true" tabindex="-1"></a>through GFP variability and cell length.However,</span>
<span id="cb1-784"><a href="#cb1-784" aria-hidden="true" tabindex="-1"></a>@fig-just-initial-values and @fig-metric-differences showed us the</span>
<span id="cb1-785"><a href="#cb1-785" aria-hidden="true" tabindex="-1"></a>possibility of other factors relevant to the phenomenon under study. As</span>
<span id="cb1-786"><a href="#cb1-786" aria-hidden="true" tabindex="-1"></a>some papers in the literature suggest, some of these other factors may</span>
<span id="cb1-787"><a href="#cb1-787" aria-hidden="true" tabindex="-1"></a>be cell division and chronological age (*i.e.*, how much time has passed</span>
<span id="cb1-788"><a href="#cb1-788" aria-hidden="true" tabindex="-1"></a>since the last cell division at the time of exposure to a toxic agent)</span>
<span id="cb1-789"><a href="#cb1-789" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@moger-reischer2019; @roostalu2008; @heinrich2015</span><span class="co">]</span>. Therefore, we chose</span>
<span id="cb1-790"><a href="#cb1-790" aria-hidden="true" tabindex="-1"></a>to observe these two metrics in experiments at a purely qualitative</span>
<span id="cb1-791"><a href="#cb1-791" aria-hidden="true" tabindex="-1"></a>level, i.e., without the inclusion of, e.g., metrics of membrane or cell</span>
<span id="cb1-792"><a href="#cb1-792" aria-hidden="true" tabindex="-1"></a>cycle properties <span class="co">[</span><span class="ot">@Joseleau-Petit1999</span><span class="co">]</span>.</span>
<span id="cb1-793"><a href="#cb1-793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-794"><a href="#cb1-794" aria-hidden="true" tabindex="-1"></a>Although we expected to see a small contribution, either by the number</span>
<span id="cb1-795"><a href="#cb1-795" aria-hidden="true" tabindex="-1"></a>of divisions or cell age, in @fig-number-divisions and</span>
<span id="cb1-796"><a href="#cb1-796" aria-hidden="true" tabindex="-1"></a>@fig-time-since-last-division, we could not observe a precise effect of</span>
<span id="cb1-797"><a href="#cb1-797" aria-hidden="true" tabindex="-1"></a>these variables on cell survival. Although they could have an</span>
<span id="cb1-798"><a href="#cb1-798" aria-hidden="true" tabindex="-1"></a>explanation or biological significance, we decided to omit as relevant</span>
<span id="cb1-799"><a href="#cb1-799" aria-hidden="true" tabindex="-1"></a>in the characterization of our cells, since the signal was not clear.</span>
<span id="cb1-800"><a href="#cb1-800" aria-hidden="true" tabindex="-1"></a>However, we derived from this analysis a slightly simpler variable that</span>
<span id="cb1-801"><a href="#cb1-801" aria-hidden="true" tabindex="-1"></a>tells us whether a cell underwent a cell division event or not. So it</span>
<span id="cb1-802"><a href="#cb1-802" aria-hidden="true" tabindex="-1"></a>gives us a more generalized picture of the contribution of division to</span>
<span id="cb1-803"><a href="#cb1-803" aria-hidden="true" tabindex="-1"></a>cell survival (see @fig-plasmid-pca-variable-contribution).</span>
<span id="cb1-804"><a href="#cb1-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-807"><a href="#cb1-807" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-808"><a href="#cb1-808" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-number-divisions</span></span>
<span id="cb1-809"><a href="#cb1-809" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Cell's number of divisions.</span></span>
<span id="cb1-810"><a href="#cb1-810" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-811"><a href="#cb1-811" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Cell's number of divisions.**</span></span>
<span id="cb1-812"><a href="#cb1-812" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Both chromosomal and plasmid cells exhibited a wider distribution of</span></span>
<span id="cb1-813"><a href="#cb1-813" aria-hidden="true" tabindex="-1"></a><span class="co">#|   divisions for the surviving cells against non-surviving cells. However,</span></span>
<span id="cb1-814"><a href="#cb1-814" aria-hidden="true" tabindex="-1"></a><span class="co">#|   we did not appreciate a significant change between the chromosome </span></span>
<span id="cb1-815"><a href="#cb1-815" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamented cells. Therefore, the number of cell divisions' contribution</span></span>
<span id="cb1-816"><a href="#cb1-816" aria-hidden="true" tabindex="-1"></a><span class="co">#|   to filamentation remains uncertain.</span></span>
<span id="cb1-817"><a href="#cb1-817" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb1-818"><a href="#cb1-818" aria-hidden="true" tabindex="-1"></a>lineages_processed_1_df <span class="sc">|&gt;</span> </span>
<span id="cb1-819"><a href="#cb1-819" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(</span>
<span id="cb1-820"><a href="#cb1-820" aria-hidden="true" tabindex="-1"></a>        time <span class="sc">&lt;=</span> antibiotic_start_time</span>
<span id="cb1-821"><a href="#cb1-821" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span> </span>
<span id="cb1-822"><a href="#cb1-822" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(experiment_id, id, cell_status, <span class="at">wt =</span> division) <span class="sc">|&gt;</span> </span>
<span id="cb1-823"><a href="#cb1-823" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb1-824"><a href="#cb1-824" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> cell_status,</span>
<span id="cb1-825"><a href="#cb1-825" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> n,</span>
<span id="cb1-826"><a href="#cb1-826" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> cell_status  </span>
<span id="cb1-827"><a href="#cb1-827" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb1-828"><a href="#cb1-828" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_eye</span>() <span class="sc">+</span></span>
<span id="cb1-829"><a href="#cb1-829" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-830"><a href="#cb1-830" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-831"><a href="#cb1-831" aria-hidden="true" tabindex="-1"></a>      <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-832"><a href="#cb1-832" aria-hidden="true" tabindex="-1"></a>      <span class="at">ref.group =</span> <span class="st">"Filamented - Survived"</span>,</span>
<span id="cb1-833"><a href="#cb1-833" aria-hidden="true" tabindex="-1"></a>      <span class="at">label.y.npc =</span> <span class="fl">0.75</span>,</span>
<span id="cb1-834"><a href="#cb1-834" aria-hidden="true" tabindex="-1"></a>      <span class="at">hide.ns =</span> <span class="cn">TRUE</span></span>
<span id="cb1-835"><a href="#cb1-835" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-836"><a href="#cb1-836" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-837"><a href="#cb1-837" aria-hidden="true" tabindex="-1"></a>      <span class="at">method =</span> <span class="st">"t.test"</span>,</span>
<span id="cb1-838"><a href="#cb1-838" aria-hidden="true" tabindex="-1"></a>      <span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"Not filamented - Survived"</span>, <span class="st">"Not filamented - Not survived"</span>)),</span>
<span id="cb1-839"><a href="#cb1-839" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-840"><a href="#cb1-840" aria-hidden="true" tabindex="-1"></a>      <span class="at">label.y.npc =</span> <span class="fl">0.6</span>,</span>
<span id="cb1-841"><a href="#cb1-841" aria-hidden="true" tabindex="-1"></a>      <span class="at">hide.ns =</span> <span class="cn">TRUE</span></span>
<span id="cb1-842"><a href="#cb1-842" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-843"><a href="#cb1-843" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(. <span class="sc">~</span> experiment_id) <span class="sc">+</span></span>
<span id="cb1-844"><a href="#cb1-844" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(</span>
<span id="cb1-845"><a href="#cb1-845" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-846"><a href="#cb1-846" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-847"><a href="#cb1-847" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb1-848"><a href="#cb1-848" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-849"><a href="#cb1-849" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-850"><a href="#cb1-850" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-851"><a href="#cb1-851" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-852"><a href="#cb1-852" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey92"</span>),</span>
<span id="cb1-853"><a href="#cb1-853" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">0.5</span>))</span>
<span id="cb1-854"><a href="#cb1-854" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-855"><a href="#cb1-855" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-856"><a href="#cb1-856" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-857"><a href="#cb1-857" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-858"><a href="#cb1-858" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-859"><a href="#cb1-859" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(</span>
<span id="cb1-860"><a href="#cb1-860" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">6</span></span>
<span id="cb1-861"><a href="#cb1-861" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-862"><a href="#cb1-862" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb1-863"><a href="#cb1-863" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Number of divisions"</span>,</span>
<span id="cb1-864"><a href="#cb1-864" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-865"><a href="#cb1-865" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-866"><a href="#cb1-866" aria-hidden="true" tabindex="-1"></a>    <span class="cn">NULL</span></span>
<span id="cb1-867"><a href="#cb1-867" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-868"><a href="#cb1-868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-871"><a href="#cb1-871" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-872"><a href="#cb1-872" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-time-since-last-division</span></span>
<span id="cb1-873"><a href="#cb1-873" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Time elapsed since the last division at the beginning of the experiment.</span></span>
<span id="cb1-874"><a href="#cb1-874" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-875"><a href="#cb1-875" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Time elapsed since the last division at the beginning of the experiment.**</span></span>
<span id="cb1-876"><a href="#cb1-876" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The mean time of the last division before starting the experiment</span></span>
<span id="cb1-877"><a href="#cb1-877" aria-hidden="true" tabindex="-1"></a><span class="co">#|   indicates that it did not influence the final result for chromosomal</span></span>
<span id="cb1-878"><a href="#cb1-878" aria-hidden="true" tabindex="-1"></a><span class="co">#|   cells. There is a slight difference between the filamented-not survived</span></span>
<span id="cb1-879"><a href="#cb1-879" aria-hidden="true" tabindex="-1"></a><span class="co">#|   cells and the rest for cells with plasmids. However, the signal does not</span></span>
<span id="cb1-880"><a href="#cb1-880" aria-hidden="true" tabindex="-1"></a><span class="co">#|   appear to be strong on the survival role. Therefore, we conclude that we</span></span>
<span id="cb1-881"><a href="#cb1-881" aria-hidden="true" tabindex="-1"></a><span class="co">#|   have no evidence to support that the time of the last division at the</span></span>
<span id="cb1-882"><a href="#cb1-882" aria-hidden="true" tabindex="-1"></a><span class="co">#|   beginning of the experiment influences the final classification results.</span></span>
<span id="cb1-883"><a href="#cb1-883" aria-hidden="true" tabindex="-1"></a><span class="co">#|   </span></span>
<span id="cb1-884"><a href="#cb1-884" aria-hidden="true" tabindex="-1"></a>p_time_since_last_division <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-885"><a href="#cb1-885" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_since_last_division_to_experiment_start)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-886"><a href="#cb1-886" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(</span>
<span id="cb1-887"><a href="#cb1-887" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb1-888"><a href="#cb1-888" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> cell_status,</span>
<span id="cb1-889"><a href="#cb1-889" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> time_since_last_division_to_experiment_start,</span>
<span id="cb1-890"><a href="#cb1-890" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> cell_status</span>
<span id="cb1-891"><a href="#cb1-891" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-892"><a href="#cb1-892" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-893"><a href="#cb1-893" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_eye</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb1-894"><a href="#cb1-894" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>experiment_id) <span class="sc">+</span>    </span>
<span id="cb1-895"><a href="#cb1-895" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-896"><a href="#cb1-896" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Experiment"</span>,</span>
<span id="cb1-897"><a href="#cb1-897" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Time since last division to experiment start"</span>,</span>
<span id="cb1-898"><a href="#cb1-898" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-899"><a href="#cb1-899" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-900"><a href="#cb1-900" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-901"><a href="#cb1-901" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-902"><a href="#cb1-902" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-903"><a href="#cb1-903" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-904"><a href="#cb1-904" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-905"><a href="#cb1-905" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-906"><a href="#cb1-906" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-907"><a href="#cb1-907" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-908"><a href="#cb1-908" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-909"><a href="#cb1-909" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey92"</span>),</span>
<span id="cb1-910"><a href="#cb1-910" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">0.5</span>))</span>
<span id="cb1-911"><a href="#cb1-911" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-912"><a href="#cb1-912" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-913"><a href="#cb1-913" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-914"><a href="#cb1-914" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-915"><a href="#cb1-915" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-916"><a href="#cb1-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-917"><a href="#cb1-917" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-918"><a href="#cb1-918" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-919"><a href="#cb1-919" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-920"><a href="#cb1-920" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-921"><a href="#cb1-921" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-922"><a href="#cb1-922" aria-hidden="true" tabindex="-1"></a>p_time_since_last_division</span>
<span id="cb1-923"><a href="#cb1-923" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-924"><a href="#cb1-924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-925"><a href="#cb1-925" aria-hidden="true" tabindex="-1"></a><span class="fu">### Time to reach filamentation matters in determining cell survival</span></span>
<span id="cb1-926"><a href="#cb1-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-927"><a href="#cb1-927" aria-hidden="true" tabindex="-1"></a>In @fig-dsred-temporal-distribution, @fig-gfp-temporal-distribution, and</span>
<span id="cb1-928"><a href="#cb1-928" aria-hidden="true" tabindex="-1"></a>@fig-length-temporal-distribution, we showed how, at the time of</span>
<span id="cb1-929"><a href="#cb1-929" aria-hidden="true" tabindex="-1"></a>filamentation, DsRed and GFP levels appeared indifferent to the cells.</span>
<span id="cb1-930"><a href="#cb1-930" aria-hidden="true" tabindex="-1"></a>Therefore, we hypothesized that a possible variable determining cell</span>
<span id="cb1-931"><a href="#cb1-931" aria-hidden="true" tabindex="-1"></a>survival could be its time to activate its anti-stress response system</span>
<span id="cb1-932"><a href="#cb1-932" aria-hidden="true" tabindex="-1"></a>that causes filamentation. Furthermore, we also guided our hypothesis by</span>
<span id="cb1-933"><a href="#cb1-933" aria-hidden="true" tabindex="-1"></a>previous reports showing us how the gene expression level can induce</span>
<span id="cb1-934"><a href="#cb1-934" aria-hidden="true" tabindex="-1"></a>filamentation with tight temporal coordination <span class="sc">\[</span>x<span class="sc">\]</span>.</span>
<span id="cb1-935"><a href="#cb1-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-936"><a href="#cb1-936" aria-hidden="true" tabindex="-1"></a>While, for our analyses, we did not measure the concentration of</span>
<span id="cb1-937"><a href="#cb1-937" aria-hidden="true" tabindex="-1"></a>antibiotic that triggers filamentation per se, we indirectly quantified</span>
<span id="cb1-938"><a href="#cb1-938" aria-hidden="true" tabindex="-1"></a>its effect by using the time it took for a cell to reach a length at</span>
<span id="cb1-939"><a href="#cb1-939" aria-hidden="true" tabindex="-1"></a>which it is already considered a filamentating cell. Furthermore, to</span>
<span id="cb1-940"><a href="#cb1-940" aria-hidden="true" tabindex="-1"></a>recognize that the observed effect was a product of the experiment, we</span>
<span id="cb1-941"><a href="#cb1-941" aria-hidden="true" tabindex="-1"></a>decided to keep only filamented cells just once antibiotic exposure</span>
<span id="cb1-942"><a href="#cb1-942" aria-hidden="true" tabindex="-1"></a>began.</span>
<span id="cb1-943"><a href="#cb1-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-944"><a href="#cb1-944" aria-hidden="true" tabindex="-1"></a>@fig-time-to-filamentation-filtered shows how filamentation times are</span>
<span id="cb1-945"><a href="#cb1-945" aria-hidden="true" tabindex="-1"></a>narrower for chromosomal cells than for plasmid-bearing cells. Then, we</span>
<span id="cb1-946"><a href="#cb1-946" aria-hidden="true" tabindex="-1"></a>hypothesize that the effect could come from the heterogeneity in the</span>
<span id="cb1-947"><a href="#cb1-947" aria-hidden="true" tabindex="-1"></a>plasmid copy number in the population. Also, interestingly, we observed</span>
<span id="cb1-948"><a href="#cb1-948" aria-hidden="true" tabindex="-1"></a>that, for both experiments, cells that survived had longer filamentation</span>
<span id="cb1-949"><a href="#cb1-949" aria-hidden="true" tabindex="-1"></a>times than those that died. These differences in response times suggest</span>
<span id="cb1-950"><a href="#cb1-950" aria-hidden="true" tabindex="-1"></a>the following: 1) if the cell grows too fast, it will reach a limit and</span>
<span id="cb1-951"><a href="#cb1-951" aria-hidden="true" tabindex="-1"></a>start to accumulate antibiotics constantly, and 2) if the cell grows too</span>
<span id="cb1-952"><a href="#cb1-952" aria-hidden="true" tabindex="-1"></a>fast, likely, the cost of maintaining an ample length for prolonged</span>
<span id="cb1-953"><a href="#cb1-953" aria-hidden="true" tabindex="-1"></a>periods of exposure will become counterproductive.</span>
<span id="cb1-954"><a href="#cb1-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-957"><a href="#cb1-957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-958"><a href="#cb1-958" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-time-to-filamentation-filtered</span></span>
<span id="cb1-959"><a href="#cb1-959" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Time to filamentation filtered.</span></span>
<span id="cb1-960"><a href="#cb1-960" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-961"><a href="#cb1-961" aria-hidden="true" tabindex="-1"></a><span class="co">#|  **Time to filamentation filtered.**</span></span>
<span id="cb1-962"><a href="#cb1-962" aria-hidden="true" tabindex="-1"></a><span class="co">#|  We only keep cells that filamented during the antibiotic exposure to</span></span>
<span id="cb1-963"><a href="#cb1-963" aria-hidden="true" tabindex="-1"></a><span class="co">#|  quantify their time to filamentation and its effect on survival. </span></span>
<span id="cb1-964"><a href="#cb1-964" aria-hidden="true" tabindex="-1"></a><span class="co">#|  In this way, we normalize the start times for the calculation of the</span></span>
<span id="cb1-965"><a href="#cb1-965" aria-hidden="true" tabindex="-1"></a><span class="co">#|  filamentation time. For both strains, the filamentation time had a more</span></span>
<span id="cb1-966"><a href="#cb1-966" aria-hidden="true" tabindex="-1"></a><span class="co">#|  significant delay in the surviving cells.</span></span>
<span id="cb1-967"><a href="#cb1-967" aria-hidden="true" tabindex="-1"></a>p_time_to_filamentation_filtered <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-968"><a href="#cb1-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb1-969"><a href="#cb1-969" aria-hidden="true" tabindex="-1"></a>    filamented_id <span class="sc">==</span> <span class="st">"Filamented"</span>,</span>
<span id="cb1-970"><a href="#cb1-970" aria-hidden="true" tabindex="-1"></a>    time_sos <span class="sc">&gt;</span> antibiotic_start_time</span>
<span id="cb1-971"><a href="#cb1-971" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-972"><a href="#cb1-972" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-973"><a href="#cb1-973" aria-hidden="true" tabindex="-1"></a>    <span class="at">time_to_sos =</span> time_sos <span class="sc">-</span> antibiotic_start_time,</span>
<span id="cb1-974"><a href="#cb1-974" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-975"><a href="#cb1-975" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>( <span class="at">x =</span> cell_status, <span class="at">y =</span> time_to_sos, <span class="at">fill =</span> survived)) <span class="sc">+</span></span>
<span id="cb1-976"><a href="#cb1-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_eye</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb1-977"><a href="#cb1-977" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb1-978"><a href="#cb1-978" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"p.signif"</span>,</span>
<span id="cb1-979"><a href="#cb1-979" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y.npc =</span> <span class="fl">0.8</span>,</span>
<span id="cb1-980"><a href="#cb1-980" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparisons =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"Filamented - Not survived"</span>, <span class="st">"Filamented - Survived"</span>))</span>
<span id="cb1-981"><a href="#cb1-981" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-982"><a href="#cb1-982" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> experiment_id) <span class="sc">+</span></span>
<span id="cb1-983"><a href="#cb1-983" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-984"><a href="#cb1-984" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Experiment"</span>,</span>
<span id="cb1-985"><a href="#cb1-985" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Time to filamentation (minutes)"</span>,</span>
<span id="cb1-986"><a href="#cb1-986" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-987"><a href="#cb1-987" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-988"><a href="#cb1-988" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( </span>
<span id="cb1-989"><a href="#cb1-989" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-990"><a href="#cb1-990" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-991"><a href="#cb1-991" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-992"><a href="#cb1-992" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-993"><a href="#cb1-993" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey92"</span>),</span>
<span id="cb1-994"><a href="#cb1-994" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">0.5</span>))</span>
<span id="cb1-995"><a href="#cb1-995" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-996"><a href="#cb1-996" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-997"><a href="#cb1-997" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#dd5129"</span>, <span class="st">"#43b284"</span>)</span>
<span id="cb1-998"><a href="#cb1-998" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-999"><a href="#cb1-999" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1000"><a href="#cb1-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1001"><a href="#cb1-1001" aria-hidden="true" tabindex="-1"></a>p_time_to_filamentation_filtered</span>
<span id="cb1-1002"><a href="#cb1-1002" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1003"><a href="#cb1-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1004"><a href="#cb1-1004" aria-hidden="true" tabindex="-1"></a>In Figure @fig-initial-values-with-time, we decided to project the</span>
<span id="cb1-1005"><a href="#cb1-1005" aria-hidden="true" tabindex="-1"></a>results of Figure @fig-time-to-filamentation-filtered in a space similar</span>
<span id="cb1-1006"><a href="#cb1-1006" aria-hidden="true" tabindex="-1"></a>to the one described in Figure @fig-just-initial-values). Thus, we</span>
<span id="cb1-1007"><a href="#cb1-1007" aria-hidden="true" tabindex="-1"></a>separated our data into cells that survived and cells that did not and</span>
<span id="cb1-1008"><a href="#cb1-1008" aria-hidden="true" tabindex="-1"></a>painted them when it took them to reach their filamented state. We</span>
<span id="cb1-1009"><a href="#cb1-1009" aria-hidden="true" tabindex="-1"></a>realized that, by adding this temporal component to the initial</span>
<span id="cb1-1010"><a href="#cb1-1010" aria-hidden="true" tabindex="-1"></a>variables of length and GFP, we could separate surviving cells from dead</span>
<span id="cb1-1011"><a href="#cb1-1011" aria-hidden="true" tabindex="-1"></a>cells to a greater degree. However, it may still not be enough, and</span>
<span id="cb1-1012"><a href="#cb1-1012" aria-hidden="true" tabindex="-1"></a>there are still many other variables that play a crucial role in</span>
<span id="cb1-1013"><a href="#cb1-1013" aria-hidden="true" tabindex="-1"></a>understanding the ecology of stress and how some cells will be survivors</span>
<span id="cb1-1014"><a href="#cb1-1014" aria-hidden="true" tabindex="-1"></a>or not.</span>
<span id="cb1-1015"><a href="#cb1-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1018"><a href="#cb1-1018" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1019"><a href="#cb1-1019" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-initial-values-with-time</span></span>
<span id="cb1-1020"><a href="#cb1-1020" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Experiment initial values with time to filamentation.</span></span>
<span id="cb1-1021"><a href="#cb1-1021" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1022"><a href="#cb1-1022" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Experiment initial values with time to filamentation.**</span></span>
<span id="cb1-1023"><a href="#cb1-1023" aria-hidden="true" tabindex="-1"></a><span class="co">#|   As in @fig-just-initial-values, including the time it will take for cells</span></span>
<span id="cb1-1024"><a href="#cb1-1024" aria-hidden="true" tabindex="-1"></a><span class="co">#|   to filament allows us to better understand the phenomenon of survival.</span></span>
<span id="cb1-1025"><a href="#cb1-1025" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Cells that filamented and survived generally have a much higher delay</span></span>
<span id="cb1-1026"><a href="#cb1-1026" aria-hidden="true" tabindex="-1"></a><span class="co">#|   than their non-filamented peers for both strains</span></span>
<span id="cb1-1027"><a href="#cb1-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#|   (see @fig-time-to-filamentation-filtered).</span></span>
<span id="cb1-1028"><a href="#cb1-1028" aria-hidden="true" tabindex="-1"></a>p_initial_values_with_time <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-1029"><a href="#cb1-1029" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb1-1030"><a href="#cb1-1030" aria-hidden="true" tabindex="-1"></a>    filamented_id <span class="sc">==</span> <span class="st">"Filamented"</span>,</span>
<span id="cb1-1031"><a href="#cb1-1031" aria-hidden="true" tabindex="-1"></a>    time_sos <span class="sc">&gt;</span> antibiotic_start_time</span>
<span id="cb1-1032"><a href="#cb1-1032" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1033"><a href="#cb1-1033" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">time_to_sos =</span> time_sos <span class="sc">-</span> antibiotic_start_time) <span class="sc">%&gt;%</span></span>
<span id="cb1-1034"><a href="#cb1-1034" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> length_first, <span class="at">y =</span> <span class="fu">log</span>(gfp_first), <span class="at">z =</span> time_to_sos, <span class="at">color =</span> time_to_sos)) <span class="sc">+</span></span>
<span id="cb1-1035"><a href="#cb1-1035" aria-hidden="true" tabindex="-1"></a>  <span class="co"># stat_summary_2d() +</span></span>
<span id="cb1-1036"><a href="#cb1-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-1037"><a href="#cb1-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> survived) <span class="sc">+</span></span>
<span id="cb1-1038"><a href="#cb1-1038" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">"inferno"</span>) <span class="sc">+</span></span>
<span id="cb1-1039"><a href="#cb1-1039" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1040"><a href="#cb1-1040" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Initial length"</span>,</span>
<span id="cb1-1041"><a href="#cb1-1041" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Initial GFP"</span>,</span>
<span id="cb1-1042"><a href="#cb1-1042" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Time to filamentation (minutes)"</span></span>
<span id="cb1-1043"><a href="#cb1-1043" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1044"><a href="#cb1-1044" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1045"><a href="#cb1-1045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1046"><a href="#cb1-1046" aria-hidden="true" tabindex="-1"></a>p_initial_values_with_time</span>
<span id="cb1-1047"><a href="#cb1-1047" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1048"><a href="#cb1-1048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1049"><a href="#cb1-1049" aria-hidden="true" tabindex="-1"></a><span class="fu">### Increasing the system's complexity and analyzing it in an unsupervised way allows a correct classification of cell states {#sec-unsupervised-classification}</span></span>
<span id="cb1-1050"><a href="#cb1-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1051"><a href="#cb1-1051" aria-hidden="true" tabindex="-1"></a>In the experiments, we observed the importance of GFP filamentation and</span>
<span id="cb1-1052"><a href="#cb1-1052" aria-hidden="true" tabindex="-1"></a>variability for cell survival. Similarly, we realized that other</span>
<span id="cb1-1053"><a href="#cb1-1053" aria-hidden="true" tabindex="-1"></a>variables must be affecting the final results. Filamentation and GFP</span>
<span id="cb1-1054"><a href="#cb1-1054" aria-hidden="true" tabindex="-1"></a>variability alone did not fully recapitulate the expected behavior of</span>
<span id="cb1-1055"><a href="#cb1-1055" aria-hidden="true" tabindex="-1"></a>the data. That is, the target variables did not capture the system's</span>
<span id="cb1-1056"><a href="#cb1-1056" aria-hidden="true" tabindex="-1"></a>heterogeneity.</span>
<span id="cb1-1057"><a href="#cb1-1057" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1058"><a href="#cb1-1058" aria-hidden="true" tabindex="-1"></a>The inability to reproduce cell classification led us to question two</span>
<span id="cb1-1059"><a href="#cb1-1059" aria-hidden="true" tabindex="-1"></a>things: 1) the possibility that our sorting was wrong beforehand, and 2)</span>
<span id="cb1-1060"><a href="#cb1-1060" aria-hidden="true" tabindex="-1"></a>we did not have enough variables to capture the study phenomenon. We</span>
<span id="cb1-1061"><a href="#cb1-1061" aria-hidden="true" tabindex="-1"></a>decided to take the unsupervised learning way to answer these subjects</span>
<span id="cb1-1062"><a href="#cb1-1062" aria-hidden="true" tabindex="-1"></a>because it allows us to project our data without prior knowledge.</span>
<span id="cb1-1063"><a href="#cb1-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1064"><a href="#cb1-1064" aria-hidden="true" tabindex="-1"></a>We opted for the path of dimensionality reduction techniques where each</span>
<span id="cb1-1065"><a href="#cb1-1065" aria-hidden="true" tabindex="-1"></a>variable or feature is equivalent to one dimension. The essence of</span>
<span id="cb1-1066"><a href="#cb1-1066" aria-hidden="true" tabindex="-1"></a>dimensionality reduction is that it is not feasible to analyze each</span>
<span id="cb1-1067"><a href="#cb1-1067" aria-hidden="true" tabindex="-1"></a>dimension with many dimensions. Furthermore, dimensionality reduction</span>
<span id="cb1-1068"><a href="#cb1-1068" aria-hidden="true" tabindex="-1"></a>helps us counteract several problems, such as reducing the complexity of</span>
<span id="cb1-1069"><a href="#cb1-1069" aria-hidden="true" tabindex="-1"></a>a model, reducing the possibility of overfitting a model, removing all</span>
<span id="cb1-1070"><a href="#cb1-1070" aria-hidden="true" tabindex="-1"></a>correlated variables, and visualizing our data in a two- or</span>
<span id="cb1-1071"><a href="#cb1-1071" aria-hidden="true" tabindex="-1"></a>three-dimensional space for better appreciation. Improved visualization</span>
<span id="cb1-1072"><a href="#cb1-1072" aria-hidden="true" tabindex="-1"></a>and identification of essential variables are the main reasons to guide</span>
<span id="cb1-1073"><a href="#cb1-1073" aria-hidden="true" tabindex="-1"></a>and complement our research with this technique.</span>
<span id="cb1-1074"><a href="#cb1-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1075"><a href="#cb1-1075" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Principal Component Analysis (PCA) emphasizes the importance of cell length and its GFP in cell survival</span></span>
<span id="cb1-1076"><a href="#cb1-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1077"><a href="#cb1-1077" aria-hidden="true" tabindex="-1"></a><span class="in">```{r experiment-03-split-datasets}</span></span>
<span id="cb1-1078"><a href="#cb1-1078" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1079"><a href="#cb1-1079" aria-hidden="true" tabindex="-1"></a>experiment_datasets <span class="ot">&lt;-</span> cells_df <span class="sc">%&gt;%</span> </span>
<span id="cb1-1080"><a href="#cb1-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(experiment_id, cell_status, divided, <span class="fu">contains</span>(<span class="st">"first"</span>), <span class="fu">contains</span>(<span class="st">"last"</span>), <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"time"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1081"><a href="#cb1-1081" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">divided =</span> <span class="fu">as.numeric</span>(divided)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1082"><a href="#cb1-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~!</span><span class="fu">any</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1083"><a href="#cb1-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">glimpse</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-1084"><a href="#cb1-1084" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-1085"><a href="#cb1-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(experiment_id) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1086"><a href="#cb1-1086" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb1-1087"><a href="#cb1-1087" aria-hidden="true" tabindex="-1"></a>    grouped_data <span class="ot">&lt;-</span> .</span>
<span id="cb1-1088"><a href="#cb1-1088" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_split</span>(grouped_data) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1089"><a href="#cb1-1089" aria-hidden="true" tabindex="-1"></a>      <span class="fu">set_names</span>(<span class="at">nm =</span> <span class="fu">group_keys</span>(grouped_data) <span class="sc">%&gt;%</span> <span class="fu">pull</span>())</span>
<span id="cb1-1090"><a href="#cb1-1090" aria-hidden="true" tabindex="-1"></a>  } <span class="sc">%&gt;%</span> </span>
<span id="cb1-1091"><a href="#cb1-1091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(select, <span class="sc">-</span>experiment_id) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1092"><a href="#cb1-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>()</span>
<span id="cb1-1093"><a href="#cb1-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1094"><a href="#cb1-1094" aria-hidden="true" tabindex="-1"></a>chromosome_df <span class="ot">&lt;-</span> experiment_datasets<span class="sc">$</span>Chromosome</span>
<span id="cb1-1095"><a href="#cb1-1095" aria-hidden="true" tabindex="-1"></a>plasmid_df <span class="ot">&lt;-</span> experiment_datasets<span class="sc">$</span>Plasmid</span>
<span id="cb1-1096"><a href="#cb1-1096" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1097"><a href="#cb1-1097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1098"><a href="#cb1-1098" aria-hidden="true" tabindex="-1"></a>The first dimensionality reduction technique we decided to use was</span>
<span id="cb1-1099"><a href="#cb1-1099" aria-hidden="true" tabindex="-1"></a>Principal Component Analysis (PCA) <span class="co">[</span><span class="ot">@pearson1901; @hotelling1936</span><span class="co">]</span>.</span>
<span id="cb1-1100"><a href="#cb1-1100" aria-hidden="true" tabindex="-1"></a>Scientist mainly uses PCA to create predictive models or in Exploratory</span>
<span id="cb1-1101"><a href="#cb1-1101" aria-hidden="true" tabindex="-1"></a>Data Analysis (EDA). In our case, we only use it as an EDA.</span>
<span id="cb1-1102"><a href="#cb1-1102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1103"><a href="#cb1-1103" aria-hidden="true" tabindex="-1"></a>For chromosomal and plasmid strain, in</span>
<span id="cb1-1104"><a href="#cb1-1104" aria-hidden="true" tabindex="-1"></a>@fig-chromosome-pca-new-coordinates and</span>
<span id="cb1-1105"><a href="#cb1-1105" aria-hidden="true" tabindex="-1"></a>@fig-plasmid-pca-new-coordinates, we show the projection of the first</span>
<span id="cb1-1106"><a href="#cb1-1106" aria-hidden="true" tabindex="-1"></a>two principal components (PCs), respectively.</span>
<span id="cb1-1107"><a href="#cb1-1107" aria-hidden="true" tabindex="-1"></a>@fig-chromosome-pca-new-coordinates separates the manually annotated</span>
<span id="cb1-1108"><a href="#cb1-1108" aria-hidden="true" tabindex="-1"></a>classes, surviving cells separated from non-surviving cells. However,</span>
<span id="cb1-1109"><a href="#cb1-1109" aria-hidden="true" tabindex="-1"></a>for @fig-plasmid-pca-new-coordinates, the class separation was a bit</span>
<span id="cb1-1110"><a href="#cb1-1110" aria-hidden="true" tabindex="-1"></a>rougher but allowed us to separate the surviving filament cells from the</span>
<span id="cb1-1111"><a href="#cb1-1111" aria-hidden="true" tabindex="-1"></a>dead ones.</span>
<span id="cb1-1112"><a href="#cb1-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1113"><a href="#cb1-1113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r experiment-03-chromosome-pca-prep}</span></span>
<span id="cb1-1114"><a href="#cb1-1114" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1115"><a href="#cb1-1115" aria-hidden="true" tabindex="-1"></a>c_pca_rec <span class="ot">&lt;-</span> <span class="fu">recipe</span>(cell_status <span class="sc">~</span> ., <span class="at">data =</span> chromosome_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1116"><a href="#cb1-1116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_naomit</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1117"><a href="#cb1-1117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb1-1118"><a href="#cb1-1118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_pca</span>(<span class="fu">all_predictors</span>())</span>
<span id="cb1-1119"><a href="#cb1-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1120"><a href="#cb1-1120" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-1121"><a href="#cb1-1121" aria-hidden="true" tabindex="-1"></a>c_pca_prep <span class="ot">&lt;-</span> <span class="fu">prep</span>(c_pca_rec)</span>
<span id="cb1-1122"><a href="#cb1-1122" aria-hidden="true" tabindex="-1"></a>c_pca_prep</span>
<span id="cb1-1123"><a href="#cb1-1123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1124"><a href="#cb1-1124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1127"><a href="#cb1-1127" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1128"><a href="#cb1-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-chromosome-pca-new-coordinates</span></span>
<span id="cb1-1129"><a href="#cb1-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Principal Component Analysis of chromosomal strain.</span></span>
<span id="cb1-1130"><a href="#cb1-1130" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1131"><a href="#cb1-1131" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Principal Component Analysis of chromosomal strain.**</span></span>
<span id="cb1-1132"><a href="#cb1-1132" aria-hidden="true" tabindex="-1"></a><span class="co">#|   When integrating the information of different variables in a</span></span>
<span id="cb1-1133"><a href="#cb1-1133" aria-hidden="true" tabindex="-1"></a><span class="co">#|   dimensionality reduction analysis, we observed a clear separation</span></span>
<span id="cb1-1134"><a href="#cb1-1134" aria-hidden="true" tabindex="-1"></a><span class="co">#|   between the surviving cells and those that did not. The contributions</span></span>
<span id="cb1-1135"><a href="#cb1-1135" aria-hidden="true" tabindex="-1"></a><span class="co">#|   that determined this phenomenon come mainly from the last amount of</span></span>
<span id="cb1-1136"><a href="#cb1-1136" aria-hidden="true" tabindex="-1"></a><span class="co">#|   DsRed, GFP, and cell length (see @fig-chromosome-pca-variable-contribution).</span></span>
<span id="cb1-1137"><a href="#cb1-1137" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Although it seems obvious, it effectively confirms that the temporal</span></span>
<span id="cb1-1138"><a href="#cb1-1138" aria-hidden="true" tabindex="-1"></a><span class="co">#|   classification that we carry out makes sense. Longer length represents</span></span>
<span id="cb1-1139"><a href="#cb1-1139" aria-hidden="true" tabindex="-1"></a><span class="co">#|   a greater uptake of antibiotics, but in a much larger volume, so the net</span></span>
<span id="cb1-1140"><a href="#cb1-1140" aria-hidden="true" tabindex="-1"></a><span class="co">#|   effect is an internal reduction of antibiotics</span></span>
<span id="cb1-1141"><a href="#cb1-1141" aria-hidden="true" tabindex="-1"></a><span class="co">#|   (see @fig-cell-dimensions-relationship).</span></span>
<span id="cb1-1142"><a href="#cb1-1142" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1143"><a href="#cb1-1143" aria-hidden="true" tabindex="-1"></a>p_c_pca <span class="ot">&lt;-</span> c_pca_prep <span class="sc">%&gt;%</span> </span>
<span id="cb1-1144"><a href="#cb1-1144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">juice</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-1145"><a href="#cb1-1145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-1146"><a href="#cb1-1146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb1-1147"><a href="#cb1-1147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb1-1148"><a href="#cb1-1148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">seed =</span> <span class="dv">42</span>)) <span class="sc">+</span></span>
<span id="cb1-1149"><a href="#cb1-1149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1150"><a href="#cb1-1150" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>)),</span>
<span id="cb1-1151"><a href="#cb1-1151" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>))</span>
<span id="cb1-1152"><a href="#cb1-1152" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1153"><a href="#cb1-1153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1154"><a href="#cb1-1154" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1155"><a href="#cb1-1155" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1156"><a href="#cb1-1156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1157"><a href="#cb1-1157" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1158"><a href="#cb1-1158" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1159"><a href="#cb1-1159" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1160"><a href="#cb1-1160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1161"><a href="#cb1-1161" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1162"><a href="#cb1-1162" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1163"><a href="#cb1-1163" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-1164"><a href="#cb1-1164" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1165"><a href="#cb1-1165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1166"><a href="#cb1-1166" aria-hidden="true" tabindex="-1"></a>p_c_pca</span>
<span id="cb1-1167"><a href="#cb1-1167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1168"><a href="#cb1-1168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1169"><a href="#cb1-1169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r experiment-03-plasmid-pca-prep}</span></span>
<span id="cb1-1170"><a href="#cb1-1170" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1171"><a href="#cb1-1171" aria-hidden="true" tabindex="-1"></a>p_pca_rec <span class="ot">&lt;-</span> <span class="fu">recipe</span>(cell_status <span class="sc">~</span> ., <span class="at">data =</span> plasmid_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1172"><a href="#cb1-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_naomit</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1173"><a href="#cb1-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb1-1174"><a href="#cb1-1174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_pca</span>(<span class="fu">all_predictors</span>())</span>
<span id="cb1-1175"><a href="#cb1-1175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1176"><a href="#cb1-1176" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-1177"><a href="#cb1-1177" aria-hidden="true" tabindex="-1"></a>p_pca_prep <span class="ot">&lt;-</span> <span class="fu">prep</span>(p_pca_rec)</span>
<span id="cb1-1178"><a href="#cb1-1178" aria-hidden="true" tabindex="-1"></a>p_pca_prep</span>
<span id="cb1-1179"><a href="#cb1-1179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1180"><a href="#cb1-1180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1183"><a href="#cb1-1183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1184"><a href="#cb1-1184" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plasmid-pca-new-coordinates</span></span>
<span id="cb1-1185"><a href="#cb1-1185" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Principal Component Analysis of plasmid strain.</span></span>
<span id="cb1-1186"><a href="#cb1-1186" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1187"><a href="#cb1-1187" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Principal Component Analysis of plasmid strain.**</span></span>
<span id="cb1-1188"><a href="#cb1-1188" aria-hidden="true" tabindex="-1"></a><span class="co">#|   By integrating the information from different variables in a</span></span>
<span id="cb1-1189"><a href="#cb1-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#|   dimensionality reduction analysis, we observed a clear separation</span></span>
<span id="cb1-1190"><a href="#cb1-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#|   between the filamented and non-filamented cells. Said class separation</span></span>
<span id="cb1-1191"><a href="#cb1-1191" aria-hidden="true" tabindex="-1"></a><span class="co">#|   is given by component 2 (Y-axis), which is determined primarily by the</span></span>
<span id="cb1-1192"><a href="#cb1-1192" aria-hidden="true" tabindex="-1"></a><span class="co">#|   initial and final lengths of the cells (see</span></span>
<span id="cb1-1193"><a href="#cb1-1193" aria-hidden="true" tabindex="-1"></a><span class="co">#|   @fig-plasmid-pca-variable-contribution). Furthermore, the classification</span></span>
<span id="cb1-1194"><a href="#cb1-1194" aria-hidden="true" tabindex="-1"></a><span class="co">#|   also allows us to separate those filamented cells that died from those</span></span>
<span id="cb1-1195"><a href="#cb1-1195" aria-hidden="true" tabindex="-1"></a><span class="co">#|   that survived. Therefore, despite the increase in the system's</span></span>
<span id="cb1-1196"><a href="#cb1-1196" aria-hidden="true" tabindex="-1"></a><span class="co">#|   complexity, length plays a role in determining survival.</span></span>
<span id="cb1-1197"><a href="#cb1-1197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1198"><a href="#cb1-1198" aria-hidden="true" tabindex="-1"></a>p_p_pca <span class="ot">&lt;-</span> p_pca_prep <span class="sc">%&gt;%</span> </span>
<span id="cb1-1199"><a href="#cb1-1199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">juice</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb1-1200"><a href="#cb1-1200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-1201"><a href="#cb1-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb1-1202"><a href="#cb1-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb1-1203"><a href="#cb1-1203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb1-1204"><a href="#cb1-1204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="cn">NA</span>,  <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb1-1205"><a href="#cb1-1205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="dv">7</span>)) <span class="sc">+</span></span>
<span id="cb1-1206"><a href="#cb1-1206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">seed =</span> <span class="dv">42</span>)) <span class="sc">+</span></span>
<span id="cb1-1207"><a href="#cb1-1207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1208"><a href="#cb1-1208" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>)),</span>
<span id="cb1-1209"><a href="#cb1-1209" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>))</span>
<span id="cb1-1210"><a href="#cb1-1210" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1211"><a href="#cb1-1211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1212"><a href="#cb1-1212" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1213"><a href="#cb1-1213" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1214"><a href="#cb1-1214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1215"><a href="#cb1-1215" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1216"><a href="#cb1-1216" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1217"><a href="#cb1-1217" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1218"><a href="#cb1-1218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1219"><a href="#cb1-1219" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1220"><a href="#cb1-1220" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1221"><a href="#cb1-1221" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-1222"><a href="#cb1-1222" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1223"><a href="#cb1-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1224"><a href="#cb1-1224" aria-hidden="true" tabindex="-1"></a>p_p_pca</span>
<span id="cb1-1225"><a href="#cb1-1225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1226"><a href="#cb1-1226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1227"><a href="#cb1-1227" aria-hidden="true" tabindex="-1"></a>For their part, in @fig-chromosome-pca-variable-contribution and</span>
<span id="cb1-1228"><a href="#cb1-1228" aria-hidden="true" tabindex="-1"></a>@fig-plasmid-pca-variable-contribution, we show the total contribution</span>
<span id="cb1-1229"><a href="#cb1-1229" aria-hidden="true" tabindex="-1"></a>of each variable per PC for the chromosomal and plasmid strain,</span>
<span id="cb1-1230"><a href="#cb1-1230" aria-hidden="true" tabindex="-1"></a>respectively. Finding that filamentation plays a crucial role in</span>
<span id="cb1-1231"><a href="#cb1-1231" aria-hidden="true" tabindex="-1"></a>determining cell survival. For example, for PC2, we appreciated how the</span>
<span id="cb1-1232"><a href="#cb1-1232" aria-hidden="true" tabindex="-1"></a>variable end DsRed directed the dots to the positive side, while the</span>
<span id="cb1-1233"><a href="#cb1-1233" aria-hidden="true" tabindex="-1"></a>variable end and start length directed the dots to the opposing side.</span>
<span id="cb1-1234"><a href="#cb1-1234" aria-hidden="true" tabindex="-1"></a>Therefore, we can support that filamentation has a role in moving cells</span>
<span id="cb1-1235"><a href="#cb1-1235" aria-hidden="true" tabindex="-1"></a>away from having higher amounts of DsRed.</span>
<span id="cb1-1236"><a href="#cb1-1236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1239"><a href="#cb1-1239" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1240"><a href="#cb1-1240" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-chromosome-pca-variable-contribution</span></span>
<span id="cb1-1241"><a href="#cb1-1241" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Variables contribution of Principal Component Analysis of chromosomal strain.</span></span>
<span id="cb1-1242"><a href="#cb1-1242" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1243"><a href="#cb1-1243" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Variables contribution of Principal Component Analysis of chromosomal strain.**</span></span>
<span id="cb1-1244"><a href="#cb1-1244" aria-hidden="true" tabindex="-1"></a><span class="co">#|   In @fig-chromosome-pca-new-coordinates, we see that the classes we</span></span>
<span id="cb1-1245"><a href="#cb1-1245" aria-hidden="true" tabindex="-1"></a><span class="co">#|   created manually reflected what we observed when performing a reduction</span></span>
<span id="cb1-1246"><a href="#cb1-1246" aria-hidden="true" tabindex="-1"></a><span class="co">#|   of dimensions analysis. Here we show the individual contribution of each</span></span>
<span id="cb1-1247"><a href="#cb1-1247" aria-hidden="true" tabindex="-1"></a><span class="co">#|   variable for the first two components. The variables that most affected</span></span>
<span id="cb1-1248"><a href="#cb1-1248" aria-hidden="true" tabindex="-1"></a><span class="co">#|   components 1 and 2 (X-axis and Y-axis, respectively) are the final</span></span>
<span id="cb1-1249"><a href="#cb1-1249" aria-hidden="true" tabindex="-1"></a><span class="co">#|   measurements of DsRed, GFP, length, and the initial amount of GFP.</span></span>
<span id="cb1-1250"><a href="#cb1-1250" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Given that they are chromosomal strains, we should note that this</span></span>
<span id="cb1-1251"><a href="#cb1-1251" aria-hidden="true" tabindex="-1"></a><span class="co">#|   variability could be produced by intrinsic experimental noise that we</span></span>
<span id="cb1-1252"><a href="#cb1-1252" aria-hidden="true" tabindex="-1"></a><span class="co">#|   could not remove. With that in mind, having the DsRed and the final</span></span>
<span id="cb1-1253"><a href="#cb1-1253" aria-hidden="true" tabindex="-1"></a><span class="co">#|   length highlights the inherent role of cells by having increased their size.</span></span>
<span id="cb1-1254"><a href="#cb1-1254" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1255"><a href="#cb1-1255" aria-hidden="true" tabindex="-1"></a>c_tidied_pca <span class="ot">&lt;-</span> <span class="fu">tidy</span>(c_pca_prep, <span class="dv">3</span>)</span>
<span id="cb1-1256"><a href="#cb1-1256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1257"><a href="#cb1-1257" aria-hidden="true" tabindex="-1"></a>p_c_titied_pca <span class="ot">&lt;-</span> c_tidied_pca <span class="sc">%&gt;%</span></span>
<span id="cb1-1258"><a href="#cb1-1258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(component <span class="sc">%in%</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-1259"><a href="#cb1-1259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1260"><a href="#cb1-1260" aria-hidden="true" tabindex="-1"></a>    <span class="at">component =</span> <span class="fu">fct_inorder</span>(component),</span>
<span id="cb1-1261"><a href="#cb1-1261" aria-hidden="true" tabindex="-1"></a>    <span class="at">terms =</span> <span class="fu">reorder_within</span>(terms, <span class="fu">abs</span>(value), component)</span>
<span id="cb1-1262"><a href="#cb1-1262" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1263"><a href="#cb1-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(</span>
<span id="cb1-1264"><a href="#cb1-1264" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb1-1265"><a href="#cb1-1265" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">abs</span>(value), </span>
<span id="cb1-1266"><a href="#cb1-1266" aria-hidden="true" tabindex="-1"></a>      <span class="at">y=</span> terms,</span>
<span id="cb1-1267"><a href="#cb1-1267" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> value <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb1-1268"><a href="#cb1-1268" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-1269"><a href="#cb1-1269" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1270"><a href="#cb1-1270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb1-1271"><a href="#cb1-1271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(<span class="fu">abs</span>(value), <span class="at">digits =</span> <span class="dv">2</span>)), <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb1-1272"><a href="#cb1-1272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(component <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb1-1273"><a href="#cb1-1273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reordered</span>() <span class="sc">+</span></span>
<span id="cb1-1274"><a href="#cb1-1274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb1-1275"><a href="#cb1-1275" aria-hidden="true" tabindex="-1"></a>      <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb1-1276"><a href="#cb1-1276" aria-hidden="true" tabindex="-1"></a>      <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.8</span>)</span>
<span id="cb1-1277"><a href="#cb1-1277" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1278"><a href="#cb1-1278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1279"><a href="#cb1-1279" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Absolute value of contribution"</span>,</span>
<span id="cb1-1280"><a href="#cb1-1280" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Metric"</span>,</span>
<span id="cb1-1281"><a href="#cb1-1281" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Is positive?"</span></span>
<span id="cb1-1282"><a href="#cb1-1282" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1283"><a href="#cb1-1283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb1-1284"><a href="#cb1-1284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1285"><a href="#cb1-1285" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1286"><a href="#cb1-1286" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-1287"><a href="#cb1-1287" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1288"><a href="#cb1-1288" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-1289"><a href="#cb1-1289" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1290"><a href="#cb1-1290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1291"><a href="#cb1-1291" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#dd5129"</span>, <span class="st">"#43b284"</span>)</span>
<span id="cb1-1292"><a href="#cb1-1292" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-1293"><a href="#cb1-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1294"><a href="#cb1-1294" aria-hidden="true" tabindex="-1"></a>p_c_titied_pca</span>
<span id="cb1-1295"><a href="#cb1-1295" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1296"><a href="#cb1-1296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1299"><a href="#cb1-1299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1300"><a href="#cb1-1300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plasmid-pca-variable-contribution</span></span>
<span id="cb1-1301"><a href="#cb1-1301" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Variables contribution of Principal Component Analysis of plasmid strain.</span></span>
<span id="cb1-1302"><a href="#cb1-1302" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1303"><a href="#cb1-1303" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Variables contribution of Principal Component Analysis of plasmid strain.**</span></span>
<span id="cb1-1304"><a href="#cb1-1304" aria-hidden="true" tabindex="-1"></a><span class="co">#|   In @fig-plasmid-pca-new-coordinates, we saw that we could separate the</span></span>
<span id="cb1-1305"><a href="#cb1-1305" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamented cells from the non-filamented ones. The reduction analysis also</span></span>
<span id="cb1-1306"><a href="#cb1-1306" aria-hidden="true" tabindex="-1"></a><span class="co">#|   shows a slight difference between surviving and dead cells within the</span></span>
<span id="cb1-1307"><a href="#cb1-1307" aria-hidden="true" tabindex="-1"></a><span class="co">#|   small group of filamented cells. Here we offer the individual</span></span>
<span id="cb1-1308"><a href="#cb1-1308" aria-hidden="true" tabindex="-1"></a><span class="co">#|   contribution of each variable for the first two components. For the first</span></span>
<span id="cb1-1309"><a href="#cb1-1309" aria-hidden="true" tabindex="-1"></a><span class="co">#|   component (x-axis in  @fig-chromosome-pca-new-coordinates), the initial</span></span>
<span id="cb1-1310"><a href="#cb1-1310" aria-hidden="true" tabindex="-1"></a><span class="co">#|   and final GFP measurements mainly received the component's variability.</span></span>
<span id="cb1-1311"><a href="#cb1-1311" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We expected this component's importance since it is a chromosomal strain,</span></span>
<span id="cb1-1312"><a href="#cb1-1312" aria-hidden="true" tabindex="-1"></a><span class="co">#|   so we hope its inherent variation will be inherited. On the</span></span>
<span id="cb1-1313"><a href="#cb1-1313" aria-hidden="true" tabindex="-1"></a><span class="co">#|   other hand, the second component (Y-axis in</span></span>
<span id="cb1-1314"><a href="#cb1-1314" aria-hidden="true" tabindex="-1"></a><span class="co">#|   @fig-chromosome-pca-new-coordinates) was determined by the length of</span></span>
<span id="cb1-1315"><a href="#cb1-1315" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the cell. Factors that, in the chromosomal strain (see</span></span>
<span id="cb1-1316"><a href="#cb1-1316" aria-hidden="true" tabindex="-1"></a><span class="co">#|   @fig-chromosome-pca-variable-contribution), determined with the</span></span>
<span id="cb1-1317"><a href="#cb1-1317" aria-hidden="true" tabindex="-1"></a><span class="co">#|   help of DsRed the separation between surviving and dead cells.</span></span>
<span id="cb1-1318"><a href="#cb1-1318" aria-hidden="true" tabindex="-1"></a><span class="co">#|   </span></span>
<span id="cb1-1319"><a href="#cb1-1319" aria-hidden="true" tabindex="-1"></a>p_tidied_pca <span class="ot">&lt;-</span> <span class="fu">tidy</span>(p_pca_prep, <span class="dv">3</span>)</span>
<span id="cb1-1320"><a href="#cb1-1320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1321"><a href="#cb1-1321" aria-hidden="true" tabindex="-1"></a>p_p_titied_pca <span class="ot">&lt;-</span> p_tidied_pca <span class="sc">%&gt;%</span></span>
<span id="cb1-1322"><a href="#cb1-1322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(component <span class="sc">%in%</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-1323"><a href="#cb1-1323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1324"><a href="#cb1-1324" aria-hidden="true" tabindex="-1"></a>    <span class="at">component =</span> <span class="fu">fct_inorder</span>(component),</span>
<span id="cb1-1325"><a href="#cb1-1325" aria-hidden="true" tabindex="-1"></a>    <span class="at">terms =</span> <span class="fu">reorder_within</span>(terms, <span class="fu">abs</span>(value), component)</span>
<span id="cb1-1326"><a href="#cb1-1326" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1327"><a href="#cb1-1327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(</span>
<span id="cb1-1328"><a href="#cb1-1328" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb1-1329"><a href="#cb1-1329" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">abs</span>(value), </span>
<span id="cb1-1330"><a href="#cb1-1330" aria-hidden="true" tabindex="-1"></a>      <span class="at">y=</span> terms,</span>
<span id="cb1-1331"><a href="#cb1-1331" aria-hidden="true" tabindex="-1"></a>      <span class="at">fill =</span> value <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb1-1332"><a href="#cb1-1332" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-1333"><a href="#cb1-1333" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1334"><a href="#cb1-1334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb1-1335"><a href="#cb1-1335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(<span class="fu">abs</span>(value), <span class="at">digits =</span> <span class="dv">2</span>)), <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb1-1336"><a href="#cb1-1336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(component <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb1-1337"><a href="#cb1-1337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reordered</span>() <span class="sc">+</span></span>
<span id="cb1-1338"><a href="#cb1-1338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb1-1339"><a href="#cb1-1339" aria-hidden="true" tabindex="-1"></a>      <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb1-1340"><a href="#cb1-1340" aria-hidden="true" tabindex="-1"></a>      <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.8</span>)</span>
<span id="cb1-1341"><a href="#cb1-1341" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1342"><a href="#cb1-1342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1343"><a href="#cb1-1343" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Absolute value of contribution"</span>,</span>
<span id="cb1-1344"><a href="#cb1-1344" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Metric"</span>,</span>
<span id="cb1-1345"><a href="#cb1-1345" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Is positive?"</span></span>
<span id="cb1-1346"><a href="#cb1-1346" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1347"><a href="#cb1-1347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb1-1348"><a href="#cb1-1348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1349"><a href="#cb1-1349" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1350"><a href="#cb1-1350" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-1351"><a href="#cb1-1351" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1352"><a href="#cb1-1352" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-1353"><a href="#cb1-1353" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1354"><a href="#cb1-1354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1355"><a href="#cb1-1355" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#dd5129"</span>, <span class="st">"#43b284"</span>)</span>
<span id="cb1-1356"><a href="#cb1-1356" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-1357"><a href="#cb1-1357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1358"><a href="#cb1-1358" aria-hidden="true" tabindex="-1"></a>p_p_titied_pca</span>
<span id="cb1-1359"><a href="#cb1-1359" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1360"><a href="#cb1-1360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1361"><a href="#cb1-1361" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Uniform Manifold Approximation and Projection (UMAP) correctly represents the local structure of cell states</span></span>
<span id="cb1-1362"><a href="#cb1-1362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1363"><a href="#cb1-1363" aria-hidden="true" tabindex="-1"></a>Staying with only a one-dimensionality reduction technique was not an</span>
<span id="cb1-1364"><a href="#cb1-1364" aria-hidden="true" tabindex="-1"></a>option, so we used the UMAP technique <span class="co">[</span><span class="ot">@mcinnes2018umap</span><span class="co">]</span>. We mainly</span>
<span id="cb1-1365"><a href="#cb1-1365" aria-hidden="true" tabindex="-1"></a>decided to use UMAP for clustering purposes and see if the annotated</span>
<span id="cb1-1366"><a href="#cb1-1366" aria-hidden="true" tabindex="-1"></a>clusters corresponded to the manually annotated ones. UMAP has certain</span>
<span id="cb1-1367"><a href="#cb1-1367" aria-hidden="true" tabindex="-1"></a>advantages for these purposes, e.g., it preserves the global structure</span>
<span id="cb1-1368"><a href="#cb1-1368" aria-hidden="true" tabindex="-1"></a>across the whole space, so the distances between clusters matter.</span>
<span id="cb1-1369"><a href="#cb1-1369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1370"><a href="#cb1-1370" aria-hidden="true" tabindex="-1"></a>In @fig-chromosome-umap-new-coordinates and</span>
<span id="cb1-1371"><a href="#cb1-1371" aria-hidden="true" tabindex="-1"></a>@fig-plasmid-umap-new-coordinates, we show how, using the same variables</span>
<span id="cb1-1372"><a href="#cb1-1372" aria-hidden="true" tabindex="-1"></a>used in the "PCA" section, UMAP accomplished clustering the four</span>
<span id="cb1-1373"><a href="#cb1-1373" aria-hidden="true" tabindex="-1"></a>proposed classes correctly. Interestingly, in</span>
<span id="cb1-1374"><a href="#cb1-1374" aria-hidden="true" tabindex="-1"></a>@fig-chromosome-umap-new-coordinates, UMAP formed three general groups</span>
<span id="cb1-1375"><a href="#cb1-1375" aria-hidden="true" tabindex="-1"></a>and four for @fig-plasmid-umap-new-coordinates. However, in general,</span>
<span id="cb1-1376"><a href="#cb1-1376" aria-hidden="true" tabindex="-1"></a>UMAP clustered the surviving cells from those that did not survive. On</span>
<span id="cb1-1377"><a href="#cb1-1377" aria-hidden="true" tabindex="-1"></a>investigating why this separation occurred, we found that the large</span>
<span id="cb1-1378"><a href="#cb1-1378" aria-hidden="true" tabindex="-1"></a>groups coalesced into one another if we eliminated the division</span>
<span id="cb1-1379"><a href="#cb1-1379" aria-hidden="true" tabindex="-1"></a>variable. So, in a way, the division also has a role in determining</span>
<span id="cb1-1380"><a href="#cb1-1380" aria-hidden="true" tabindex="-1"></a>survival, but it is not essential or at least not over-represented in</span>
<span id="cb1-1381"><a href="#cb1-1381" aria-hidden="true" tabindex="-1"></a>our data.</span>
<span id="cb1-1382"><a href="#cb1-1382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1385"><a href="#cb1-1385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1386"><a href="#cb1-1386" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: chromosome-umap-prep</span></span>
<span id="cb1-1387"><a href="#cb1-1387" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1388"><a href="#cb1-1388" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1389"><a href="#cb1-1389" aria-hidden="true" tabindex="-1"></a>c_umap_rec <span class="ot">&lt;-</span> <span class="fu">recipe</span>(cell_status <span class="sc">~</span> ., <span class="at">data =</span> chromosome_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1390"><a href="#cb1-1390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_naomit</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1391"><a href="#cb1-1391" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb1-1392"><a href="#cb1-1392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_umap</span>(<span class="fu">all_predictors</span>())</span>
<span id="cb1-1393"><a href="#cb1-1393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1394"><a href="#cb1-1394" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-1395"><a href="#cb1-1395" aria-hidden="true" tabindex="-1"></a>c_umap_prep <span class="ot">&lt;-</span> <span class="fu">prep</span>(c_umap_rec)</span>
<span id="cb1-1396"><a href="#cb1-1396" aria-hidden="true" tabindex="-1"></a>c_umap_prep</span>
<span id="cb1-1397"><a href="#cb1-1397" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1398"><a href="#cb1-1398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1401"><a href="#cb1-1401" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1402"><a href="#cb1-1402" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-chromosome-umap-new-coordinates</span></span>
<span id="cb1-1403"><a href="#cb1-1403" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: UMAP coordinates of chromosome strain.</span></span>
<span id="cb1-1404"><a href="#cb1-1404" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1405"><a href="#cb1-1405" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **UMAP coordinates of chromosome strain.**</span></span>
<span id="cb1-1406"><a href="#cb1-1406" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We represented the cells in a low-dimensional space. This new </span></span>
<span id="cb1-1407"><a href="#cb1-1407" aria-hidden="true" tabindex="-1"></a><span class="co">#|   projection allowed it to group the cells that survived and</span></span>
<span id="cb1-1408"><a href="#cb1-1408" aria-hidden="true" tabindex="-1"></a><span class="co">#|   those that did not. Therefore, as in PCA </span></span>
<span id="cb1-1409"><a href="#cb1-1409" aria-hidden="true" tabindex="-1"></a><span class="co">#|   @fig-chromosome-pca-new-coordinates, this technique supports the manual</span></span>
<span id="cb1-1410"><a href="#cb1-1410" aria-hidden="true" tabindex="-1"></a><span class="co">#|   classification that we carry out.</span></span>
<span id="cb1-1411"><a href="#cb1-1411" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1412"><a href="#cb1-1412" aria-hidden="true" tabindex="-1"></a>p_c_umap <span class="ot">&lt;-</span> <span class="fu">juice</span>(c_umap_prep) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1413"><a href="#cb1-1413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(UMAP1, UMAP2)) <span class="sc">+</span></span>
<span id="cb1-1414"><a href="#cb1-1414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> cell_status), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb1-1415"><a href="#cb1-1415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1416"><a href="#cb1-1416" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>)),</span>
<span id="cb1-1417"><a href="#cb1-1417" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-1418"><a href="#cb1-1418" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1419"><a href="#cb1-1419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1420"><a href="#cb1-1420" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"UMAP 1"</span>,</span>
<span id="cb1-1421"><a href="#cb1-1421" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"UMAP 2"</span>,</span>
<span id="cb1-1422"><a href="#cb1-1422" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1423"><a href="#cb1-1423" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1424"><a href="#cb1-1424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1425"><a href="#cb1-1425" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey92"</span>),</span>
<span id="cb1-1426"><a href="#cb1-1426" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">0.5</span>))</span>
<span id="cb1-1427"><a href="#cb1-1427" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1428"><a href="#cb1-1428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1429"><a href="#cb1-1429" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1430"><a href="#cb1-1430" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1431"><a href="#cb1-1431" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1432"><a href="#cb1-1432" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1433"><a href="#cb1-1433" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1434"><a href="#cb1-1434" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1435"><a href="#cb1-1435" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-1436"><a href="#cb1-1436" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1437"><a href="#cb1-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1438"><a href="#cb1-1438" aria-hidden="true" tabindex="-1"></a>p_c_umap</span>
<span id="cb1-1439"><a href="#cb1-1439" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1440"><a href="#cb1-1440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1443"><a href="#cb1-1443" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1444"><a href="#cb1-1444" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plasmid-umap-prep</span></span>
<span id="cb1-1445"><a href="#cb1-1445" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1446"><a href="#cb1-1446" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1447"><a href="#cb1-1447" aria-hidden="true" tabindex="-1"></a>p_umap_rec <span class="ot">&lt;-</span> <span class="fu">recipe</span>(cell_status <span class="sc">~</span> ., <span class="at">data =</span> plasmid_df) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1448"><a href="#cb1-1448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_naomit</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1449"><a href="#cb1-1449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_predictors</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb1-1450"><a href="#cb1-1450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_umap</span>(<span class="fu">all_predictors</span>())</span>
<span id="cb1-1451"><a href="#cb1-1451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1452"><a href="#cb1-1452" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-1453"><a href="#cb1-1453" aria-hidden="true" tabindex="-1"></a>p_umap_prep <span class="ot">&lt;-</span> <span class="fu">prep</span>(p_umap_rec)</span>
<span id="cb1-1454"><a href="#cb1-1454" aria-hidden="true" tabindex="-1"></a>p_umap_prep</span>
<span id="cb1-1455"><a href="#cb1-1455" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1456"><a href="#cb1-1456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1459"><a href="#cb1-1459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1460"><a href="#cb1-1460" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plasmid-umap-new-coordinates</span></span>
<span id="cb1-1461"><a href="#cb1-1461" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: UMAP coordinates of plasmid strain.</span></span>
<span id="cb1-1462"><a href="#cb1-1462" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1463"><a href="#cb1-1463" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **UMAP coordinates of plasmid strain.**</span></span>
<span id="cb1-1464"><a href="#cb1-1464" aria-hidden="true" tabindex="-1"></a><span class="co">#|   As in @fig-chromosome-umap-new-coordinates, the representation</span></span>
<span id="cb1-1465"><a href="#cb1-1465" aria-hidden="true" tabindex="-1"></a><span class="co">#|   in a low-dimensional space helped classify the cells into four groups,</span></span>
<span id="cb1-1466"><a href="#cb1-1466" aria-hidden="true" tabindex="-1"></a><span class="co">#|   two survivors and two non-survivors.</span></span>
<span id="cb1-1467"><a href="#cb1-1467" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The variable *division* marks the separation of classes. The *division*</span></span>
<span id="cb1-1468"><a href="#cb1-1468" aria-hidden="true" tabindex="-1"></a><span class="co">#|   variable indicates whether a cell divided during its lifetime or not.</span></span>
<span id="cb1-1469"><a href="#cb1-1469" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Together, the UMAP represents the manually assigned classes.</span></span>
<span id="cb1-1470"><a href="#cb1-1470" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1471"><a href="#cb1-1471" aria-hidden="true" tabindex="-1"></a>p_p_umap <span class="ot">&lt;-</span> <span class="fu">juice</span>(p_umap_prep) <span class="sc">%&gt;%</span></span>
<span id="cb1-1472"><a href="#cb1-1472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(UMAP1, UMAP2)) <span class="sc">+</span></span>
<span id="cb1-1473"><a href="#cb1-1473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> cell_status), <span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb1-1474"><a href="#cb1-1474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1475"><a href="#cb1-1475" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"UMAP 1"</span>,</span>
<span id="cb1-1476"><a href="#cb1-1476" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"UMAP 2"</span>,</span>
<span id="cb1-1477"><a href="#cb1-1477" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1478"><a href="#cb1-1478" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1479"><a href="#cb1-1479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1480"><a href="#cb1-1480" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>)),</span>
<span id="cb1-1481"><a href="#cb1-1481" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-1482"><a href="#cb1-1482" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1483"><a href="#cb1-1483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1484"><a href="#cb1-1484" aria-hidden="true" tabindex="-1"></a>       <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"grey92"</span>),</span>
<span id="cb1-1485"><a href="#cb1-1485" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">0.5</span>))</span>
<span id="cb1-1486"><a href="#cb1-1486" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1487"><a href="#cb1-1487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1488"><a href="#cb1-1488" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1489"><a href="#cb1-1489" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1490"><a href="#cb1-1490" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1491"><a href="#cb1-1491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1492"><a href="#cb1-1492" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1493"><a href="#cb1-1493" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1494"><a href="#cb1-1494" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-1495"><a href="#cb1-1495" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1496"><a href="#cb1-1496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1497"><a href="#cb1-1497" aria-hidden="true" tabindex="-1"></a>p_p_umap</span>
<span id="cb1-1498"><a href="#cb1-1498" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1499"><a href="#cb1-1499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1500"><a href="#cb1-1500" aria-hidden="true" tabindex="-1"></a><span class="fu">### Population dynamics reveal how filamentation contributes cell survival</span></span>
<span id="cb1-1501"><a href="#cb1-1501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1504"><a href="#cb1-1504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1505"><a href="#cb1-1505" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb1-1506"><a href="#cb1-1506" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-fitness-cost-measured-in-single-cell-data</span></span>
<span id="cb1-1507"><a href="#cb1-1507" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Fitness cost measured in single-cell data.</span></span>
<span id="cb1-1508"><a href="#cb1-1508" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1509"><a href="#cb1-1509" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Fitness cost measured in single-cell data.**</span></span>
<span id="cb1-1510"><a href="#cb1-1510" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Number of cell divisions before drug.</span></span>
<span id="cb1-1511"><a href="#cb1-1511" aria-hidden="true" tabindex="-1"></a><span class="co">#|   exposure for MG/pBGT (green) and MG:GT (blue). Note that the plasmid-bearing</span></span>
<span id="cb1-1512"><a href="#cb1-1512" aria-hidden="true" tabindex="-1"></a><span class="co">#|   strain presented significantly fewer divisions compared to the chromosomal strain,</span></span>
<span id="cb1-1513"><a href="#cb1-1513" aria-hidden="true" tabindex="-1"></a><span class="co">#|   consistent with prior studies showing that carrying plasmids is associated with</span></span>
<span id="cb1-1514"><a href="#cb1-1514" aria-hidden="true" tabindex="-1"></a><span class="co">#|   a fitness cost in non-selective conditions.</span></span>
<span id="cb1-1515"><a href="#cb1-1515" aria-hidden="true" tabindex="-1"></a>lineages_processed_1_df <span class="sc">|&gt;</span> </span>
<span id="cb1-1516"><a href="#cb1-1516" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(</span>
<span id="cb1-1517"><a href="#cb1-1517" aria-hidden="true" tabindex="-1"></a>        time <span class="sc">&lt;=</span> antibiotic_start_time</span>
<span id="cb1-1518"><a href="#cb1-1518" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span> </span>
<span id="cb1-1519"><a href="#cb1-1519" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(experiment_id, id, <span class="at">wt =</span> division) <span class="sc">|&gt;</span> </span>
<span id="cb1-1520"><a href="#cb1-1520" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb1-1521"><a href="#cb1-1521" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> n,</span>
<span id="cb1-1522"><a href="#cb1-1522" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> experiment_id</span>
<span id="cb1-1523"><a href="#cb1-1523" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb1-1524"><a href="#cb1-1524" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_halfeye</span>(</span>
<span id="cb1-1525"><a href="#cb1-1525" aria-hidden="true" tabindex="-1"></a>        <span class="at">adjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb1-1526"><a href="#cb1-1526" aria-hidden="true" tabindex="-1"></a>        <span class="at">.width =</span> <span class="dv">0</span>,</span>
<span id="cb1-1527"><a href="#cb1-1527" aria-hidden="true" tabindex="-1"></a>        <span class="at">height =</span> <span class="fl">0.6</span>,</span>
<span id="cb1-1528"><a href="#cb1-1528" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_color =</span> <span class="cn">NA</span>,</span>
<span id="cb1-1529"><a href="#cb1-1529" aria-hidden="true" tabindex="-1"></a>        <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">y =</span> <span class="fl">0.3</span>)</span>
<span id="cb1-1530"><a href="#cb1-1530" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1531"><a href="#cb1-1531" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(</span>
<span id="cb1-1532"><a href="#cb1-1532" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb1-1533"><a href="#cb1-1533" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> experiment_id,</span>
<span id="cb1-1534"><a href="#cb1-1534" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> experiment_id</span>
<span id="cb1-1535"><a href="#cb1-1535" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb1-1536"><a href="#cb1-1536" aria-hidden="true" tabindex="-1"></a>        <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">y =</span> <span class="fl">0.2</span>),</span>
<span id="cb1-1537"><a href="#cb1-1537" aria-hidden="true" tabindex="-1"></a>        <span class="at">width =</span> <span class="fl">0.15</span>,</span>
<span id="cb1-1538"><a href="#cb1-1538" aria-hidden="true" tabindex="-1"></a>        <span class="at">outlier.shape =</span> <span class="cn">NA</span></span>
<span id="cb1-1539"><a href="#cb1-1539" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1540"><a href="#cb1-1540" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(</span>
<span id="cb1-1541"><a href="#cb1-1541" aria-hidden="true" tabindex="-1"></a>        <span class="at">position =</span> <span class="fu">position_jitter</span>(</span>
<span id="cb1-1542"><a href="#cb1-1542" aria-hidden="true" tabindex="-1"></a>            <span class="at">width =</span> <span class="fl">0.4</span>,</span>
<span id="cb1-1543"><a href="#cb1-1543" aria-hidden="true" tabindex="-1"></a>            <span class="at">height =</span> <span class="fl">0.03</span>,</span>
<span id="cb1-1544"><a href="#cb1-1544" aria-hidden="true" tabindex="-1"></a>            <span class="at">seed =</span> <span class="dv">42</span></span>
<span id="cb1-1545"><a href="#cb1-1545" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb1-1546"><a href="#cb1-1546" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">200</span></span>
<span id="cb1-1547"><a href="#cb1-1547" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1548"><a href="#cb1-1548" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(</span>
<span id="cb1-1549"><a href="#cb1-1549" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">6</span>,</span>
<span id="cb1-1550"><a href="#cb1-1550" aria-hidden="true" tabindex="-1"></a>        <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb1-1551"><a href="#cb1-1551" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1552"><a href="#cb1-1552" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(</span>
<span id="cb1-1553"><a href="#cb1-1553" aria-hidden="true" tabindex="-1"></a>        <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb1-1554"><a href="#cb1-1554" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1555"><a href="#cb1-1555" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1556"><a href="#cb1-1556" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#D0DEEB"</span>, <span class="st">"#DAEAD5"</span>)</span>
<span id="cb1-1557"><a href="#cb1-1557" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1558"><a href="#cb1-1558" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1559"><a href="#cb1-1559" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#D0DEEB"</span>, <span class="st">"#DAEAD5"</span>)</span>
<span id="cb1-1560"><a href="#cb1-1560" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1561"><a href="#cb1-1561" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb1-1562"><a href="#cb1-1562" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb1-1563"><a href="#cb1-1563" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_line</span>(),</span>
<span id="cb1-1564"><a href="#cb1-1564" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1565"><a href="#cb1-1565" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1566"><a href="#cb1-1566" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1567"><a href="#cb1-1567" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(</span>
<span id="cb1-1568"><a href="#cb1-1568" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">3.5</span></span>
<span id="cb1-1569"><a href="#cb1-1569" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb1-1570"><a href="#cb1-1570" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb1-1571"><a href="#cb1-1571" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb1-1572"><a href="#cb1-1572" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb1-1573"><a href="#cb1-1573" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Number of divisions prior to drug exposure"</span></span>
<span id="cb1-1574"><a href="#cb1-1574" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-1575"><a href="#cb1-1575" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1576"><a href="#cb1-1576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1579"><a href="#cb1-1579" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1580"><a href="#cb1-1580" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: population-create-status-over-time-dataset</span></span>
<span id="cb1-1581"><a href="#cb1-1581" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb1-1582"><a href="#cb1-1582" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb1-1583"><a href="#cb1-1583" aria-hidden="true" tabindex="-1"></a>status_time_df <span class="ot">&lt;-</span> lineages_processed_1_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1584"><a href="#cb1-1584" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(experiment_id, id, time) <span class="sc">%&gt;%</span></span>
<span id="cb1-1585"><a href="#cb1-1585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(experiment_id) <span class="sc">%&gt;%</span></span>
<span id="cb1-1586"><a href="#cb1-1586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_modify</span>(<span class="sc">~</span> <span class="fu">complete</span>(<span class="fu">expand</span>(.x, id, time))) <span class="sc">%&gt;%</span></span>
<span id="cb1-1587"><a href="#cb1-1587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-1588"><a href="#cb1-1588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(lineages_processed_1_df) <span class="sc">%&gt;%</span></span>
<span id="cb1-1589"><a href="#cb1-1589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">cell_status_at_time =</span> filamented_at_time) <span class="sc">%&gt;%</span></span>
<span id="cb1-1590"><a href="#cb1-1590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fill</span>(time_first, time_last, gfp_first, filamentation_threshold, ds_red_threshold, <span class="at">.direction =</span> <span class="st">"up"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-1591"><a href="#cb1-1591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fill</span>(antibiotic_start_time, antibiotic_end_time, <span class="at">.direction =</span> <span class="st">"down"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-1592"><a href="#cb1-1592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">&gt;=</span> time_first) <span class="sc">%&gt;%</span></span>
<span id="cb1-1593"><a href="#cb1-1593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1594"><a href="#cb1-1594" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status_at_time =</span> <span class="fu">as.character</span>(cell_status_at_time),</span>
<span id="cb1-1595"><a href="#cb1-1595" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status_at_time =</span> <span class="fu">replace_na</span>(cell_status_at_time, <span class="st">"Dead"</span>),</span>
<span id="cb1-1596"><a href="#cb1-1596" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_status_at_time =</span> <span class="fu">factor</span>(</span>
<span id="cb1-1597"><a href="#cb1-1597" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> cell_status_at_time,</span>
<span id="cb1-1598"><a href="#cb1-1598" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Not filamented"</span>, <span class="st">"Filamented"</span>, <span class="st">"Dead"</span>)</span>
<span id="cb1-1599"><a href="#cb1-1599" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-1600"><a href="#cb1-1600" aria-hidden="true" tabindex="-1"></a>    <span class="at">time =</span> <span class="fu">factor</span>(time)</span>
<span id="cb1-1601"><a href="#cb1-1601" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1602"><a href="#cb1-1602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">glimpse</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-1603"><a href="#cb1-1603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>()</span>
<span id="cb1-1604"><a href="#cb1-1604" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1605"><a href="#cb1-1605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1606"><a href="#cb1-1606" aria-hidden="true" tabindex="-1"></a>From the full tracking dataset, we evaluated how the different cell</span>
<span id="cb1-1607"><a href="#cb1-1607" aria-hidden="true" tabindex="-1"></a>states behaved over time---for example, understanding how the cells</span>
<span id="cb1-1608"><a href="#cb1-1608" aria-hidden="true" tabindex="-1"></a>absorbed antibiotics or how they elongated in time. In contrast to the</span>
<span id="cb1-1609"><a href="#cb1-1609" aria-hidden="true" tabindex="-1"></a>dataset generated in the @sec-length-gfp-crucial, we did not truncate</span>
<span id="cb1-1610"><a href="#cb1-1610" aria-hidden="true" tabindex="-1"></a>the results 10 minutes after the antibiotic exposure. In this way, we</span>
<span id="cb1-1611"><a href="#cb1-1611" aria-hidden="true" tabindex="-1"></a>could observe cell behavior before and after the presence of the toxic</span>
<span id="cb1-1612"><a href="#cb1-1612" aria-hidden="true" tabindex="-1"></a>agent.</span>
<span id="cb1-1613"><a href="#cb1-1613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1614"><a href="#cb1-1614" aria-hidden="true" tabindex="-1"></a>In @fig-status-with-dead, we observed a small fraction of filamentous</span>
<span id="cb1-1615"><a href="#cb1-1615" aria-hidden="true" tabindex="-1"></a>cells without exposure to the toxic agent in both cell strains. However,</span>
<span id="cb1-1616"><a href="#cb1-1616" aria-hidden="true" tabindex="-1"></a>after antibiotic exposure at minute 60, we observed increases in the</span>
<span id="cb1-1617"><a href="#cb1-1617" aria-hidden="true" tabindex="-1"></a>proportion of filamented cells. It is interesting to note how filamented</span>
<span id="cb1-1618"><a href="#cb1-1618" aria-hidden="true" tabindex="-1"></a>cells grew after antibiotic exposure for the chromosomal strain. We</span>
<span id="cb1-1619"><a href="#cb1-1619" aria-hidden="true" tabindex="-1"></a>speculate that this post-antibiotic growth exists because, once the SOS</span>
<span id="cb1-1620"><a href="#cb1-1620" aria-hidden="true" tabindex="-1"></a>system that triggers filamentation is activated, the system continues to</span>
<span id="cb1-1621"><a href="#cb1-1621" aria-hidden="true" tabindex="-1"></a>grow until it reaches a limit regardless of whether the damaging agent</span>
<span id="cb1-1622"><a href="#cb1-1622" aria-hidden="true" tabindex="-1"></a>is still present [@justiceMorphologicalPlasticityBacterial2008;</span>
<span id="cb1-1623"><a href="#cb1-1623" aria-hidden="true" tabindex="-1"></a>@mückl2018]. Moreover, we observed how the cells start to divide again</span>
<span id="cb1-1624"><a href="#cb1-1624" aria-hidden="true" tabindex="-1"></a>after some time because the proportion of non-filament cells starts to</span>
<span id="cb1-1625"><a href="#cb1-1625" aria-hidden="true" tabindex="-1"></a>grow while the filament cells start to divide. We observed the same</span>
<span id="cb1-1626"><a href="#cb1-1626" aria-hidden="true" tabindex="-1"></a>effects for the plasmid strain. However, the number of filament cells</span>
<span id="cb1-1627"><a href="#cb1-1627" aria-hidden="true" tabindex="-1"></a>expected was much lower by experimental design.</span>
<span id="cb1-1628"><a href="#cb1-1628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1631"><a href="#cb1-1631" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1632"><a href="#cb1-1632" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-status-with-dead</span></span>
<span id="cb1-1633"><a href="#cb1-1633" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Population status over time.</span></span>
<span id="cb1-1634"><a href="#cb1-1634" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1635"><a href="#cb1-1635" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Population status over time.**</span></span>
<span id="cb1-1636"><a href="#cb1-1636" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We calculate how many cells of each type existed for each time point:</span></span>
<span id="cb1-1637"><a href="#cb1-1637" aria-hidden="true" tabindex="-1"></a><span class="co">#|   non-filamented and filamented living cells (green and orange areas,</span></span>
<span id="cb1-1638"><a href="#cb1-1638" aria-hidden="true" tabindex="-1"></a><span class="co">#|   respectively) and dead cells (red area; we considered *dead* cells as those</span></span>
<span id="cb1-1639"><a href="#cb1-1639" aria-hidden="true" tabindex="-1"></a><span class="co">#|   that existed at one time and then stopped tracking). The gray vertical</span></span>
<span id="cb1-1640"><a href="#cb1-1640" aria-hidden="true" tabindex="-1"></a><span class="co">#|   lines represent each experiment's start and end of antibiotic exposure.</span></span>
<span id="cb1-1641"><a href="#cb1-1641" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The experiment was finalized with the resolution of the cells when they</span></span>
<span id="cb1-1642"><a href="#cb1-1642" aria-hidden="true" tabindex="-1"></a><span class="co">#|   returned to their non-filamented state. The effect of filamentation and </span></span>
<span id="cb1-1643"><a href="#cb1-1643" aria-hidden="true" tabindex="-1"></a><span class="co">#|   its spread after exposure to the antibiotic is evident for the chromosomal </span></span>
<span id="cb1-1644"><a href="#cb1-1644" aria-hidden="true" tabindex="-1"></a><span class="co">#|   strain. For its part, for the plasmid strain, it is observed how the</span></span>
<span id="cb1-1645"><a href="#cb1-1645" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamented cells begin to appear slowly. Their proportion is as expected,</span></span>
<span id="cb1-1646"><a href="#cb1-1646" aria-hidden="true" tabindex="-1"></a><span class="co">#|   given that the population had a wide distribution of GFP that allowed</span></span>
<span id="cb1-1647"><a href="#cb1-1647" aria-hidden="true" tabindex="-1"></a><span class="co">#|   them to combat exposure to the antibiotic.</span></span>
<span id="cb1-1648"><a href="#cb1-1648" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1649"><a href="#cb1-1649" aria-hidden="true" tabindex="-1"></a>p_status_with_dead <span class="ot">&lt;-</span> status_time_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1650"><a href="#cb1-1650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">fill =</span> cell_status_at_time)) <span class="sc">+</span></span>
<span id="cb1-1651"><a href="#cb1-1651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb1-1652"><a href="#cb1-1652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">factor</span>(antibiotic_start_time)), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1653"><a href="#cb1-1653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">factor</span>(antibiotic_end_time)), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1654"><a href="#cb1-1654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(experiment_id <span class="sc">~</span> .) <span class="sc">+</span></span>
<span id="cb1-1655"><a href="#cb1-1655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb1-1656"><a href="#cb1-1656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb1-1657"><a href="#cb1-1657" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1658"><a href="#cb1-1658" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.spacing.y =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb1-1659"><a href="#cb1-1659" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1660"><a href="#cb1-1660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb1-1661"><a href="#cb1-1661" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">data.frame</span>(</span>
<span id="cb1-1662"><a href="#cb1-1662" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">c</span>(<span class="fl">7.5</span>, <span class="fl">15.5</span>),</span>
<span id="cb1-1663"><a href="#cb1-1663" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">c</span>(<span class="fl">0.75</span>, <span class="fl">0.25</span>),</span>
<span id="cb1-1664"><a href="#cb1-1664" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> <span class="fu">c</span>(<span class="st">"Start"</span>, <span class="st">"End"</span>),</span>
<span id="cb1-1665"><a href="#cb1-1665" aria-hidden="true" tabindex="-1"></a>      <span class="at">experiment_id =</span> <span class="st">"Plasmid"</span></span>
<span id="cb1-1666"><a href="#cb1-1666" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-1667"><a href="#cb1-1667" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label),</span>
<span id="cb1-1668"><a href="#cb1-1668" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">6</span>,</span>
<span id="cb1-1669"><a href="#cb1-1669" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> 0L,</span>
<span id="cb1-1670"><a href="#cb1-1670" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> 0L,</span>
<span id="cb1-1671"><a href="#cb1-1671" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"white"</span>,</span>
<span id="cb1-1672"><a href="#cb1-1672" aria-hidden="true" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span></span>
<span id="cb1-1673"><a href="#cb1-1673" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1674"><a href="#cb1-1674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1675"><a href="#cb1-1675" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Time (minutes)"</span>,</span>
<span id="cb1-1676"><a href="#cb1-1676" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Percentage of cells"</span>,</span>
<span id="cb1-1677"><a href="#cb1-1677" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1678"><a href="#cb1-1678" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1679"><a href="#cb1-1679" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1680"><a href="#cb1-1680" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#43b284"</span>, <span class="st">"#fab255"</span>, <span class="st">"#dd5129"</span>)</span>
<span id="cb1-1681"><a href="#cb1-1681" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1682"><a href="#cb1-1682" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1683"><a href="#cb1-1683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1684"><a href="#cb1-1684" aria-hidden="true" tabindex="-1"></a>p_status_with_dead</span>
<span id="cb1-1685"><a href="#cb1-1685" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1686"><a href="#cb1-1686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1687"><a href="#cb1-1687" aria-hidden="true" tabindex="-1"></a>In @fig-metrics-over-time, we showed that once antibiotics exposure</span>
<span id="cb1-1688"><a href="#cb1-1688" aria-hidden="true" tabindex="-1"></a>began, those cells that died had a much faster increase in DsRed than</span>
<span id="cb1-1689"><a href="#cb1-1689" aria-hidden="true" tabindex="-1"></a>those that did manage to live, regardless of whether they were</span>
<span id="cb1-1690"><a href="#cb1-1690" aria-hidden="true" tabindex="-1"></a>filamented. On the other hand, surviving cells maintained their</span>
<span id="cb1-1691"><a href="#cb1-1691" aria-hidden="true" tabindex="-1"></a>relatively stable DsRed levels. We noted that length was critical for</span>
<span id="cb1-1692"><a href="#cb1-1692" aria-hidden="true" tabindex="-1"></a>the surviving cells for the chromosomal strain by turning to the GFP and</span>
<span id="cb1-1693"><a href="#cb1-1693" aria-hidden="true" tabindex="-1"></a>length variables for a temporal explanation. Even cells categorized as</span>
<span id="cb1-1694"><a href="#cb1-1694" aria-hidden="true" tabindex="-1"></a>non-filamented reached the filamentation threshold minutes after</span>
<span id="cb1-1695"><a href="#cb1-1695" aria-hidden="true" tabindex="-1"></a>antibiotic exposure. However, the distinction between live or dead</span>
<span id="cb1-1696"><a href="#cb1-1696" aria-hidden="true" tabindex="-1"></a>filamented cells was not as evident as expected. As for cells with</span>
<span id="cb1-1697"><a href="#cb1-1697" aria-hidden="true" tabindex="-1"></a>plasmids, the effect on GFP for surviving cells was maintained for</span>
<span id="cb1-1698"><a href="#cb1-1698" aria-hidden="true" tabindex="-1"></a>filamented cells and decreased for non-filamented cells. For the</span>
<span id="cb1-1699"><a href="#cb1-1699" aria-hidden="true" tabindex="-1"></a>filament cells that died, we showed that they had, on average, a much</span>
<span id="cb1-1700"><a href="#cb1-1700" aria-hidden="true" tabindex="-1"></a>longer initial length than the surviving cells. We also consider it</span>
<span id="cb1-1701"><a href="#cb1-1701" aria-hidden="true" tabindex="-1"></a>necessary to understand which variables affect cell survival.</span>
<span id="cb1-1702"><a href="#cb1-1702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1705"><a href="#cb1-1705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1706"><a href="#cb1-1706" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-metrics-over-time</span></span>
<span id="cb1-1707"><a href="#cb1-1707" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Population measurements over time.</span></span>
<span id="cb1-1708"><a href="#cb1-1708" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1709"><a href="#cb1-1709" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Population measurements over time.**</span></span>
<span id="cb1-1710"><a href="#cb1-1710" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The colored lines symbolize the average value of each metric at each</span></span>
<span id="cb1-1711"><a href="#cb1-1711" aria-hidden="true" tabindex="-1"></a><span class="co">#|   instant of time, while its surrounding shaded area represents the</span></span>
<span id="cb1-1712"><a href="#cb1-1712" aria-hidden="true" tabindex="-1"></a><span class="co">#|   95% confidence interval. The vertical lines represent the start and</span></span>
<span id="cb1-1713"><a href="#cb1-1713" aria-hidden="true" tabindex="-1"></a><span class="co">#|   end of antibiotic exposure. The horizontal line in the length metric</span></span>
<span id="cb1-1714"><a href="#cb1-1714" aria-hidden="true" tabindex="-1"></a><span class="co">#|   symbolizes the threshold to consider a cell filament.</span></span>
<span id="cb1-1715"><a href="#cb1-1715" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Regarding the GFP metric, the behavior is relatively stable for the</span></span>
<span id="cb1-1716"><a href="#cb1-1716" aria-hidden="true" tabindex="-1"></a><span class="co">#|   chromosomal strain. We observed a faster increase of DsRed for the</span></span>
<span id="cb1-1717"><a href="#cb1-1717" aria-hidden="true" tabindex="-1"></a><span class="co">#|   non-surviving populations in both experiments.</span></span>
<span id="cb1-1718"><a href="#cb1-1718" aria-hidden="true" tabindex="-1"></a><span class="co">#|   In contrast, for the plasmid strain, a decline in GFP</span></span>
<span id="cb1-1719"><a href="#cb1-1719" aria-hidden="true" tabindex="-1"></a><span class="co">#|   is observed for the population that did not survive. For the length metric,</span></span>
<span id="cb1-1720"><a href="#cb1-1720" aria-hidden="true" tabindex="-1"></a><span class="co">#|   it is interesting to note how the chromosome cells that did not filament</span></span>
<span id="cb1-1721"><a href="#cb1-1721" aria-hidden="true" tabindex="-1"></a><span class="co">#|   continued to grow past the filamentation threshold once the exposure to</span></span>
<span id="cb1-1722"><a href="#cb1-1722" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the antibiotic in the chromosomal strain had ended. On the other hand, the</span></span>
<span id="cb1-1723"><a href="#cb1-1723" aria-hidden="true" tabindex="-1"></a><span class="co">#|   filamented and dead cells seem to have a greater length from the beginning</span></span>
<span id="cb1-1724"><a href="#cb1-1724" aria-hidden="true" tabindex="-1"></a><span class="co">#|   for the plasmid strain.</span></span>
<span id="cb1-1725"><a href="#cb1-1725" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1726"><a href="#cb1-1726" aria-hidden="true" tabindex="-1"></a>p_metrics_over_time <span class="ot">&lt;-</span> lineages_processed_1_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1727"><a href="#cb1-1727" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(experiment_id, cell_status, time, length, gfp, ds_red) <span class="sc">%&gt;%</span></span>
<span id="cb1-1728"><a href="#cb1-1728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb1-1729"><a href="#cb1-1729" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(length, gfp, ds_red),</span>
<span id="cb1-1730"><a href="#cb1-1730" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"metric"</span></span>
<span id="cb1-1731"><a href="#cb1-1731" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1732"><a href="#cb1-1732" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1733"><a href="#cb1-1733" aria-hidden="true" tabindex="-1"></a>    <span class="at">metric =</span> <span class="fu">case_when</span>(</span>
<span id="cb1-1734"><a href="#cb1-1734" aria-hidden="true" tabindex="-1"></a>      metric <span class="sc">==</span> <span class="st">"ds_red"</span> <span class="sc">~</span> <span class="st">"DsRed"</span>,</span>
<span id="cb1-1735"><a href="#cb1-1735" aria-hidden="true" tabindex="-1"></a>      metric <span class="sc">==</span> <span class="st">"gfp"</span> <span class="sc">~</span> <span class="st">"GFP"</span>,</span>
<span id="cb1-1736"><a href="#cb1-1736" aria-hidden="true" tabindex="-1"></a>      metric <span class="sc">==</span> <span class="st">"length"</span> <span class="sc">~</span> <span class="st">"Length"</span></span>
<span id="cb1-1737"><a href="#cb1-1737" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-1738"><a href="#cb1-1738" aria-hidden="true" tabindex="-1"></a>    <span class="at">filamentation_threshold =</span> <span class="fu">ifelse</span>(metric <span class="sc">==</span> <span class="st">"length"</span>, filamentation_threshold, <span class="cn">NA</span>)</span>
<span id="cb1-1739"><a href="#cb1-1739" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1740"><a href="#cb1-1740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(experiment_id, cell_status, time, metric) <span class="sc">%&gt;%</span></span>
<span id="cb1-1741"><a href="#cb1-1741" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb1-1742"><a href="#cb1-1742" aria-hidden="true" tabindex="-1"></a>    <span class="at">ci =</span> <span class="fu">list</span>(<span class="fu">mean_cl_normal</span>(value)),</span>
<span id="cb1-1743"><a href="#cb1-1743" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb1-1744"><a href="#cb1-1744" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1745"><a href="#cb1-1745" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(ci)) <span class="sc">%&gt;%</span></span>
<span id="cb1-1746"><a href="#cb1-1746" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb1-1747"><a href="#cb1-1747" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> lineages_processed_1_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1748"><a href="#cb1-1748" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(experiment_id, antibiotic_start_time, antibiotic_end_time, filamentation_threshold) <span class="sc">%&gt;%</span></span>
<span id="cb1-1749"><a href="#cb1-1749" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(),</span>
<span id="cb1-1750"><a href="#cb1-1750" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"experiment_id"</span>)</span>
<span id="cb1-1751"><a href="#cb1-1751" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1752"><a href="#cb1-1752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1753"><a href="#cb1-1753" aria-hidden="true" tabindex="-1"></a>    <span class="at">filamentation_threshold =</span> <span class="fu">ifelse</span>(metric <span class="sc">==</span> <span class="st">"Length"</span>, filamentation_threshold, <span class="cn">NA</span>)</span>
<span id="cb1-1754"><a href="#cb1-1754" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1755"><a href="#cb1-1755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> y, <span class="at">ymin =</span> ymin, <span class="at">ymax =</span> ymax, <span class="at">color =</span> cell_status, <span class="at">fill =</span> cell_status)) <span class="sc">+</span></span>
<span id="cb1-1756"><a href="#cb1-1756" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> antibiotic_start_time), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1757"><a href="#cb1-1757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> antibiotic_end_time), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1758"><a href="#cb1-1758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="fu">as.numeric</span>(filamentation_threshold)), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1759"><a href="#cb1-1759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"loess"</span>) <span class="sc">+</span></span>
<span id="cb1-1760"><a href="#cb1-1760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> experiment_id, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb1-1761"><a href="#cb1-1761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1762"><a href="#cb1-1762" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Time (minutes)"</span>,</span>
<span id="cb1-1763"><a href="#cb1-1763" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Value"</span>,</span>
<span id="cb1-1764"><a href="#cb1-1764" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span>,</span>
<span id="cb1-1765"><a href="#cb1-1765" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-1766"><a href="#cb1-1766" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1767"><a href="#cb1-1767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1768"><a href="#cb1-1768" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>),</span>
<span id="cb1-1769"><a href="#cb1-1769" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-1770"><a href="#cb1-1770" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1771"><a href="#cb1-1771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb1-1772"><a href="#cb1-1772" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1773"><a href="#cb1-1773" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1774"><a href="#cb1-1774" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1775"><a href="#cb1-1775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-1776"><a href="#cb1-1776" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> cell_status_pallete,</span>
<span id="cb1-1777"><a href="#cb1-1777" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> cell_status_legend_order</span>
<span id="cb1-1778"><a href="#cb1-1778" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb1-1779"><a href="#cb1-1779" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1780"><a href="#cb1-1780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1781"><a href="#cb1-1781" aria-hidden="true" tabindex="-1"></a>p_metrics_over_time</span>
<span id="cb1-1782"><a href="#cb1-1782" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1783"><a href="#cb1-1783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1784"><a href="#cb1-1784" aria-hidden="true" tabindex="-1"></a><span class="fu">### Heterogeneity in plasmid copy-number allows various forms of survival in addition to filamentation</span></span>
<span id="cb1-1785"><a href="#cb1-1785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1786"><a href="#cb1-1786" aria-hidden="true" tabindex="-1"></a>We are confident that filamentation has a fundamental role in</span>
<span id="cb1-1787"><a href="#cb1-1787" aria-hidden="true" tabindex="-1"></a>determining cell survival, as we have shown so far. However, for plasmid</span>
<span id="cb1-1788"><a href="#cb1-1788" aria-hidden="true" tabindex="-1"></a>cells, we have a component of our complete interest; heterogeneity. Each</span>
<span id="cb1-1789"><a href="#cb1-1789" aria-hidden="true" tabindex="-1"></a>cell can possess a different plasmid copy number; thus, each could show</span>
<span id="cb1-1790"><a href="#cb1-1790" aria-hidden="true" tabindex="-1"></a>a different behavior under stress <span class="co">[</span><span class="ot">@sanmillan2016</span><span class="co">]</span>. For instance,</span>
<span id="cb1-1791"><a href="#cb1-1791" aria-hidden="true" tabindex="-1"></a>heterogeneity can produce resistant cells that do not suffer damage,</span>
<span id="cb1-1792"><a href="#cb1-1792" aria-hidden="true" tabindex="-1"></a>susceptible cells, and cells that form filaments to mitigate</span>
<span id="cb1-1793"><a href="#cb1-1793" aria-hidden="true" tabindex="-1"></a>environmental stress.</span>
<span id="cb1-1794"><a href="#cb1-1794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1795"><a href="#cb1-1795" aria-hidden="true" tabindex="-1"></a>To study the effect of variability in plasmid copy number on the</span>
<span id="cb1-1796"><a href="#cb1-1796" aria-hidden="true" tabindex="-1"></a>survival probability of the population, we decided to group cells by the</span>
<span id="cb1-1797"><a href="#cb1-1797" aria-hidden="true" tabindex="-1"></a>proportion of initial GFP with respect to the population maximum. We</span>
<span id="cb1-1798"><a href="#cb1-1798" aria-hidden="true" tabindex="-1"></a>defined 100% of the population as the number of total cells at the onset</span>
<span id="cb1-1799"><a href="#cb1-1799" aria-hidden="true" tabindex="-1"></a>of antibiotic exposure. @fig-proportion-living-cells-gfp-by-row shows</span>
<span id="cb1-1800"><a href="#cb1-1800" aria-hidden="true" tabindex="-1"></a>how the cells with the highest amount of GFP remained unchanged once</span>
<span id="cb1-1801"><a href="#cb1-1801" aria-hidden="true" tabindex="-1"></a>antibiotic exposure began, while the rest of the cells started to</span>
<span id="cb1-1802"><a href="#cb1-1802" aria-hidden="true" tabindex="-1"></a>decrease their percentage of surviving cells. However, the decrease was</span>
<span id="cb1-1803"><a href="#cb1-1803" aria-hidden="true" tabindex="-1"></a>not linear. On the contrary, we observed a bi-modal distribution in the</span>
<span id="cb1-1804"><a href="#cb1-1804" aria-hidden="true" tabindex="-1"></a>reduction of live cells. An average GFP point provided higher survival</span>
<span id="cb1-1805"><a href="#cb1-1805" aria-hidden="true" tabindex="-1"></a>than a point below or above the average (except for cells very close to</span>
<span id="cb1-1806"><a href="#cb1-1806" aria-hidden="true" tabindex="-1"></a>the population maximum).</span>
<span id="cb1-1807"><a href="#cb1-1807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1810"><a href="#cb1-1810" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1811"><a href="#cb1-1811" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-proportion-living-cells-gfp-by-row</span></span>
<span id="cb1-1812"><a href="#cb1-1812" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Population survivals binned by initial GFP over time.</span></span>
<span id="cb1-1813"><a href="#cb1-1813" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1814"><a href="#cb1-1814" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Population survivals binned by initial GFP over time.**</span></span>
<span id="cb1-1815"><a href="#cb1-1815" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We categorized the cells' GFP into ranges of proportions 0.05 concerning</span></span>
<span id="cb1-1816"><a href="#cb1-1816" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the maximum amount of GFP in the population. 100% cells per bin of GFP</span></span>
<span id="cb1-1817"><a href="#cb1-1817" aria-hidden="true" tabindex="-1"></a><span class="co">#|   was taken as the number of cells one frame before the start of exposure</span></span>
<span id="cb1-1818"><a href="#cb1-1818" aria-hidden="true" tabindex="-1"></a><span class="co">#|   to the antibiotic (minute 50). Therefore, dark to light colors represent a</span></span>
<span id="cb1-1819"><a href="#cb1-1819" aria-hidden="true" tabindex="-1"></a><span class="co">#|   generation of new cells, and light to dark colors the death of cells.</span></span>
<span id="cb1-1820"><a href="#cb1-1820" aria-hidden="true" tabindex="-1"></a><span class="co">#|   The black vertical bars represent the start and end of the antibiotic</span></span>
<span id="cb1-1821"><a href="#cb1-1821" aria-hidden="true" tabindex="-1"></a><span class="co">#|   exposure. Bars size and color on the right represent the percentage of</span></span>
<span id="cb1-1822"><a href="#cb1-1822" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the living cells 10 minutes after the end of the experiment. As shown in</span></span>
<span id="cb1-1823"><a href="#cb1-1823" aria-hidden="true" tabindex="-1"></a><span class="co">#|   @fig-gfp-survival-probability, we showed that the surviving cells appear</span></span>
<span id="cb1-1824"><a href="#cb1-1824" aria-hidden="true" tabindex="-1"></a><span class="co">#|   to follow something similar to a bimodal distribution. More cells survive</span></span>
<span id="cb1-1825"><a href="#cb1-1825" aria-hidden="true" tabindex="-1"></a><span class="co">#|   with a moderate amount of GFP or with an amount close to the maximum of</span></span>
<span id="cb1-1826"><a href="#cb1-1826" aria-hidden="true" tabindex="-1"></a><span class="co">#|   the population.</span></span>
<span id="cb1-1827"><a href="#cb1-1827" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1828"><a href="#cb1-1828" aria-hidden="true" tabindex="-1"></a>counts_survived_by_gfp <span class="ot">&lt;-</span> status_time_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1829"><a href="#cb1-1829" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(experiment_id <span class="sc">==</span> <span class="st">"Plasmid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-1830"><a href="#cb1-1830" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1831"><a href="#cb1-1831" aria-hidden="true" tabindex="-1"></a>    <span class="at">gfp_first =</span> gfp_first <span class="sc">/</span> <span class="fu">max</span>(gfp_first),</span>
<span id="cb1-1832"><a href="#cb1-1832" aria-hidden="true" tabindex="-1"></a>    <span class="at">gfp_first =</span> <span class="fu">cut</span>(gfp_first, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.05</span>))</span>
<span id="cb1-1833"><a href="#cb1-1833" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1834"><a href="#cb1-1834" aria-hidden="true" tabindex="-1"></a>  <span class="fu">with_groups</span>(</span>
<span id="cb1-1835"><a href="#cb1-1835" aria-hidden="true" tabindex="-1"></a>    gfp_first,</span>
<span id="cb1-1836"><a href="#cb1-1836" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">mutate</span>(.x, <span class="at">n_at_gfp =</span> <span class="fu">filter</span>(<span class="fu">cur_data</span>(), time <span class="sc">==</span> antibiotic_start_time) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>())</span>
<span id="cb1-1837"><a href="#cb1-1837" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1838"><a href="#cb1-1838" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1839"><a href="#cb1-1839" aria-hidden="true" tabindex="-1"></a>    <span class="at">gfp_first =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(</span>
<span id="cb1-1840"><a href="#cb1-1840" aria-hidden="true" tabindex="-1"></a>      <span class="at">string =</span> gfp_first,</span>
<span id="cb1-1841"><a href="#cb1-1841" aria-hidden="true" tabindex="-1"></a>      <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">d+.</span><span class="sc">\\</span><span class="st">d+"</span></span>
<span id="cb1-1842"><a href="#cb1-1842" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-1843"><a href="#cb1-1843" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1844"><a href="#cb1-1844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(time, gfp_first) <span class="sc">%&gt;%</span></span>
<span id="cb1-1845"><a href="#cb1-1845" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb1-1846"><a href="#cb1-1846" aria-hidden="true" tabindex="-1"></a>    <span class="at">percentage_alive =</span> <span class="fu">sum</span>(cell_status_at_time <span class="sc">!=</span> <span class="st">"Dead"</span>) <span class="sc">/</span> <span class="fu">first</span>(n_at_gfp),</span>
<span id="cb1-1847"><a href="#cb1-1847" aria-hidden="true" tabindex="-1"></a>    <span class="at">antibiotic_start_time =</span> <span class="fu">first</span>(antibiotic_start_time),</span>
<span id="cb1-1848"><a href="#cb1-1848" aria-hidden="true" tabindex="-1"></a>    <span class="at">antibiotic_end_time =</span> <span class="fu">first</span>(antibiotic_end_time),</span>
<span id="cb1-1849"><a href="#cb1-1849" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb1-1850"><a href="#cb1-1850" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1851"><a href="#cb1-1851" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>()</span>
<span id="cb1-1852"><a href="#cb1-1852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1853"><a href="#cb1-1853" aria-hidden="true" tabindex="-1"></a>survived_p1 <span class="ot">&lt;-</span> counts_survived_by_gfp <span class="sc">%&gt;%</span></span>
<span id="cb1-1854"><a href="#cb1-1854" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1855"><a href="#cb1-1855" aria-hidden="true" tabindex="-1"></a>    <span class="at">time =</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(time)) <span class="sc">-</span> antibiotic_start_time,</span>
<span id="cb1-1856"><a href="#cb1-1856" aria-hidden="true" tabindex="-1"></a>    <span class="at">antibiotic_end_time =</span> antibiotic_end_time <span class="sc">-</span> antibiotic_start_time</span>
<span id="cb1-1857"><a href="#cb1-1857" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1858"><a href="#cb1-1858" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb1-1859"><a href="#cb1-1859" aria-hidden="true" tabindex="-1"></a>    time <span class="sc">&gt;=</span> <span class="dv">0</span></span>
<span id="cb1-1860"><a href="#cb1-1860" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1861"><a href="#cb1-1861" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1862"><a href="#cb1-1862" aria-hidden="true" tabindex="-1"></a>    <span class="at">time =</span> <span class="fu">as.factor</span>(time)</span>
<span id="cb1-1863"><a href="#cb1-1863" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1864"><a href="#cb1-1864" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> gfp_first, <span class="at">fill =</span> percentage_alive)) <span class="sc">+</span></span>
<span id="cb1-1865"><a href="#cb1-1865" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb1-1866"><a href="#cb1-1866" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">factor</span>(antibiotic_end_time)), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb1-1867"><a href="#cb1-1867" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb1-1868"><a href="#cb1-1868" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb1-1869"><a href="#cb1-1869" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>(</span>
<span id="cb1-1870"><a href="#cb1-1870" aria-hidden="true" tabindex="-1"></a>    <span class="at">option =</span> <span class="st">"inferno"</span>,</span>
<span id="cb1-1871"><a href="#cb1-1871" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_percent</span>(),</span>
<span id="cb1-1872"><a href="#cb1-1872" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1873"><a href="#cb1-1873" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1874"><a href="#cb1-1874" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Time (minutes)"</span>,</span>
<span id="cb1-1875"><a href="#cb1-1875" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Proportion of GFP"</span>,</span>
<span id="cb1-1876"><a href="#cb1-1876" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Percentage of living cells"</span></span>
<span id="cb1-1877"><a href="#cb1-1877" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1878"><a href="#cb1-1878" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1879"><a href="#cb1-1879" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.spacing.x =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"cm"</span>),</span>
<span id="cb1-1880"><a href="#cb1-1880" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb1-1881"><a href="#cb1-1881" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="st">"right"</span></span>
<span id="cb1-1882"><a href="#cb1-1882" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1883"><a href="#cb1-1883" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb1-1884"><a href="#cb1-1884" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_colorbar</span>(</span>
<span id="cb1-1885"><a href="#cb1-1885" aria-hidden="true" tabindex="-1"></a>      <span class="at">barwidth =</span> <span class="dv">8</span>,</span>
<span id="cb1-1886"><a href="#cb1-1886" aria-hidden="true" tabindex="-1"></a>      <span class="at">title.position =</span> <span class="st">"top"</span></span>
<span id="cb1-1887"><a href="#cb1-1887" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-1888"><a href="#cb1-1888" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1889"><a href="#cb1-1889" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(</span></span>
<span id="cb1-1890"><a href="#cb1-1890" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   data = data.frame(x = 4.76, y = 17.6, label = "Antibiotic Exposure"),</span></span>
<span id="cb1-1891"><a href="#cb1-1891" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   mapping = aes(x = x, y = y, label = label),</span></span>
<span id="cb1-1892"><a href="#cb1-1892" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   size = 4,</span></span>
<span id="cb1-1893"><a href="#cb1-1893" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   fontface = 2,</span></span>
<span id="cb1-1894"><a href="#cb1-1894" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   color = "black",</span></span>
<span id="cb1-1895"><a href="#cb1-1895" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   alpha = 0.8,</span></span>
<span id="cb1-1896"><a href="#cb1-1896" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   inherit.aes = FALSE</span></span>
<span id="cb1-1897"><a href="#cb1-1897" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ) +</span></span>
<span id="cb1-1898"><a href="#cb1-1898" aria-hidden="true" tabindex="-1"></a>  cowplot<span class="sc">::</span><span class="fu">draw_label</span>(</span>
<span id="cb1-1899"><a href="#cb1-1899" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Antibiotic exposure"</span>,</span>
<span id="cb1-1900"><a href="#cb1-1900" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">4.7</span>,</span>
<span id="cb1-1901"><a href="#cb1-1901" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fl">17.5</span>,</span>
<span id="cb1-1902"><a href="#cb1-1902" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">12</span></span>
<span id="cb1-1903"><a href="#cb1-1903" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1904"><a href="#cb1-1904" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(</span>
<span id="cb1-1905"><a href="#cb1-1905" aria-hidden="true" tabindex="-1"></a>    <span class="at">grob =</span> grid<span class="sc">::</span><span class="fu">linesGrob</span>(</span>
<span id="cb1-1906"><a href="#cb1-1906" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="st">"npc"</span>),</span>
<span id="cb1-1907"><a href="#cb1-1907" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="st">"npc"</span>),</span>
<span id="cb1-1908"><a href="#cb1-1908" aria-hidden="true" tabindex="-1"></a>      <span class="at">gp =</span> grid<span class="sc">::</span><span class="fu">gpar</span>(<span class="at">lty =</span> <span class="st">"dashed"</span>)</span>
<span id="cb1-1909"><a href="#cb1-1909" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-1910"><a href="#cb1-1910" aria-hidden="true" tabindex="-1"></a>    <span class="at">xmin =</span> <span class="fl">0.5</span>,</span>
<span id="cb1-1911"><a href="#cb1-1911" aria-hidden="true" tabindex="-1"></a>    <span class="at">xmax =</span> <span class="fl">1.7</span>,</span>
<span id="cb1-1912"><a href="#cb1-1912" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> <span class="dv">17</span>,</span>
<span id="cb1-1913"><a href="#cb1-1913" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymax =</span> <span class="fl">15.5</span></span>
<span id="cb1-1914"><a href="#cb1-1914" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1915"><a href="#cb1-1915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(</span>
<span id="cb1-1916"><a href="#cb1-1916" aria-hidden="true" tabindex="-1"></a>    <span class="at">grob =</span> grid<span class="sc">::</span><span class="fu">linesGrob</span>(</span>
<span id="cb1-1917"><a href="#cb1-1917" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="st">"npc"</span>),</span>
<span id="cb1-1918"><a href="#cb1-1918" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>), <span class="st">"npc"</span>),</span>
<span id="cb1-1919"><a href="#cb1-1919" aria-hidden="true" tabindex="-1"></a>      <span class="at">gp =</span> grid<span class="sc">::</span><span class="fu">gpar</span>(<span class="at">lty =</span> <span class="st">"dashed"</span>)</span>
<span id="cb1-1920"><a href="#cb1-1920" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-1921"><a href="#cb1-1921" aria-hidden="true" tabindex="-1"></a>    <span class="at">xmin =</span> <span class="fl">7.8</span>,</span>
<span id="cb1-1922"><a href="#cb1-1922" aria-hidden="true" tabindex="-1"></a>    <span class="at">xmax =</span> <span class="dv">9</span>,</span>
<span id="cb1-1923"><a href="#cb1-1923" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> <span class="dv">17</span>,</span>
<span id="cb1-1924"><a href="#cb1-1924" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymax =</span> <span class="cn">Inf</span></span>
<span id="cb1-1925"><a href="#cb1-1925" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1926"><a href="#cb1-1926" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">"off"</span>) <span class="sc">+</span></span>
<span id="cb1-1927"><a href="#cb1-1927" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1928"><a href="#cb1-1928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1929"><a href="#cb1-1929" aria-hidden="true" tabindex="-1"></a>survived_p2 <span class="ot">&lt;-</span> counts_survived_by_gfp <span class="sc">%&gt;%</span></span>
<span id="cb1-1930"><a href="#cb1-1930" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> antibiotic_end_time <span class="sc">+</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-1931"><a href="#cb1-1931" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> percentage_alive <span class="sc">*</span> <span class="dv">100</span>, <span class="at">y =</span> gfp_first, <span class="at">fill =</span> percentage_alive)) <span class="sc">+</span></span>
<span id="cb1-1932"><a href="#cb1-1932" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb1-1933"><a href="#cb1-1933" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>(</span>
<span id="cb1-1934"><a href="#cb1-1934" aria-hidden="true" tabindex="-1"></a>    <span class="at">option =</span> <span class="st">"inferno"</span>,</span>
<span id="cb1-1935"><a href="#cb1-1935" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> scales<span class="sc">::</span>percent,</span>
<span id="cb1-1936"><a href="#cb1-1936" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="fu">min</span>(counts_survived_by_gfp<span class="sc">$</span>percentage_alive), <span class="fu">max</span>(counts_survived_by_gfp<span class="sc">$</span>percentage_alive))</span>
<span id="cb1-1937"><a href="#cb1-1937" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1938"><a href="#cb1-1938" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb1-1939"><a href="#cb1-1939" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb1-1940"><a href="#cb1-1940" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1941"><a href="#cb1-1941" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-1942"><a href="#cb1-1942" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb1-1943"><a href="#cb1-1943" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1944"><a href="#cb1-1944" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-1945"><a href="#cb1-1945" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Survival (%)"</span></span>
<span id="cb1-1946"><a href="#cb1-1946" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-1947"><a href="#cb1-1947" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_number</span>(<span class="at">digits =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb1-1948"><a href="#cb1-1948" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb1-1949"><a href="#cb1-1949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1950"><a href="#cb1-1950" aria-hidden="true" tabindex="-1"></a>p_proportions_by_gfp_row <span class="ot">&lt;-</span> (survived_p1 <span class="sc">|</span> survived_p2) <span class="sc">+</span></span>
<span id="cb1-1951"><a href="#cb1-1951" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(</span>
<span id="cb1-1952"><a href="#cb1-1952" aria-hidden="true" tabindex="-1"></a>    <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">3</span>)</span>
<span id="cb1-1953"><a href="#cb1-1953" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-1954"><a href="#cb1-1954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1955"><a href="#cb1-1955" aria-hidden="true" tabindex="-1"></a>p_proportions_by_gfp_row</span>
<span id="cb1-1956"><a href="#cb1-1956" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1957"><a href="#cb1-1957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1958"><a href="#cb1-1958" aria-hidden="true" tabindex="-1"></a>Therefore, what we observed was a bimodal distribution for GFP-dependent</span>
<span id="cb1-1959"><a href="#cb1-1959" aria-hidden="true" tabindex="-1"></a>cell survival. In order to show this effect more clearly, in</span>
<span id="cb1-1960"><a href="#cb1-1960" aria-hidden="true" tabindex="-1"></a>@fig-gfp-survival-probability, we plotted the survival probability for</span>
<span id="cb1-1961"><a href="#cb1-1961" aria-hidden="true" tabindex="-1"></a>each GFP bin without normalizing for the population maximum. This new</span>
<span id="cb1-1962"><a href="#cb1-1962" aria-hidden="true" tabindex="-1"></a>plot allowed us to observe how the bimodal survival distribution occurs</span>
<span id="cb1-1963"><a href="#cb1-1963" aria-hidden="true" tabindex="-1"></a>for cells that did not grow as filaments, whereas cells that filament</span>
<span id="cb1-1964"><a href="#cb1-1964" aria-hidden="true" tabindex="-1"></a>increase their survival probability gradually as they have more initial</span>
<span id="cb1-1965"><a href="#cb1-1965" aria-hidden="true" tabindex="-1"></a>GFP (see also @fig-gfp-temporal-distribution).-distribution).</span>
<span id="cb1-1966"><a href="#cb1-1966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1969"><a href="#cb1-1969" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1970"><a href="#cb1-1970" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-gfp-survival-probability</span></span>
<span id="cb1-1971"><a href="#cb1-1971" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Plasmid initial GFP survival probability.</span></span>
<span id="cb1-1972"><a href="#cb1-1972" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-1973"><a href="#cb1-1973" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Plasmid initial GFP survival probability.**</span></span>
<span id="cb1-1974"><a href="#cb1-1974" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We calculated the survival probability after comparing the population</span></span>
<span id="cb1-1975"><a href="#cb1-1975" aria-hidden="true" tabindex="-1"></a><span class="co">#|   distributions of GFP with those of the cells that managed to survive.</span></span>
<span id="cb1-1976"><a href="#cb1-1976" aria-hidden="true" tabindex="-1"></a><span class="co">#|   To assess survival by GFP, we only used plasmid cells. For non-filamented</span></span>
<span id="cb1-1977"><a href="#cb1-1977" aria-hidden="true" tabindex="-1"></a><span class="co">#|   cells (blue dots), a bell forms with an upturned tail. On the other hand,</span></span>
<span id="cb1-1978"><a href="#cb1-1978" aria-hidden="true" tabindex="-1"></a><span class="co">#|   for the filamented cells (red dots), a continuous increase in survival is</span></span>
<span id="cb1-1979"><a href="#cb1-1979" aria-hidden="true" tabindex="-1"></a><span class="co">#|   shown just when it seems that the probability of the non-filamented cells</span></span>
<span id="cb1-1980"><a href="#cb1-1980" aria-hidden="true" tabindex="-1"></a><span class="co">#|   has decreased. In global, much GFP has higher resistance, but an average</span></span>
<span id="cb1-1981"><a href="#cb1-1981" aria-hidden="true" tabindex="-1"></a><span class="co">#|   GFP value without filamentation also increases the probability of survival.</span></span>
<span id="cb1-1982"><a href="#cb1-1982" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-1983"><a href="#cb1-1983" aria-hidden="true" tabindex="-1"></a>step <span class="ot">&lt;-</span> <span class="fl">0.04</span></span>
<span id="cb1-1984"><a href="#cb1-1984" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(lineages_processed_1_df<span class="sc">$</span>gfp) <span class="sc">-</span> step, <span class="fu">max</span>(lineages_processed_1_df<span class="sc">$</span>gfp) <span class="sc">+</span> step, step)</span>
<span id="cb1-1985"><a href="#cb1-1985" aria-hidden="true" tabindex="-1"></a>hist_gfp_control_info <span class="ot">&lt;-</span> <span class="fu">hist</span>(</span>
<span id="cb1-1986"><a href="#cb1-1986" aria-hidden="true" tabindex="-1"></a>  lineages_processed_1_df<span class="sc">$</span>gfp,</span>
<span id="cb1-1987"><a href="#cb1-1987" aria-hidden="true" tabindex="-1"></a>  <span class="at">breaks =</span> breaks,</span>
<span id="cb1-1988"><a href="#cb1-1988" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span></span>
<span id="cb1-1989"><a href="#cb1-1989" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1990"><a href="#cb1-1990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1991"><a href="#cb1-1991" aria-hidden="true" tabindex="-1"></a>p_survival_probability_gfp <span class="ot">&lt;-</span> lineages_processed_1_df <span class="sc">%&gt;%</span></span>
<span id="cb1-1992"><a href="#cb1-1992" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(experiment_id <span class="sc">==</span> <span class="st">"Plasmid"</span>, survived <span class="sc">==</span> <span class="st">"Survived"</span>, time <span class="sc">==</span> time_first) <span class="sc">%&gt;%</span></span>
<span id="cb1-1993"><a href="#cb1-1993" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(filamented_id) <span class="sc">%&gt;%</span></span>
<span id="cb1-1994"><a href="#cb1-1994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb1-1995"><a href="#cb1-1995" aria-hidden="true" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">list</span>(<span class="fu">hist</span>(gfp, <span class="at">plot =</span> <span class="cn">FALSE</span>, <span class="at">breaks =</span> breaks)<span class="sc">$</span>counts)</span>
<span id="cb1-1996"><a href="#cb1-1996" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-1997"><a href="#cb1-1997" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(counts) <span class="sc">%&gt;%</span></span>
<span id="cb1-1998"><a href="#cb1-1998" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-1999"><a href="#cb1-1999" aria-hidden="true" tabindex="-1"></a>    <span class="at">mids =</span> <span class="fu">rep</span>(hist_gfp_control_info<span class="sc">$</span>mids, <span class="dv">2</span>),</span>
<span id="cb1-2000"><a href="#cb1-2000" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_counts =</span> <span class="fu">rep</span>(hist_gfp_control_info<span class="sc">$</span>counts, <span class="at">times =</span> <span class="dv">2</span>),</span>
<span id="cb1-2001"><a href="#cb1-2001" aria-hidden="true" tabindex="-1"></a>    <span class="at">survival_probability =</span> counts <span class="sc">/</span> control_counts</span>
<span id="cb1-2002"><a href="#cb1-2002" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-2003"><a href="#cb1-2003" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-2004"><a href="#cb1-2004" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter(survival_probability != 0) %&gt;%</span></span>
<span id="cb1-2005"><a href="#cb1-2005" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mids, <span class="at">y =</span> survival_probability, <span class="at">color =</span> filamented_id)) <span class="sc">+</span></span>
<span id="cb1-2006"><a href="#cb1-2006" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-2007"><a href="#cb1-2007" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(</span>
<span id="cb1-2008"><a href="#cb1-2008" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-2009"><a href="#cb1-2009" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb1-2010"><a href="#cb1-2010" aria-hidden="true" tabindex="-1"></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span></span>
<span id="cb1-2011"><a href="#cb1-2011" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-2012"><a href="#cb1-2012" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>() <span class="sc">+</span></span>
<span id="cb1-2013"><a href="#cb1-2013" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb1-2014"><a href="#cb1-2014" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-2015"><a href="#cb1-2015" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#43b284"</span>, <span class="st">"#fab255"</span>)</span>
<span id="cb1-2016"><a href="#cb1-2016" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-2017"><a href="#cb1-2017" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-2018"><a href="#cb1-2018" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Initial GFP"</span>,</span>
<span id="cb1-2019"><a href="#cb1-2019" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Survival probability"</span>,</span>
<span id="cb1-2020"><a href="#cb1-2020" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-2021"><a href="#cb1-2021" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-2022"><a href="#cb1-2022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2023"><a href="#cb1-2023" aria-hidden="true" tabindex="-1"></a>p_survival_probability_gfp</span>
<span id="cb1-2024"><a href="#cb1-2024" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-2025"><a href="#cb1-2025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2026"><a href="#cb1-2026" aria-hidden="true" tabindex="-1"></a>As in @fig-gfp-survival-probability, in</span>
<span id="cb1-2027"><a href="#cb1-2027" aria-hidden="true" tabindex="-1"></a>@fig-length-survival-probability, we show the survival probability given</span>
<span id="cb1-2028"><a href="#cb1-2028" aria-hidden="true" tabindex="-1"></a>an initial length. We observe that survival is higher for cells that did</span>
<span id="cb1-2029"><a href="#cb1-2029" aria-hidden="true" tabindex="-1"></a>not grow as filaments if the initial length was less than the average.</span>
<span id="cb1-2030"><a href="#cb1-2030" aria-hidden="true" tabindex="-1"></a>In contrast, for filamented cells, the survival probability increased as</span>
<span id="cb1-2031"><a href="#cb1-2031" aria-hidden="true" tabindex="-1"></a>cell length was longer at the beginning of the experiment (see also</span>
<span id="cb1-2032"><a href="#cb1-2032" aria-hidden="true" tabindex="-1"></a>@fig-length-temporal-distribution). However, it is noteworthy that the</span>
<span id="cb1-2033"><a href="#cb1-2033" aria-hidden="true" tabindex="-1"></a>probability of survival had a limit in which a higher initial length</span>
<span id="cb1-2034"><a href="#cb1-2034" aria-hidden="true" tabindex="-1"></a>meant a lower probability of survival (see red dotted lines in</span>
<span id="cb1-2035"><a href="#cb1-2035" aria-hidden="true" tabindex="-1"></a>@fig-length-survival-probability).</span>
<span id="cb1-2036"><a href="#cb1-2036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2039"><a href="#cb1-2039" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-2040"><a href="#cb1-2040" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-length-survival-probability</span></span>
<span id="cb1-2041"><a href="#cb1-2041" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-scap: Plasmid initial length survival probability.</span></span>
<span id="cb1-2042"><a href="#cb1-2042" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: &gt;</span></span>
<span id="cb1-2043"><a href="#cb1-2043" aria-hidden="true" tabindex="-1"></a><span class="co">#|   **Plasmid initial length survival probability.**</span></span>
<span id="cb1-2044"><a href="#cb1-2044" aria-hidden="true" tabindex="-1"></a><span class="co">#|   We calculated the survival probability after comparing the population</span></span>
<span id="cb1-2045"><a href="#cb1-2045" aria-hidden="true" tabindex="-1"></a><span class="co">#|   distributions of length with those of the cells that managed to survive.</span></span>
<span id="cb1-2046"><a href="#cb1-2046" aria-hidden="true" tabindex="-1"></a><span class="co">#|   For non-filamented cells (blue dots), the survival probability is higher</span></span>
<span id="cb1-2047"><a href="#cb1-2047" aria-hidden="true" tabindex="-1"></a><span class="co">#|   for those cells with small initial lengths, while It seems to decrease with</span></span>
<span id="cb1-2048"><a href="#cb1-2048" aria-hidden="true" tabindex="-1"></a><span class="co">#|   a more extensive initial size. For their part, for filamented cells</span></span>
<span id="cb1-2049"><a href="#cb1-2049" aria-hidden="true" tabindex="-1"></a><span class="co">#|   (red dots), the probability of survival increases according to their</span></span>
<span id="cb1-2050"><a href="#cb1-2050" aria-hidden="true" tabindex="-1"></a><span class="co">#|   length but then declines when the cells are too long at first (see red</span></span>
<span id="cb1-2051"><a href="#cb1-2051" aria-hidden="true" tabindex="-1"></a><span class="co">#|   dotted line). Therefore, in general, a small and moderate length or an</span></span>
<span id="cb1-2052"><a href="#cb1-2052" aria-hidden="true" tabindex="-1"></a><span class="co">#|   initial length already filamented from the beginning increases the</span></span>
<span id="cb1-2053"><a href="#cb1-2053" aria-hidden="true" tabindex="-1"></a><span class="co">#|   chances of survival.</span></span>
<span id="cb1-2054"><a href="#cb1-2054" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb1-2055"><a href="#cb1-2055" aria-hidden="true" tabindex="-1"></a>step <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb1-2056"><a href="#cb1-2056" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(lineages_processed_1_df<span class="sc">$</span>length) <span class="sc">-</span> step, <span class="fu">max</span>(lineages_processed_1_df<span class="sc">$</span>length) <span class="sc">+</span> step, step)</span>
<span id="cb1-2057"><a href="#cb1-2057" aria-hidden="true" tabindex="-1"></a>hist_length_control_info <span class="ot">&lt;-</span> <span class="fu">hist</span>(</span>
<span id="cb1-2058"><a href="#cb1-2058" aria-hidden="true" tabindex="-1"></a>  lineages_processed_1_df<span class="sc">$</span>length,</span>
<span id="cb1-2059"><a href="#cb1-2059" aria-hidden="true" tabindex="-1"></a>  <span class="at">breaks =</span> breaks,</span>
<span id="cb1-2060"><a href="#cb1-2060" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span></span>
<span id="cb1-2061"><a href="#cb1-2061" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-2062"><a href="#cb1-2062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2063"><a href="#cb1-2063" aria-hidden="true" tabindex="-1"></a>survival_probability_length <span class="ot">&lt;-</span> lineages_processed_1_df <span class="sc">%&gt;%</span></span>
<span id="cb1-2064"><a href="#cb1-2064" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(experiment_id <span class="sc">==</span> <span class="st">"Plasmid"</span>, survived <span class="sc">==</span> <span class="st">"Survived"</span>, time <span class="sc">==</span> time_first) <span class="sc">%&gt;%</span></span>
<span id="cb1-2065"><a href="#cb1-2065" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(filamented_id) <span class="sc">%&gt;%</span></span>
<span id="cb1-2066"><a href="#cb1-2066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb1-2067"><a href="#cb1-2067" aria-hidden="true" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">list</span>(<span class="fu">hist</span>(length, <span class="at">plot =</span> <span class="cn">FALSE</span>, <span class="at">breaks =</span> breaks)<span class="sc">$</span>counts)</span>
<span id="cb1-2068"><a href="#cb1-2068" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-2069"><a href="#cb1-2069" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(counts) <span class="sc">%&gt;%</span></span>
<span id="cb1-2070"><a href="#cb1-2070" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-2071"><a href="#cb1-2071" aria-hidden="true" tabindex="-1"></a>    <span class="at">mids =</span> <span class="fu">rep</span>(hist_length_control_info<span class="sc">$</span>mids, <span class="dv">2</span>),</span>
<span id="cb1-2072"><a href="#cb1-2072" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_counts =</span> <span class="fu">rep</span>(hist_length_control_info<span class="sc">$</span>counts, <span class="at">times =</span> <span class="dv">2</span>),</span>
<span id="cb1-2073"><a href="#cb1-2073" aria-hidden="true" tabindex="-1"></a>    <span class="at">survival_probability =</span> counts <span class="sc">/</span> control_counts</span>
<span id="cb1-2074"><a href="#cb1-2074" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-2075"><a href="#cb1-2075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">identity</span>()</span>
<span id="cb1-2076"><a href="#cb1-2076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2077"><a href="#cb1-2077" aria-hidden="true" tabindex="-1"></a>p_survival_probability_length <span class="ot">&lt;-</span> survival_probability_length <span class="sc">%&gt;%</span></span>
<span id="cb1-2078"><a href="#cb1-2078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(survival_probability <span class="sc">!=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-2079"><a href="#cb1-2079" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mids, <span class="at">y =</span> survival_probability, <span class="at">color =</span> filamented_id)) <span class="sc">+</span></span>
<span id="cb1-2080"><a href="#cb1-2080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-2081"><a href="#cb1-2081" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(</span>
<span id="cb1-2082"><a href="#cb1-2082" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-2083"><a href="#cb1-2083" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb1-2084"><a href="#cb1-2084" aria-hidden="true" tabindex="-1"></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span></span>
<span id="cb1-2085"><a href="#cb1-2085" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-2086"><a href="#cb1-2086" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>() <span class="sc">+</span></span>
<span id="cb1-2087"><a href="#cb1-2087" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb1-2088"><a href="#cb1-2088" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb1-2089"><a href="#cb1-2089" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#43b284"</span>, <span class="st">"#fab255"</span>)</span>
<span id="cb1-2090"><a href="#cb1-2090" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-2091"><a href="#cb1-2091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb1-2092"><a href="#cb1-2092" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Initial length"</span>,</span>
<span id="cb1-2093"><a href="#cb1-2093" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Survival probability"</span>,</span>
<span id="cb1-2094"><a href="#cb1-2094" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Cell status"</span></span>
<span id="cb1-2095"><a href="#cb1-2095" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb1-2096"><a href="#cb1-2096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(</span>
<span id="cb1-2097"><a href="#cb1-2097" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="dv">120</span>),</span>
<span id="cb1-2098"><a href="#cb1-2098" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>)</span>
<span id="cb1-2099"><a href="#cb1-2099" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-2100"><a href="#cb1-2100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2101"><a href="#cb1-2101" aria-hidden="true" tabindex="-1"></a>p_survival_probability_length</span>
<span id="cb1-2102"><a href="#cb1-2102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-2103"><a href="#cb1-2103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2104"><a href="#cb1-2104" aria-hidden="true" tabindex="-1"></a><span class="fu">## Discussion</span></span>
<span id="cb1-2105"><a href="#cb1-2105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2106"><a href="#cb1-2106" aria-hidden="true" tabindex="-1"></a>Here, we evaluated different variables that could determine cell</span>
<span id="cb1-2107"><a href="#cb1-2107" aria-hidden="true" tabindex="-1"></a>survival upon exposure to toxic agents by studying two experimental</span>
<span id="cb1-2108"><a href="#cb1-2108" aria-hidden="true" tabindex="-1"></a>populations of *E. coli*, one strain with a resistance gene on the</span>
<span id="cb1-2109"><a href="#cb1-2109" aria-hidden="true" tabindex="-1"></a>chromosome and the other on multicopy plasmids. We identified two</span>
<span id="cb1-2110"><a href="#cb1-2110" aria-hidden="true" tabindex="-1"></a>variables that are predominantly responsible for cell survival: cell</span>
<span id="cb1-2111"><a href="#cb1-2111" aria-hidden="true" tabindex="-1"></a>length and GFP amount related to the cell's inherent resistance to the</span>
<span id="cb1-2112"><a href="#cb1-2112" aria-hidden="true" tabindex="-1"></a>toxic agent and heterogeneity in response times.</span>
<span id="cb1-2113"><a href="#cb1-2113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2114"><a href="#cb1-2114" aria-hidden="true" tabindex="-1"></a>On the other hand, as other studies have already mentioned</span>
<span id="cb1-2115"><a href="#cb1-2115" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@heinrich2015; @wang2009</span><span class="co">]</span>, we examined cell activity and youth in a</span>
<span id="cb1-2116"><a href="#cb1-2116" aria-hidden="true" tabindex="-1"></a>minimalistic way. While the distribution of the number of divisions</span>
<span id="cb1-2117"><a href="#cb1-2117" aria-hidden="true" tabindex="-1"></a>exemplifies a broader and more uniform range for the surviving cells,</span>
<span id="cb1-2118"><a href="#cb1-2118" aria-hidden="true" tabindex="-1"></a>the cells that died tended to have fewer divisions. However, for the</span>
<span id="cb1-2119"><a href="#cb1-2119" aria-hidden="true" tabindex="-1"></a>study of cellular youth at the time of exposure to the toxic agent, the</span>
<span id="cb1-2120"><a href="#cb1-2120" aria-hidden="true" tabindex="-1"></a>results did not show a clear pattern of behavior for cell fate</span>
<span id="cb1-2121"><a href="#cb1-2121" aria-hidden="true" tabindex="-1"></a>determination. Therefore, it would be interesting to study cellular</span>
<span id="cb1-2122"><a href="#cb1-2122" aria-hidden="true" tabindex="-1"></a>youth at a higher level of complexity in future studies to understand</span>
<span id="cb1-2123"><a href="#cb1-2123" aria-hidden="true" tabindex="-1"></a>its contribution to cell survival.</span>
<span id="cb1-2124"><a href="#cb1-2124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2125"><a href="#cb1-2125" aria-hidden="true" tabindex="-1"></a>Interestingly, when we used temporal measurements of cell length, GFP,</span>
<span id="cb1-2126"><a href="#cb1-2126" aria-hidden="true" tabindex="-1"></a>DsRed, and if a cell divided, we could recapitulate, for the most part,</span>
<span id="cb1-2127"><a href="#cb1-2127" aria-hidden="true" tabindex="-1"></a>the fates of cellular states (see @sec-length-gfp-crucial and</span>
<span id="cb1-2128"><a href="#cb1-2128" aria-hidden="true" tabindex="-1"></a>@sec-unsupervised-classification). Thus, increasing the system's</span>
<span id="cb1-2129"><a href="#cb1-2129" aria-hidden="true" tabindex="-1"></a>complexity led to better clustering of cell states, but not how these</span>
<span id="cb1-2130"><a href="#cb1-2130" aria-hidden="true" tabindex="-1"></a>factors interact biologically in determining cell survival. Therefore,</span>
<span id="cb1-2131"><a href="#cb1-2131" aria-hidden="true" tabindex="-1"></a>we decided to postulate a mathematical model that helps us understand</span>
<span id="cb1-2132"><a href="#cb1-2132" aria-hidden="true" tabindex="-1"></a>the critical components of cell survival.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>