<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.140">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bacterial filamentation: a bet for survival in stressful environments - References</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./98-appendices.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">References</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bacterial filamentation: a bet for survival in stressful environments</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/jvelezmagic/bacterial-filamentation-research" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="./Bacterial-filamentation--a-bet-for-survival-in-stressful-environments.pdf" title="Download PDF" class="sidebar-tool px-1"><i class="bi bi-file-pdf"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
            <i class="bi bi-bi-linkedin pe-1"></i>
          LinkedIn
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Welcome</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00-introduction.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-image-processing.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Image processing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-experiment-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Experiment analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-model-analysis.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Models to the rescue; filamentation abstraction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-discussion.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./98-appendices.html" class="sidebar-item-text sidebar-link">Appendix</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link active">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">References</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-ackermann2015" class="csl-entry" role="doc-biblioentry">
Ackermann, Martin. 2015. <span>“A Functional Perspective on Phenotypic
Heterogeneity in Microorganisms.”</span> <em>Nature Reviews
Microbiology</em> 13 (8): 497–508. <a href="https://doi.org/10.1038/nrmicro3491">https://doi.org/10.1038/nrmicro3491</a>.
</div>
<div id="ref-R-quarto" class="csl-entry" role="doc-biblioentry">
Allaire, JJ. 2022. <em>Quarto: R Interface to Quarto Markdown Publishing
System</em>. <a href="https://github.com/quarto-dev/quarto-r">https://github.com/quarto-dev/quarto-r</a>.
</div>
<div id="ref-R-rmarkdown" class="csl-entry" role="doc-biblioentry">
Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin
Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and
Richard Iannone. 2022. <em>Rmarkdown: Dynamic Documents for r</em>. <a href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>.
</div>
<div id="ref-altman1992" class="csl-entry" role="doc-biblioentry">
Altman, N. S. 1992. <span>“An Introduction to Kernel and
Nearest-Neighbor Nonparametric Regression.”</span> <em>The American
Statistician</em> 46 (3): 175–85. <a href="https://doi.org/10.1080/00031305.1992.10475879">https://doi.org/10.1080/00031305.1992.10475879</a>.
</div>
<div id="ref-andersson2005" class="csl-entry" role="doc-biblioentry">
Andersson, Dan I. 2005. <span>“The Ways in Which Bacteria Resist
Antibiotics.”</span> <em>International Journal of Risk and Safety in
Medicine</em> 17 (3-4): 111–16.
</div>
<div id="ref-andrewsDeterminationMinimumInhibitory2002" class="csl-entry" role="doc-biblioentry">
Andrews, J. M. 2002. <span>“Determination of Minimum Inhibitory
Concentrations.”</span> <em>Journal of Antimicrobial Chemotherapy</em>
49 (6): 1049–49. <a href="https://doi.org/10.1093/jac/dkf083">https://doi.org/10.1093/jac/dkf083</a>.
</div>
<div id="ref-andrews2001" class="csl-entry" role="doc-biblioentry">
Andrews, Jennifer M. 2001. <span>“Determination of Minimum Inhibitory
Concentrations.”</span> <em>Journal of Antimicrobial Chemotherapy</em>
48 (suppl<span>_</span>1): 5–16. <a href="https://doi.org/10.1093/jac/48.suppl_1.5">https://doi.org/10.1093/jac/48.suppl_1.5</a>.
</div>
<div id="ref-balaban2004a" class="csl-entry" role="doc-biblioentry">
Balaban, Nathalie Q., Jack Merrin, Remy Chait, Lukasz Kowalik, and
Stanislas Leibler. 2004. <span>“Bacterial Persistence as a Phenotypic
Switch.”</span> <em>Science</em> 305 (5690): 1622–25. <a href="https://doi.org/10.1126/science.1099390">https://doi.org/10.1126/science.1099390</a>.
</div>
<div id="ref-Julia-2017" class="csl-entry" role="doc-biblioentry">
Bezanson, Jeff, Alan Edelman, Stefan Karpinski, and Viral B Shah. 2017.
<span>“Julia: A Fresh Approach to Numerical Computing.”</span> <em>SIAM
<span>R</span>eview</em> 59 (1): 65–98. <a href="https://doi.org/10.1137/141000671">https://doi.org/10.1137/141000671</a>.
</div>
<div id="ref-bosEmergenceAntibioticResistance2015" class="csl-entry" role="doc-biblioentry">
Bos, Julia, Qiucen Zhang, Saurabh Vyawahare, Elizabeth Rogers, Susan M.
Rosenberg, and Robert H. Austin. 2015. <span>“Emergence of Antibiotic
Resistance from Multinucleated Bacterial Filaments.”</span>
<em>Proceedings of the National Academy of Sciences</em> 112 (1):
178–83. <a href="https://doi.org/10.1073/pnas.1420702111">https://doi.org/10.1073/pnas.1420702111</a>.
</div>
<div id="ref-10.5555/1386553" class="csl-entry" role="doc-biblioentry">
Brinkmann, Ron. 2008. <em>The Art and Science of Digital Compositing,
Second Edition: Techniques for Visual Effects, Animation and Motion
Graphics (the Morgan Kaufmann Series in ... Morgan Kaufmann Series in
Computer Graphics)</em>. 2nd ed. San Francisco, CA, USA: Morgan Kaufmann
Publishers Inc.
</div>
<div id="ref-bruggeman" class="csl-entry" role="doc-biblioentry">
Bruggeman, Frank J., Jorrit J. Hornberg, Fred C. Boogerd, and Hans V.
Westerhoff. 2007. <span>“Introduction to Systems Biology.”</span> In,
1–19. Birkhäuser Basel. <a href="https://doi.org/10.1007/978-3-7643-7439-6_1">https://doi.org/10.1007/978-3-7643-7439-6_1</a>.
</div>
<div id="ref-Caicedo2017" class="csl-entry" role="doc-biblioentry">
Caicedo, Juan C, Sam Cooper, Florian Heigwer, Scott Warchal, Peng Qiu,
Csaba Molnar, Aliaksei S Vasilevich, et al. 2017. <span>“Data-Analysis
Strategies for Image-Based Cell Profiling.”</span> <em>Nature
Methods</em> 14 (9): 849–63. <a href="https://doi.org/10.1038/nmeth.4397">https://doi.org/10.1038/nmeth.4397</a>.
</div>
<div id="ref-camposConstantSizeExtension2014" class="csl-entry" role="doc-biblioentry">
Campos, Manuel, Ivan V. Surovtsev, Setsu Kato, Ahmad Paintdakhi, Bruno
Beltran, Sarah E. Ebmeier, and Christine Jacobs-Wagner. 2014. <span>“A
<span>Constant Size Extension Drives Bacterial Cell Size
Homeostasis</span>.”</span> <em>Cell</em> 159 (6): 1433–46. <a href="https://doi.org/10.1016/j.cell.2014.11.022">https://doi.org/10.1016/j.cell.2014.11.022</a>.
</div>
<div id="ref-cayron2020" class="csl-entry" role="doc-biblioentry">
Cayron, Julien, Annick Dedieu, and Christian Lesterlin. 2020.
<span>“Bacterial Filament Division Dynamics Allows Rapid Post-Stress
Cell Proliferation.”</span> <a href="http://dx.doi.org/10.1101/2020.03.16.993345">http://dx.doi.org/10.1101/2020.03.16.993345</a>.
</div>
<div id="ref-convery2019" class="csl-entry" role="doc-biblioentry">
Convery, Neil, and Nikolaj Gadegaard. 2019. <span>“30 Years of
Microfluidics.”</span> <em>Micro and Nano Engineering</em> 2 (March):
76–91. <a href="https://doi.org/10.1016/j.mne.2019.01.003">https://doi.org/10.1016/j.mne.2019.01.003</a>.
</div>
<div id="ref-datseris2020" class="csl-entry" role="doc-biblioentry">
Datseris, George, Jonas Isensee, Sebastian Pech, and Tamás Gál. 2020.
<span>“DrWatson: The Perfect Sidekick for Your Scientific
Inquiries.”</span> <em>Journal of Open Source Software</em> 5 (54):
2673. <a href="https://doi.org/10.21105/joss.02673">https://doi.org/10.21105/joss.02673</a>.
</div>
<div id="ref-dever1991" class="csl-entry" role="doc-biblioentry">
Dever, L. A., and T. S. Dermody. 1991. <span>“Mechanisms of bacterial
resistance to antibiotics.”</span> <em>Archives of Internal
Medicine</em> 151 (5): 886–95.
</div>
<div id="ref-EditorialBoard2014" class="csl-entry" role="doc-biblioentry">
<span>“Editorial <span>Board</span>.”</span> 2014. <em>Journal of Global
Antimicrobial Resistance</em> 2 (2): ii. <a href="https://doi.org/10.1016/S2213-7165(14)00044-7">https://doi.org/10.1016/S2213-7165(14)00044-7</a>.
</div>
<div id="ref-elowitz2002" class="csl-entry" role="doc-biblioentry">
Elowitz, Michael B., Arnold J. Levine, Eric D. Siggia, and Peter S.
Swain. 2002. <span>“Stochastic Gene Expression in a Single Cell.”</span>
<em>Science</em> 297 (5584): 1183–86. <a href="https://doi.org/10.1126/science.1070919">https://doi.org/10.1126/science.1070919</a>.
</div>
<div id="ref-R-janitor" class="csl-entry" role="doc-biblioentry">
Firke, Sam. 2021. <em>Janitor: Simple Tools for Examining and Cleaning
Dirty Data</em>. <a href="https://github.com/sfirke/janitor">https://github.com/sfirke/janitor</a>.
</div>
<div id="ref-friedmanPreciseTemporalModulation2005" class="csl-entry" role="doc-biblioentry">
Friedman, Nir, Shuki Vardi, Michal Ronen, Uri Alon, and Joel Stavans.
2005. <span>“Precise <span>Temporal Modulation</span> in the
<span>Response</span> of the <span>SOS DNA Repair Network</span> in
<span>Individual Bacteria</span>.”</span> Edited by B’en’edicte Michel.
<em>PLoS Biology</em> 3 (7): e238. <a href="https://doi.org/10.1371/journal.pbio.0030238">https://doi.org/10.1371/journal.pbio.0030238</a>.
</div>
<div id="ref-shapely2007" class="csl-entry" role="doc-biblioentry">
Gillies, Sean et al. 2007--. <span>“Shapely: Manipulation and Analysis
of Geometric Objects.”</span> toblerity.org. <a href="https://github.com/Toblerity/Shapely">https://github.com/Toblerity/Shapely</a>.
</div>
<div id="ref-hagberg2008exploring" class="csl-entry" role="doc-biblioentry">
Hagberg, Aric, Pieter Swart, and Daniel S Chult. 2008. <span>“Exploring
Network Structure, Dynamics, and Function Using NetworkX.”</span> Los
Alamos National Lab.(LANL), Los Alamos, NM (United States).
</div>
<div id="ref-2020NumPy-Array" class="csl-entry" role="doc-biblioentry">
Harris, Charles R., K. Jarrod Millman, Stéfan J van der Walt, Ralf
Gommers, Pauli Virtanen, David Cournapeau, Eric Wieser, et al. 2020.
<span>“Array Programming with <span>NumPy</span>.”</span>
<em>Nature</em> 585: 357–62. <a href="https://doi.org/10.1038/s41586-020-2649-2">https://doi.org/10.1038/s41586-020-2649-2</a>.
</div>
<div id="ref-harrisRelativeRatesSurface2016" class="csl-entry" role="doc-biblioentry">
Harris, Leigh K., and Julie A. Theriot. 2016. <span>“Relative
<span>Rates</span> of <span>Surface</span> and <span>Volume Synthesis
Set Bacterial Cell Size</span>.”</span> <em>Cell</em> 165 (6): 1479–92.
<a href="https://doi.org/10.1016/j.cell.2016.05.045">https://doi.org/10.1016/j.cell.2016.05.045</a>.
</div>
<div id="ref-heinrich2015" class="csl-entry" role="doc-biblioentry">
Heinrich, Kristina, David J. Leslie, and Kristina Jonas. 2015.
<span>“Modulation of Bacterial Proliferation as a Survival
Strategy.”</span> In, 127–71. Elsevier. <a href="https://doi.org/10.1016/bs.aambs.2015.02.004">https://doi.org/10.1016/bs.aambs.2015.02.004</a>.
</div>
<div id="ref-R-fs" class="csl-entry" role="doc-biblioentry">
Hester, Jim, Hadley Wickham, and Gábor Csárdi. 2021. <em>Fs:
Cross-Platform File System Operations Based on Libuv</em>. <a href="https://CRAN.R-project.org/package=fs">https://CRAN.R-project.org/package=fs</a>.
</div>
<div id="ref-hotelling1936" class="csl-entry" role="doc-biblioentry">
Hotelling, Harold. 1936. <span>“Relations Between Two Sets of
Variates.”</span> <em>Biometrika</em> 28 (3/4): 321. <a href="https://doi.org/10.2307/2333955">https://doi.org/10.2307/2333955</a>.
</div>
<div id="ref-hunter2007matplotlib" class="csl-entry" role="doc-biblioentry">
Hunter, John D. 2007. <span>“Matplotlib: A 2D Graphics
Environment.”</span> <em>Computing in Science &amp; Engineering</em> 9
(3): 90–95.
</div>
<div id="ref-R-embed" class="csl-entry" role="doc-biblioentry">
Hvitfeldt, Emil, and Max Kuhn. 2022. <em>Embed: Extra Recipes for
Encoding Predictors</em>. <a href="https://CRAN.R-project.org/package=embed">https://CRAN.R-project.org/package=embed</a>.
</div>
<div id="ref-jaimes-lizcano2014" class="csl-entry" role="doc-biblioentry">
Jaimes-Lizcano, Yuly A., Dayton D. Hunn, and Kyriakos D. Papadopoulos.
2014. <span>“Filamentous Escherichia Coli Cells Swimming in Tapered
Microcapillaries.”</span> <em>Research in Microbiology</em> 165 (3):
166–74. <a href="https://doi.org/10.1016/j.resmic.2014.01.007">https://doi.org/10.1016/j.resmic.2014.01.007</a>.
</div>
<div id="ref-Joseleau-Petit1999" class="csl-entry" role="doc-biblioentry">
Joseleau-Petit, Daniel̀e, Daniel Vinella, and Richard D’Ari. 1999.
<span>“Metabolic Alarms and Cell Division in Escherichia Coli.”</span>
<em>Journal of Bacteriology</em> 181 (1): 9–14. <a href="https://doi.org/10.1128/jb.181.1.9-14.1999">https://doi.org/10.1128/jb.181.1.9-14.1999</a>.
</div>
<div id="ref-justiceFilamentationEscherichiaColi2006" class="csl-entry" role="doc-biblioentry">
Justice, S. S., D. A. Hunstad, P. C. Seed, and S. J. Hultgren. 2006.
<span>“Filamentation by <span>Escherichia</span> Coli Subverts Innate
Defenses During Urinary Tract Infection.”</span> <em>Proceedings of the
National Academy of Sciences</em> 103 (52): 19884–89. <a href="https://doi.org/10.1073/pnas.0606329104">https://doi.org/10.1073/pnas.0606329104</a>.
</div>
<div id="ref-justiceMorphologicalPlasticityBacterial2008" class="csl-entry" role="doc-biblioentry">
Justice, Sheryl S., David A. Hunstad, Lynette Cegelski, and Scott J.
Hultgren. 2008. <span>“Morphological Plasticity as a Bacterial Survival
Strategy.”</span> <em>Nature Reviews Microbiology</em> 6 (2): 162–68. <a href="https://doi.org/10.1038/nrmicro1820">https://doi.org/10.1038/nrmicro1820</a>.
</div>
<div id="ref-R-ggpubr" class="csl-entry" role="doc-biblioentry">
Kassambara, Alboukadel. 2020. <em>Ggpubr: Ggplot2 Based Publication
Ready Plots</em>. <a href="https://rpkgs.datanovia.com/ggpubr/">https://rpkgs.datanovia.com/ggpubr/</a>.
</div>
<div id="ref-R-ggdist" class="csl-entry" role="doc-biblioentry">
Kay, Matthew. 2022. <em>Ggdist: Visualizations of Distributions and
Uncertainty</em>. <a href="https://CRAN.R-project.org/package=ggdist">https://CRAN.R-project.org/package=ggdist</a>.
</div>
<div id="ref-kroemer2008" class="csl-entry" role="doc-biblioentry">
Kroemer, G, L Galluzzi, P Vandenabeele, J Abrams, E S Alnemri, E H
Baehrecke, M V Blagosklonny, et al. 2008. <span>“Classification of Cell
Death: Recommendations of the Nomenclature Committee on Cell Death
2009.”</span> <em>Cell Death &amp; Differentiation</em> 16 (1): 3–11. <a href="https://doi.org/10.1038/cdd.2008.150">https://doi.org/10.1038/cdd.2008.150</a>.
</div>
<div id="ref-R-tidymodels" class="csl-entry" role="doc-biblioentry">
Kuhn, Max, and Hadley Wickham. 2022. <em>Tidymodels: Easily Install and
Load the Tidymodels Packages</em>. <a href="https://CRAN.R-project.org/package=tidymodels">https://CRAN.R-project.org/package=tidymodels</a>.
</div>
<div id="ref-lambertBacterialResistanceAntibiotics2005" class="csl-entry" role="doc-biblioentry">
Lambert, P. 2005. <span>“Bacterial Resistance to Antibiotics:
<span>Modified</span> Target Sites.”</span> <em>Advanced Drug Delivery
Reviews</em> 57 (10): 1471–85. <a href="https://doi.org/10.1016/j.addr.2005.04.003">https://doi.org/10.1016/j.addr.2005.04.003</a>.
</div>
<div id="ref-mcinnes2018umap" class="csl-entry" role="doc-biblioentry">
McInnes, Leland, John Healy, and James Melville. 2018. <span>“Umap:
Uniform Manifold Approximation and Projection for Dimension
Reduction.”</span> <em>arXiv Preprint arXiv:1802.03426</em>.
</div>
<div id="ref-mckinney2010data" class="csl-entry" role="doc-biblioentry">
McKinney, Wes et al. 2010. <span>“Data Structures for Statistical
Computing in Python.”</span> In <em>Proceedings of the 9th Python in
Science Conference</em>, 445:51–56. Austin, TX.
</div>
<div id="ref-meldrumFacultyOpinionsRecommendation2005" class="csl-entry" role="doc-biblioentry">
Meldrum, Deirdre. 2005. <span>“Faculty <span>Opinions</span>
Recommendation of <span>Bacterial</span> Persistence as a Phenotypic
Switch.”</span> <a href="https://doi.org/10.3410/f.1021664.264823">https://doi.org/10.3410/f.1021664.264823</a>.
</div>
<div id="ref-millerSOSResponseInduction2004" class="csl-entry" role="doc-biblioentry">
Miller, C. 2004. <span>“<span>SOS Response Induction</span> by
-<span>Lactams</span> and <span>Bacterial Defense Against Antibiotic
Lethality</span>.”</span> <em>Science</em> 305 (5690): 1629–31. <a href="https://doi.org/10.1126/science.1101630">https://doi.org/10.1126/science.1101630</a>.
</div>
<div id="ref-million-weaverMechanismsPlasmidSegregation2014" class="csl-entry" role="doc-biblioentry">
Million-Weaver, Samuel, and Manel Camps. 2014. <span>“Mechanisms of
Plasmid Segregation: <span>Have</span> Multicopy Plasmids Been
Overlooked?”</span> <em>Plasmid</em> 75 (September): 27–36. <a href="https://doi.org/10.1016/j.plasmid.2014.07.002">https://doi.org/10.1016/j.plasmid.2014.07.002</a>.
</div>
<div id="ref-moger-reischer2019" class="csl-entry" role="doc-biblioentry">
Moger-Reischer, Roy Z., and Jay T. Lennon. 2019. <span>“Microbial Ageing
and Longevity.”</span> <em>Nature Reviews Microbiology</em> 17 (11):
679–90. <a href="https://doi.org/10.1038/s41579-019-0253-y">https://doi.org/10.1038/s41579-019-0253-y</a>.
</div>
<div id="ref-mückl2018" class="csl-entry" role="doc-biblioentry">
Mückl, Andrea, Matthaeus Schwarz-Schilling, Katrin Fischer, and
Friedrich C. Simmel. 2018. <span>“Filamentation and Restoration of
Normal Growth in Escherichia Coli Using a Combined CRISPRi
sgRNA/Antisense RNA Approach.”</span> Edited by Mark Isalan. <em>PLOS
ONE</em> 13 (9): e0198058. <a href="https://doi.org/10.1371/journal.pone.0198058">https://doi.org/10.1371/journal.pone.0198058</a>.
</div>
<div id="ref-R-here" class="csl-entry" role="doc-biblioentry">
Müller, Kirill. 2020. <em>Here: A Simpler Way to Find Your Files</em>.
<a href="https://CRAN.R-project.org/package=here">https://CRAN.R-project.org/package=here</a>.
</div>
<div id="ref-neidhardtBacterialGrowthConstant1999" class="csl-entry" role="doc-biblioentry">
Neidhardt, Frederick C. 1999. <span>“Bacterial <span>Growth</span>:
<span>Constant Obsession</span> with
<span><em>dN</em></span><span><em>/Dt</em></span>.”</span> <em>Journal
of Bacteriology</em> 181 (24): 7405–8. <a href="https://doi.org/10.1128/JB.181.24.7405-7408.1999">https://doi.org/10.1128/JB.181.24.7405-7408.1999</a>.
</div>
<div id="ref-pearson1901" class="csl-entry" role="doc-biblioentry">
Pearson, Karl. 1901. <span>“LIII. On Lines and Planes of Closest Fit to
Systems of Points in Space.”</span> <em>The London, Edinburgh, and
Dublin Philosophical Magazine and Journal of Science</em> 2 (11):
559–72. <a href="https://doi.org/10.1080/14786440109462720">https://doi.org/10.1080/14786440109462720</a>.
</div>
<div id="ref-R-patchwork" class="csl-entry" role="doc-biblioentry">
Pedersen, Thomas Lin. 2020. <em>Patchwork: The Composer of Plots</em>.
<a href="https://CRAN.R-project.org/package=patchwork">https://CRAN.R-project.org/package=patchwork</a>.
</div>
<div id="ref-perez2007ipython" class="csl-entry" role="doc-biblioentry">
Pérez, Fernando, and Brian E Granger. 2007. <span>“IPython: A System for
Interactive Scientific Computing.”</span> <em>Computing in Science &amp;
Engineering</em> 9 (3).
</div>
<div id="ref-R-base" class="csl-entry" role="doc-biblioentry">
R Core Team. 2022. <em>R: A Language and Environment for Statistical
Computing</em>. Vienna, Austria: R Foundation for Statistical Computing.
<a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-rackauckas2017adaptive" class="csl-entry" role="doc-biblioentry">
Rackauckas, Christopher, and Qing Nie. 2017a. <span>“Adaptive Methods
for Stochastic Differential Equations via Natural Embeddings and
Rejection Sampling with Memory.”</span> <em>Discrete and Continuous
Dynamical Systems. Series B</em> 22 (7): 2731.
</div>
<div id="ref-rackauckas2017differentialequations" class="csl-entry" role="doc-biblioentry">
———. 2017b. <span>“Differentialequations.jl–a Performant and
Feature-Rich Ecosystem for Solving Differential Equations in
Julia.”</span> <em>Journal of Open Research Software</em> 5 (1).
</div>
<div id="ref-rackauckas_stability-optimized_2018" class="csl-entry" role="doc-biblioentry">
———. 2018. <span>“Stability-<span>Optimized</span> <span>High</span>
<span>Order</span> <span>Methods</span> and <span>Stiffness</span>
<span>Detection</span> for <span>Pathwise</span> <span>Stiff</span>
<span>Stochastic</span> <span>Differential</span>
<span>Equations</span>.”</span> <em>arXiv:1804.04344 [Math]</em>. <a href="http://arxiv.org/abs/1804.04344">http://arxiv.org/abs/1804.04344</a>.
</div>
<div id="ref-R-tidytext" class="csl-entry" role="doc-biblioentry">
Robinson, David, and Julia Silge. 2022. <em>Tidytext: Text Mining Using
Dplyr, Ggplot2, and Other Tidy Tools</em>. <a href="https://github.com/juliasilge/tidytext">https://github.com/juliasilge/tidytext</a>.
</div>
<div id="ref-rocklin2015dask" class="csl-entry" role="doc-biblioentry">
Rocklin, Matthew. 2015. <span>“Dask: Parallel Computation with Blocked
Algorithms and Task Scheduling.”</span> In <em>Proceedings of the 14th
Python in Science Conference</em>. 130-136. Citeseer.
</div>
<div id="ref-roostalu2008" class="csl-entry" role="doc-biblioentry">
Roostalu, Johanna, Arvi Jõers, Hannes Luidalepp, Niilo Kaldalu, and
Tanel Tenson. 2008. <span>“Cell Division in Escherichia Colicultures
Monitored at Single Cell Resolution.”</span> <em>BMC Microbiology</em> 8
(1). <a href="https://doi.org/10.1186/1471-2180-8-68">https://doi.org/10.1186/1471-2180-8-68</a>.
</div>
<div id="ref-sanmillan2016" class="csl-entry" role="doc-biblioentry">
San Millan, Alvaro, Jose Antonio Escudero, Danna R. Gifford, Didier
Mazel, and R. Craig MacLean. 2016. <span>“Multicopy Plasmids Potentiate
the Evolution of Antibiotic Resistance in Bacteria.”</span> <em>Nature
Ecology &amp; Evolution</em> 1 (1). <a href="https://doi.org/10.1038/s41559-016-0010">https://doi.org/10.1038/s41559-016-0010</a>.
</div>
<div id="ref-satoOuterMembranePermeability1991" class="csl-entry" role="doc-biblioentry">
Sato, Kazuhito, and Taiji Nakae. 1991. <span>“Outer Membrane
Permeability of <span><em>Acinetobacter</em></span><span>
<em>Calcoaceticus</em></span> and Its Implication in Antibiotic
Resistance.”</span> <em>Journal of Antimicrobial Chemotherapy</em> 28
(1): 35–45. <a href="https://doi.org/10.1093/jac/28.1.35">https://doi.org/10.1093/jac/28.1.35</a>.
</div>
<div id="ref-schaechterDependencyMediumTemperature1958" class="csl-entry" role="doc-biblioentry">
Schaechter, M., O. MaalOe, and N. O. Kjeldgaard. 1958. <span>“Dependency
on <span>Medium</span> and <span>Temperature</span> of <span>Cell
Size</span> and <span>Chemical Composition</span> During <span>Balanced
Growth</span> of <span>Salmonella</span> Typhimurium.”</span>
<em>Journal of General Microbiology</em> 19 (3): 592–606. <a href="https://doi.org/10.1099/00221287-19-3-592">https://doi.org/10.1099/00221287-19-3-592</a>.
</div>
<div id="ref-schaechterGrowthCellNuclear1962" class="csl-entry" role="doc-biblioentry">
Schaechter, M., J. P. Williamson, J. R. Hood, and A. L. Koch. 1962.
<span>“Growth, <span>Cell</span> and <span>Nuclear Divisions</span> in
Some <span>Bacteria</span>.”</span> <em>Journal of General
Microbiology</em> 29 (3): 421–34. <a href="https://doi.org/10.1099/00221287-29-3-421">https://doi.org/10.1099/00221287-29-3-421</a>.
</div>
<div id="ref-schermelleh2019" class="csl-entry" role="doc-biblioentry">
Schermelleh, Lothar, Alexia Ferrand, Thomas Huser, Christian Eggeling,
Markus Sauer, Oliver Biehlmaier, and Gregor P. C. Drummen. 2019.
<span>“Super-Resolution Microscopy Demystified.”</span> <em>Nature Cell
Biology</em> 21 (1): 72–84. <a href="https://doi.org/10.1038/s41556-018-0251-8">https://doi.org/10.1038/s41556-018-0251-8</a>.
</div>
<div id="ref-R-GGally" class="csl-entry" role="doc-biblioentry">
Schloerke, Barret, Di Cook, Joseph Larmarange, Francois Briatte, Moritz
Marbach, Edwin Thoen, Amos Elberg, and Jason Crowley. 2021. <em>GGally:
Extension to Ggplot2</em>. <a href="https://CRAN.R-project.org/package=GGally">https://CRAN.R-project.org/package=GGally</a>.
</div>
<div id="ref-schneider2012" class="csl-entry" role="doc-biblioentry">
Schneider, Caroline A, Wayne S Rasband, and Kevin W Eliceiri. 2012.
<span>“NIH Image to ImageJ: 25 Years of Image Analysis.”</span>
<em>Nature Methods</em> 9 (7): 671–75. <a href="https://doi.org/10.1038/nmeth.2089">https://doi.org/10.1038/nmeth.2089</a>.
</div>
<div id="ref-R-plotly" class="csl-entry" role="doc-biblioentry">
Sievert, Carson, Chris Parmer, Toby Hocking, Scott Chamberlain, Karthik
Ram, Marianne Corvellec, and Pedro Despouy. 2021. <em>Plotly: Create
Interactive Web Graphics via Plotly.js</em>. <a href="https://CRAN.R-project.org/package=plotly">https://CRAN.R-project.org/package=plotly</a>.
</div>
<div id="ref-sliusarenkoHighthroughputSubpixelPrecision2011" class="csl-entry" role="doc-biblioentry">
Sliusarenko, Oleksii, Jennifer Heinritz, Thierry Emonet, and Christine
Jacobs-Wagner. 2011. <span>“High-Throughput, Subpixel Precision Analysis
of Bacterial Morphogenesis and Intracellular Spatio-Temporal Dynamics:
<span>Quantitative</span> Analysis of Spatio-Temporal Dynamics.”</span>
<em>Molecular Microbiology</em> 80 (3): 612–27. <a href="https://doi.org/10.1111/j.1365-2958.2011.07579.x">https://doi.org/10.1111/j.1365-2958.2011.07579.x</a>.
</div>
<div id="ref-Smith2018" class="csl-entry" role="doc-biblioentry">
Smith, Kevin, Filippo Piccinini, Tamas Balassa, Krisztian Koos, Tivadar
Danka, Hossein Azizpour, and Peter Horvath. 2018. <span>“Phenotypic
Image Analysis Software Tools for Exploring and Understanding Big Image
Data from Cell-Based Assays.”</span> <em>Cell Systems</em> 6 (6):
636–53. <a href="https://doi.org/10.1016/j.cels.2018.06.001">https://doi.org/10.1016/j.cels.2018.06.001</a>.
</div>
<div id="ref-sofroniew2021a" class="csl-entry" role="doc-biblioentry">
Sofroniew, Nicholas, Talley Lambert, Kira Evans, Juan Nunez-Iglesias,
Grzegorz Bokota, Gonzalo Peña-Castellanos, Philip Winston, et al. 2021.
<em>Napari/Napari: 0.4.12rc2</em>. Zenodo. <a href="https://doi.org/10.5281/ZENODO.3555620">https://doi.org/10.5281/ZENODO.3555620</a>.
</div>
<div id="ref-Specht2017" class="csl-entry" role="doc-biblioentry">
Specht, Elizabeth A., Esther Braselmann, and Amy E. Palmer. 2017.
<span>“A Critical and Comparative Review of Fluorescent Tools for
Live-Cell Imaging.”</span> <em>Annual Review of Physiology</em> 79 (1):
93–117. <a href="https://doi.org/10.1146/annurev-physiol-022516-034055">https://doi.org/10.1146/annurev-physiol-022516-034055</a>.
</div>
<div id="ref-taheri-araghiCellSizeControlHomeostasis2017" class="csl-entry" role="doc-biblioentry">
Taheri-Araghi, Sattar, Serena Bradde, John T. Sauls, Norbert S. Hill,
Petra Anne Levin, Johan Paulsson, Massimo Vergassola, and Suckjoon Jun.
2017. <span>“Cell-<span>Size Control</span> and <span>Homeostasis</span>
in <span>Bacteria</span>.”</span> <em>Current Biology</em> 27 (9): 1392.
<a href="https://doi.org/10.1016/j.cub.2017.04.028">https://doi.org/10.1016/j.cub.2017.04.028</a>.
</div>
<div id="ref-trevors2012" class="csl-entry" role="doc-biblioentry">
Trevors, J.T. 2012. <span>“Can Dead Bacterial Cells Be Defined and Are
Genes Expressed After Cell Death?”</span> <em>Journal of Microbiological
Methods</em> 90 (1): 25–28. <a href="https://doi.org/10.1016/j.mimet.2012.04.004">https://doi.org/10.1016/j.mimet.2012.04.004</a>.
</div>
<div id="ref-turing1950" class="csl-entry" role="doc-biblioentry">
TURING, A. M. 1950. <span>“I.<span></span>COMPUTING MACHINERY AND
INTELLIGENCE.”</span> <em>Mind</em> LIX (236): 433–60. <a href="https://doi.org/10.1093/mind/lix.236.433">https://doi.org/10.1093/mind/lix.236.433</a>.
</div>
<div id="ref-ursellRapidPreciseQuantification2017" class="csl-entry" role="doc-biblioentry">
Ursell, Tristan, Timothy K. Lee, Daisuke Shiomi, Handuo Shi, Carolina
Tropini, Russell D. Monds, Alexandre Colavin, et al. 2017. <span>“Rapid,
Precise Quantification of Bacterial Cellular Dimensions Across a
Genomic-Scale Knockout Library.”</span> <em>BMC Biology</em> 15 (1): 17.
<a href="https://doi.org/10.1186/s12915-017-0348-8">https://doi.org/10.1186/s12915-017-0348-8</a>.
</div>
<div id="ref-R-renv" class="csl-entry" role="doc-biblioentry">
Ushey, Kevin. 2022. <em>Renv: Project Environments</em>. <a href="https://rstudio.github.io/renv/">https://rstudio.github.io/renv/</a>.
</div>
<div id="ref-vanderwalt2014" class="csl-entry" role="doc-biblioentry">
van der Walt, Stéfan, Johannes L. Schönberger, Juan Nunez-Iglesias,
François Boulogne, Joshua D. Warner, Neil Yager, Emmanuelle Gouillart,
and Tony Yu. 2014. <span>“Scikit-Image: Image Processing in
Python.”</span> <em>PeerJ</em> 2 (June): e453. <a href="https://doi.org/10.7717/peerj.453">https://doi.org/10.7717/peerj.453</a>.
</div>
<div id="ref-van1995python" class="csl-entry" role="doc-biblioentry">
Van Rossum, Guido. 2020. <em>The Python Library Reference, Release
3.8.2</em>. Python Software Foundation.
</div>
<div id="ref-10.5555/1593511" class="csl-entry" role="doc-biblioentry">
Van Rossum, Guido, and Fred L. Drake. 2009. <em>Python 3 Reference
Manual</em>. Scotts Valley, CA: CreateSpace.
</div>
<div id="ref-vanvalen2016" class="csl-entry" role="doc-biblioentry">
Van Valen, David A., Takamasa Kudo, Keara M. Lane, Derek N. Macklin,
Nicolas T. Quach, Mialy M. DeFelice, Inbal Maayan, Yu Tanouchi, Euan A.
Ashley, and Markus W. Covert. 2016. <span>“Deep Learning Automates the
Quantitative Analysis of Individual Cells in Live-Cell Imaging
Experiments.”</span> Edited by Martin Meier-Schellersheim. <em>PLOS
Computational Biology</em> 12 (11): e1005177. <a href="https://doi.org/10.1371/journal.pcbi.1005177">https://doi.org/10.1371/journal.pcbi.1005177</a>.
</div>
<div id="ref-verschaffel2002" class="csl-entry" role="doc-biblioentry">
Verschaffel, Lieven, Brian Greer, and Erik de Corte. 2002.
<span>“Everyday Knowledge and Mathematical Modeling of School Word
Problems.”</span> In, 257–76. Springer Netherlands. <a href="https://doi.org/10.1007/978-94-017-3194-2_16">https://doi.org/10.1007/978-94-017-3194-2_16</a>.
</div>
<div id="ref-wang2009" class="csl-entry" role="doc-biblioentry">
Wang, Jue D., and Petra A. Levin. 2009. <span>“Metabolism, Cell Growth
and the Bacterial Cell Cycle.”</span> <em>Nature Reviews
Microbiology</em> 7 (11): 822–27. <a href="https://doi.org/10.1038/nrmicro2202">https://doi.org/10.1038/nrmicro2202</a>.
</div>
<div id="ref-Wang2014" class="csl-entry" role="doc-biblioentry">
Wang, Ying, Hong Wu, Xiaoran Jiang, and Guo-Qiang Chen. 2014.
<span>“Engineering Escherichia Coli for Enhanced Production of
Poly(3-Hydroxybutyrate-Co-4-Hydroxybutyrate) in Larger Cellular
Space.”</span> <em>Metabolic Engineering</em> 25 (September): 183–93. <a href="https://doi.org/10.1016/j.ymben.2014.07.010">https://doi.org/10.1016/j.ymben.2014.07.010</a>.
</div>
<div id="ref-Wang2014a" class="csl-entry" role="doc-biblioentry">
Wang, Ying, Jin Yin, and Guo-Qiang Chen. 2014.
<span>“Polyhydroxyalkanoates, Challenges and Opportunities.”</span>
<em>Current Opinion in Biotechnology</em> 30 (December): 59–65. <a href="https://doi.org/10.1016/j.copbio.2014.06.001">https://doi.org/10.1016/j.copbio.2014.06.001</a>.
</div>
<div id="ref-webberImportanceEffluxPumps2003" class="csl-entry" role="doc-biblioentry">
Webber, M. A. 2003. <span>“The Importance of Efflux Pumps in Bacterial
Antibiotic Resistance.”</span> <em>Journal of Antimicrobial
Chemotherapy</em> 51 (1): 9–11. <a href="https://doi.org/10.1093/jac/dkg050">https://doi.org/10.1093/jac/dkg050</a>.
</div>
<div id="ref-white2021" class="csl-entry" role="doc-biblioentry">
White, John Myles, Bogumił Kamiński, Powerdistribution, Milan
Bouchet-Valat, Sean Garborg, Jacob Quinn, Simon Kornblith, et al. 2021.
<em>JuliaData/DataFrames.jl: V1.2.2</em>. Zenodo. <a href="https://doi.org/10.5281/ZENODO.3376177">https://doi.org/10.5281/ZENODO.3376177</a>.
</div>
<div id="ref-R-stringr" class="csl-entry" role="doc-biblioentry">
Wickham, Hadley. 2019. <em>Stringr: Simple, Consistent Wrappers for
Common String Operations</em>. <a href="https://CRAN.R-project.org/package=stringr">https://CRAN.R-project.org/package=stringr</a>.
</div>
<div id="ref-R-tidyverse" class="csl-entry" role="doc-biblioentry">
———. 2022. <em>Tidyverse: Easily Install and Load the Tidyverse</em>. <a href="https://CRAN.R-project.org/package=tidyverse">https://CRAN.R-project.org/package=tidyverse</a>.
</div>
<div id="ref-R-sessioninfo" class="csl-entry" role="doc-biblioentry">
Wickham, Hadley, Winston Chang, Robert Flight, Kirill Müller, and Jim
Hester. 2021. <em>Sessioninfo: R Session Information</em>. <a href="https://CRAN.R-project.org/package=sessioninfo">https://CRAN.R-project.org/package=sessioninfo</a>.
</div>
<div id="ref-wrightBacterialResistanceAntibiotics2005" class="csl-entry" role="doc-biblioentry">
Wright, G. 2005. <span>“Bacterial Resistance to Antibiotics:
<span>Enzymatic</span> Degradation and Modification.”</span>
<em>Advanced Drug Delivery Reviews</em> 57 (10): 1451–70. <a href="https://doi.org/10.1016/j.addr.2005.04.002">https://doi.org/10.1016/j.addr.2005.04.002</a>.
</div>
<div id="ref-R-knitr" class="csl-entry" role="doc-biblioentry">
Xie, Yihui. 2022. <em>Knitr: A General-Purpose Package for Dynamic
Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-yin2012" class="csl-entry" role="doc-biblioentry">
Yin, Huabing, and Damian Marshall. 2012. <span>“Microfluidics for Single
Cell Analysis.”</span> <em>Current Opinion in Biotechnology</em> 23 (1):
110–19. <a href="https://doi.org/10.1016/j.copbio.2011.11.002">https://doi.org/10.1016/j.copbio.2011.11.002</a>.
</div>
</div>


<!-- -->


</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./98-appendices.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Appendix</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># References {.unnumbered}</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>:::</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>